<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Wrangle and analyze IMPACT and TCGA mutation data  • gnomeR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Wrangle and analyze IMPACT and TCGA mutation data ">
<meta property="og:description" content="Prepare raw mutation, cna, and fusion data for analysis. Annotate OncoKB, summarize maf files, visualize results and more. ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">gnomeR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/MSKCC-Epi-Bio/gnomeR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">

<div class="section level1">
<div class="page-header"><h1 id="gnomer">gnomeR<a class="anchor" aria-label="anchor" href="#gnomer"></a>
</h1></div>
<!-- badges: start -->

<!-- badges: end -->
<p><font size="5"><span class="emoji" data-emoji="bangbang">‼️</span> <span class="emoji" data-emoji="warning">⚠️</span> <strong>NOTE: This package is currently under active development with a new stable release expected April 30th, 2022. For code written before 2022-03-23, please use the previous stable version (v1.1.0)</strong><span class="emoji" data-emoji="warning">⚠️</span><span class="emoji" data-emoji="bangbang">‼️</span> </font></p>
<p>You can install the pre-2022-03-23 version with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">remotes</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">'MSKCC-Epi-Bio/gnomeR@v1.1.0'</span><span class="op">)</span></code></pre></div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of <code>gnomeR</code> from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># install.packages("devtools")</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"MSKCC-Epi-Bio/gnomeR"</span><span class="op">)</span></code></pre></div>
<p>Along with its companion package for cbioPortal data download:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"karissawhiting/cbioportalr"</span><span class="op">)</span></code></pre></div>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>the <code>gnomeR</code> package provides a consistent framework for genetic data processing, visualization and analysis. This is primarily targeted to IMPACT datasets but can also be applied to any genomic data provided by CbioPortal. With {gnomeR} and {cbioportalR} you can:</p>
<ul>
<li>
<a href="https://github.com/karissawhiting/cbioportalR" class="external-link"><strong>Download and gather data from CbioPortal</strong></a> - Pull from cBioPortal data base by study ID or sample ID.</li>
<li>
<strong>OncoKB annotate data</strong> - Filter genomic data for known oncogenic alterations.</li>
<li>
<strong>Process genomic data</strong> - Process retrieved mutation/maf, fusions, copy-number alteration, and segmentation data (when available) into an analysis-ready formats.</li>
<li>
<strong>Visualize processed data</strong> - Create OncoPrints, heatmaps and summary plots from processed data.</li>
<li>
<strong>Analyzing processed data</strong>- Analyze associations between genomic variables and clinical variables or outcomes with summary tables, advanced visualizations, and univariate models.</li>
</ul>
</div>
<div class="section level2">
<h2 id="getting-set-up">Getting Set up<a class="anchor" aria-label="anchor" href="#getting-set-up"></a>
</h2>
<p>{gnomeR} works with any genomic data that follows cBioPortal guidelines for <a href="https://docs.cbioportal.org/5.1-data-loading/data-loading/file-formats#data-file-5" class="external-link">mutation</a>, <a href="https://docs.cbioportal.org/5.1-data-loading/data-loading/file-formats#discrete-copy-number-data" class="external-link">CNA</a>, or <a href="https://docs.cbioportal.org/5.1-data-loading/data-loading/file-formats#structural-variant-data" class="external-link">fusion</a> data file formats.</p>
<p>If you wish to pull the data directly from CbioPortal, see how to get set up with credentials with the <a href="https://karissawhiting.github.io/cbioportalR/" class="external-link">{cBioPortalR}</a> package.</p>
<p>If you want to oncoKB annotate your data, you need to acquire credentials from the <a href="https://www.oncokb.org/apiAccess" class="external-link">oncoKB website</a>.</p>
</div>
<div class="section level2">
<h2 id="processing-genomic-data">Processing Genomic Data<a class="anchor" aria-label="anchor" href="#processing-genomic-data"></a>
</h2>
<p>The below examples uses the data sets <code>mut</code>, <code>fusion</code>, <code>cna</code> which were pulled from cBioPortal and are included in the package as example data sets. We will sample 100 samples for examples:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>

<span class="va">mut</span> <span class="op">&lt;-</span> <span class="fu">gnomeR</span><span class="fu">::</span><span class="va"><a href="reference/mut.html">mut</a></span>
<span class="va">cna</span> <span class="op">&lt;-</span> <span class="fu">gnomeR</span><span class="fu">::</span><span class="va"><a href="reference/cna.html">cna</a></span>
<span class="va">fusion</span> <span class="op">&lt;-</span> <span class="fu">gnomeR</span><span class="fu">::</span><span class="va"><a href="reference/fusion.html">fusion</a></span>

<span class="va">un</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">mut</span><span class="op">$</span><span class="va">Tumor_Sample_Barcode</span><span class="op">)</span>
<span class="va">sample_patients</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">un</span>, size <span class="op">=</span> <span class="fl">100</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></code></pre></div>
<p>The main data processing function is <code>gene_binary()</code> which takes mutation, CNA and fusion files as input, and outputs a binary matrix of N rows (number of samples) by M genes included in the data set. We can specify which patients are included which will force all patients in resulting dataframe, even if they have no alterations.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gen_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/create_gene_binary.html">create_gene_binary</a></span><span class="op">(</span>samples <span class="op">=</span> <span class="va">sample_patients</span>,
                         maf <span class="op">=</span> <span class="va">mut</span>,
                         fusion <span class="op">=</span> <span class="va">fusion</span>,
                         cna <span class="op">=</span> <span class="va">cna</span>, rm_empty <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span>

<span class="fu">kable</span><span class="op">(</span><span class="va">gen_dat</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">5</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span>, row.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">OSBPL9.fus</th>
<th align="right">ALK.fus</th>
<th align="right">EML4.fus</th>
<th align="right">MLL3.fus</th>
<th align="right">BRCA2.fus</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">P-0010604-T01-IM5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">P-0002651-T01-IM3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">P-0000270-T01-IM3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">P-0002915-T01-IM3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">P-0011099-T01-IM5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="visualize">Visualize<a class="anchor" aria-label="anchor" href="#visualize"></a>
</h2>
<p>You can visualize your processed and raw alteration data sets using {gnomeR}’s many data visualization functions.</p>
<p>Quickly visualize mutation characteristics with <code><a href="reference/ggvarclass.html">ggvarclass()</a></code>, <code><a href="reference/ggvartype.html">ggvartype()</a></code>, <code><a href="reference/ggsnvclass.html">ggsnvclass()</a></code>, <code><a href="reference/ggsamplevar.html">ggsamplevar()</a></code>, <code><a href="reference/ggtopgenes.html">ggtopgenes()</a></code>, <code><a href="reference/gggenecor.html">gggenecor()</a></code>, and <code><a href="reference/ggcomut.html">ggcomut()</a></code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu"><a href="reference/ggvarclass.html">ggvarclass</a></span><span class="op">(</span>mutation <span class="op">=</span> <span class="va">mut</span><span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-5-1.png" width="100%"></p>
<p>The wrapper function <code><a href="reference/mutation_viz.html">mutation_viz()</a></code> will generate a list with the most commonly used plot which you can pass into a plot assembly framework like {patchwork}:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">list_of_plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/mutation_viz.html">mutation_viz</a></span><span class="op">(</span>mutation <span class="op">=</span> <span class="va">mut</span><span class="op">)</span>

<span class="fu">patchwork</span><span class="fu">::</span><span class="fu">wrap_plots</span><span class="op">(</span><span class="va">list_of_plots</span>, ncol <span class="op">=</span> <span class="fl">2</span><span class="op">)</span> <span class="op">&amp;</span> 
  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>text <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">8</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-6-1.png" width="100%"></p>
<p><code><a href="reference/ggheatmap.html">ggheatmap()</a></code> allows you to visualize mutation, CNA and fusion data all in one heatmap:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co"># filter only those with alterations</span>
<span class="va">gen_dat</span> <span class="op">&lt;-</span> <span class="va">gen_dat</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span>
  <span class="fu">rowwise</span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> 
  <span class="fu">mutate</span><span class="op">(</span>total_alterations <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="fu">c_across</span><span class="op">(</span><span class="fu">everything</span><span class="op">(</span><span class="op">)</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="https://rdrr.io/r/stats/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">total_alterations</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span>
  <span class="fu">select</span><span class="op">(</span><span class="op">-</span><span class="va">total_alterations</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span>
  <span class="fu">ungroup</span><span class="op">(</span><span class="op">)</span>

<span class="va">select_gen_dat</span> <span class="op">&lt;-</span> <span class="va">gen_dat</span><span class="op">[</span>, <span class="op">(</span><span class="fu">purrr</span><span class="fu">::</span><span class="fu"><a href="https://purrr.tidyverse.org/reference/map.html" class="external-link">map</a></span><span class="op">(</span><span class="va">gen_dat</span>, <span class="op">~</span><span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">.x</span>, na.rm <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> <span class="op">&gt;</span> <span class="fl">0</span><span class="op">)</span><span class="op">]</span>

<span class="fu"><a href="reference/ggheatmap.html">ggheatmap</a></span><span class="op">(</span><span class="va">select_gen_dat</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">50</span><span class="op">]</span><span class="op">)</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-7-1.png" width="100%"></p>
</div>
<div class="section level2">
<h2 id="summarize--analyze">Summarize &amp; Analyze<a class="anchor" aria-label="anchor" href="#summarize--analyze"></a>
</h2>
<p>You can tabulate summarize your genomic data frame using the <code><a href="reference/tbl_genomic.html">tbl_genomic()</a></code> function, a wrapper for <code><a href="https://www.danieldsjoberg.com/gtsummary/reference/tbl_summary.html" class="external-link">gtsummary::tbl_summary()</a></code>.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gen_dat</span> <span class="op">&lt;-</span> <span class="va">gen_dat</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span>
  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>trt_status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pre-trt"</span>, <span class="st">"post-trt"</span><span class="op">)</span>,
       size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">gen_dat</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> </code></pre></div>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">gene_tbl_trt</span> <span class="op">&lt;-</span>  <span class="va">gen_dat</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="reference/tbl_genomic.html">tbl_genomic</a></span><span class="op">(</span>freq_cutoff <span class="op">=</span> <span class="fl">.1</span>, by <span class="op">=</span> <span class="va">trt_status</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span>
  <span class="fu">gtsummary</span><span class="fu">::</span><span class="fu"><a href="https://www.danieldsjoberg.com/gtsummary/reference/add_p.html" class="external-link">add_p</a></span><span class="op">(</span><span class="op">)</span> 

<span class="va">gene_tbl_trt</span></code></pre></div>
<p><img src="reference/figures/gene_tbl_trt.png" width="60%"></p>
<p>Additionally, you can analyze custom pathways, or a set of default gene pathways using <code><a href="reference/add_pathways.html">add_pathways()</a></code>:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">path_by_trt</span> <span class="op">&lt;-</span> <span class="va">gen_dat</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="reference/add_pathways.html">add_pathways</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span>
  <span class="fu">select</span><span class="op">(</span><span class="va">trt_status</span>, <span class="fu">contains</span><span class="op">(</span><span class="st">"pathway_"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span>
  <span class="fu"><a href="reference/tbl_genomic.html">tbl_genomic</a></span><span class="op">(</span>by <span class="op">=</span> <span class="va">trt_status</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span>
  <span class="fu">gtsummary</span><span class="fu">::</span><span class="fu"><a href="https://www.danieldsjoberg.com/gtsummary/reference/add_p.html" class="external-link">add_p</a></span><span class="op">(</span><span class="op">)</span> 


<span class="va">path_by_trt</span></code></pre></div>
<p><img src="reference/figures/path_by_trt.png" width="60%"></p>
</div>
<div class="section level2">
<h2 id="further-analytical-tools">Further analytical tools<a class="anchor" aria-label="anchor" href="#further-analytical-tools"></a>
</h2>
<p>Along with mutation, cna and fusion data, {gnomeR} also allows analysis and visualization of <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5027494/" class="external-link">FACETs</a> data. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5027494/" class="external-link">FACETs</a> is an allele-specific copy number tool and open-source software with a broad application to whole genome, whole-exome, as well as targeted panel sequencing platforms. It is a fully integrated stand-alone pipeline that includes sequencing BAM file post-processing, joint segmentation of total- and allele-specific read counts, and integer copy number calls corrected for tumor purity, ploidy and clonal heterogeneity, with comprehensive output.</p>
<p>You can vizualize this data using <code><a href="reference/facets_heatmap.html">facets_heatmap()</a></code></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">select_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">seg</span><span class="op">$</span><span class="va">ID</span><span class="op">)</span>,  <span class="fl">100</span><span class="op">)</span>
<span class="va">p.heat</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/facets_heatmap.html">facets_heatmap</a></span><span class="op">(</span>seg <span class="op">=</span> <span class="va">seg</span>, samples <span class="op">=</span> <span class="va">select_samples</span>, min_purity <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>
<span class="va">p.heat</span><span class="op">$</span><span class="va">p</span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-13-1.png" width="100%"></p>
<p>The primary goal of <code>gnomeR</code> not being in depth analysis of genomic data but rather reliable, modular and reproducible framework for processing various types of genomic data. For users interested in large scale genomic analytical methods we compiled various packages developed by <a href="https://www.mskcc.org/departments/epidemiology-biostatistics" class="external-link">Department of Epidemiology and Biostatistics</a>, Memorial Sloan-Kettering Cancer Center under an umbrella R package, <a href="https://github.com/AxelitoMartin/genomeVerse" class="external-link">gnomeVerse</a>.</p>
</div>
</div>
<div class="section level1">
<h1 id="contributing">Contributing<a class="anchor" aria-label="anchor" href="#contributing"></a>
</h1>
<p>Please note that the gnomeR project is released with a <a href="https://contributor-covenant.org/version/2/0/CODE_OF_CONDUCT.html" class="external-link">Contributor Code of Conduct</a>. By contributing to this project, you agree to abide by its terms.</p>
<p>Thank you to all contributors!</p>
<p><a href="https://github.com/akriti21" class="external-link">@akriti21</a>, <a href="https://github.com/arorarshi" class="external-link">@arorarshi</a>, <a href="https://github.com/AxelitoMartin" class="external-link">@AxelitoMartin</a>, <a href="https://github.com/carokos" class="external-link">@carokos</a>, <a href="https://github.com/ChristineZ-msk" class="external-link">@ChristineZ-msk</a>, <a href="https://github.com/edrill" class="external-link">@edrill</a>, <a href="https://github.com/jalavery" class="external-link">@jalavery</a>, <a href="https://github.com/jflynn264" class="external-link">@jflynn264</a>, <a href="https://github.com/karissawhiting" class="external-link">@karissawhiting</a>, <a href="https://github.com/michaelcurry1123" class="external-link">@michaelcurry1123</a>, and <a href="https://github.com/mljaniczek" class="external-link">@mljaniczek</a></p>
</div>

  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/MSKCC-Epi-Bio/gnomeR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/MSKCC-Epi-Bio/gnomeR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CODE_OF_CONDUCT.html">Code of conduct</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing gnomeR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Axel Martin <br><small class="roles"> Author </small> <a href="https://orcid.org/YOUR-ORCID-ID" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Karissa Whiting <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-4683-1868" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Arshi Arora <br><small class="roles"> Author </small>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/MSKCC-Epi-Bio/gnomeR/actions" class="external-link"><img src="https://github.com/MSKCC-Epi-Bio/gnomeR/workflows/R-CMD-check/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://app.codecov.io/gh/MSKCC-Epi-Bio/gnomeR?branch=main" class="external-link"><img src="https://codecov.io/gh/MSKCC-Epi-Bio/gnomeR/branch/main/graph/badge.svg" alt="Codecov test coverage"></a></li>
</ul>
</div>

  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Axel Martin, Karissa Whiting, Arshi Arora.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
