<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="Prepare raw mutation, cna, and fusion data for analysis. Annotate OncoKB, summarize maf files, visualize results and more. ">
<title>Wrangle and analyze IMPACT and TCGA mutation data  • gnomeR</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><link href="deps/Atkinson_Hyperlegible-0.4.2/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Wrangle and analyze IMPACT and TCGA mutation data ">
<meta property="og:description" content="Prepare raw mutation, cna, and fusion data for analysis. Annotate OncoKB, summarize maf files, visualize results and more. ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">gnomeR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.1.9001</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/data-processing-vignette.html">Processing Data with {gnomeR}</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MSKCC-Epi-Bio/gnomeR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="gnomer">gnomeR<a class="anchor" aria-label="anchor" href="#gnomer"></a>
</h1></div>
<!-- badges: start -->

<!-- badges: end -->
<p><font size="5"><span class="emoji" data-emoji="bangbang">‼️</span> <span class="emoji" data-emoji="warning">⚠️</span> <strong>NOTE: This package is currently under active development with a new stable release expected November 15th, 2022. For code written before 2022-03-23, please use the previous stable version (v1.1.0)</strong><span class="emoji" data-emoji="warning">⚠️</span><span class="emoji" data-emoji="bangbang">‼️</span> </font></p>
<p>You can install the pre-2022-03-23 version with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">remotes</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">'MSKCC-Epi-Bio/gnomeR@v1.1.0'</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of <code>gnomeR</code> from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"MSKCC-Epi-Bio/gnomeR"</span><span class="op">)</span></span></code></pre></div>
<p>Along with its companion package for cbioPortal data download:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"karissawhiting/cbioportalr"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>the <code>gnomeR</code> package provides a consistent framework for genetic data processing, visualization and analysis. This is primarily targeted to IMPACT datasets but can also be applied to any genomic data provided by cBioPortal. With {gnomeR} and {cbioportalR} you can:</p>
<ul>
<li>
<a href="https://www.karissawhiting.com/cbioportalR/" class="external-link"><strong>Download and gather data from CbioPortal</strong></a> - Pull from cBioPortal data base by study ID or sample ID.</li>
<li>
<strong>OncoKB annotate data</strong> - Filter genomic data for known oncogenic alterations.</li>
<li>
<strong>Process genomic data</strong> - Process retrieved mutation/maf, fusions, copy-number alteration, and segmentation data (when available) into an analysis-ready formats.</li>
<li>
<strong>Visualize processed data</strong> - Create OncoPrints, heatmaps and summary plots from processed data.</li>
<li>
<strong>Analyzing processed data</strong>- Analyze associations between genomic variables and clinical variables or outcomes with summary tables, advanced visualizations, and univariate models.</li>
</ul>
</div>
<div class="section level2">
<h2 id="getting-set-up">Getting Set up<a class="anchor" aria-label="anchor" href="#getting-set-up"></a>
</h2>
<p>{gnomeR} works with any genomic data that follows cBioPortal guidelines for <a href="https://docs.cbioportal.org/5.1-data-loading/data-loading/file-formats#data-file-5" class="external-link">mutation</a>, <a href="https://docs.cbioportal.org/5.1-data-loading/data-loading/file-formats#discrete-copy-number-data" class="external-link">CNA</a>, or <a href="https://docs.cbioportal.org/5.1-data-loading/data-loading/file-formats#structural-variant-data" class="external-link">fusion</a> data file formats.</p>
<p>If you wish to pull the data directly from cBioPortal, see how to get set up with credentials with the <a href="https://karissawhiting.github.io/cbioportalR/" class="external-link">{cbioportalR}</a> package.</p>
</div>
<div class="section level2">
<h2 id="processing-genomic-data">Processing Genomic Data<a class="anchor" aria-label="anchor" href="#processing-genomic-data"></a>
</h2>
<p>The below examples uses the data sets <code>mutatations</code>, <code>sv</code>, <code>cna</code> which were pulled from cBioPortal and are included in the package as example data sets. We will sample 100 samples for examples:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/Random.html" class="external-link">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mut</span> <span class="op">&lt;-</span> <span class="fu">gnomeR</span><span class="fu">::</span><span class="va"><a href="reference/mutations.html">mutations</a></span></span>
<span><span class="va">cna</span> <span class="op">&lt;-</span> <span class="fu">gnomeR</span><span class="fu">::</span><span class="va"><a href="reference/cna.html">cna</a></span></span>
<span><span class="va">sv</span> <span class="op">&lt;-</span> <span class="fu">gnomeR</span><span class="fu">::</span><span class="va"><a href="reference/sv.html">sv</a></span></span>
<span></span>
<span><span class="va">un</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">mut</span><span class="op">$</span><span class="va">sampleId</span><span class="op">)</span></span>
<span><span class="va">sample_patients</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="va">un</span>, size <span class="op">=</span> <span class="fl">100</span>, replace <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre></div>
<p>The main data processing function is <code><a href="reference/create_gene_binary.html">create_gene_binary()</a></code> which takes mutation, CNA and fusion files as input, and outputs a binary matrix of N rows (number of samples) by M genes included in the data set. We can specify which patients are included which will force all patients in resulting dataframe, even if they have no alterations.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gen_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/create_gene_binary.html">create_gene_binary</a></span><span class="op">(</span>samples <span class="op">=</span> <span class="va">sample_patients</span>,</span>
<span>                         maf <span class="op">=</span> <span class="va">mut</span>,</span>
<span>                         fusion <span class="op">=</span> <span class="va">sv</span>,</span>
<span>                         cna <span class="op">=</span> <span class="va">cna</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">gen_dat</span><span class="op">[</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt;                   ERG.fus KDM5A.fus KDM5D.fus GSK3B.fus EGFR.fus PBRM1.fus</span></span>
<span><span class="co">#&gt; P-0008869-T01-IM5       1         0         0         0        0         0</span></span>
<span><span class="co">#&gt; P-0001242-T01-IM3       0         0         0         0        0         0</span></span>
<span><span class="co">#&gt; P-0005806-T01-IM5       0         0         0         0        0         0</span></span>
<span><span class="co">#&gt; P-0007346-T01-IM5       0         0         0         0        0         0</span></span>
<span><span class="co">#&gt; P-0001861-T01-IM3       0         0         0         0        0         0</span></span>
<span><span class="co">#&gt; P-0001202-T01-IM3       0         0         0         0        0         0</span></span></code></pre></div>
<p>By default, mutations, CNA and fusions will be returned in separate columns. You can combine these at teh gene level using the following:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">by_gene</span> <span class="op">&lt;-</span> <span class="va">gen_dat</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="reference/summarize_by_gene.html">summarize_by_gene</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">by_gene</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span><span class="op">)</span></span>
<span><span class="co">#&gt; # A tibble: 6 x 6</span></span>
<span><span class="co">#&gt;   sample_id           ERG KDM5A KDM5D GSK3B  EGFR</span></span>
<span><span class="co">#&gt;   &lt;chr&gt;             &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;</span></span>
<span><span class="co">#&gt; 1 P-0008869-T01-IM5     1     0     0     0     0</span></span>
<span><span class="co">#&gt; 2 P-0001242-T01-IM3     0     0     0     0     0</span></span>
<span><span class="co">#&gt; 3 P-0005806-T01-IM5     0     0     0     0     0</span></span>
<span><span class="co">#&gt; 4 P-0007346-T01-IM5     0     0     0     0     0</span></span>
<span><span class="co">#&gt; 5 P-0001861-T01-IM3     0     0     0     0     0</span></span>
<span><span class="co">#&gt; 6 P-0001202-T01-IM3     0     0     0     0     0</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="visualize">Visualize<a class="anchor" aria-label="anchor" href="#visualize"></a>
</h2>
<p>You can visualize your processed and raw alteration data sets using {gnomeR}’s many data visualization functions.</p>
<p>Quickly visualize mutation characteristics with <code><a href="reference/ggvarclass.html">ggvarclass()</a></code>, <code><a href="reference/ggvartype.html">ggvartype()</a></code>, <code>ggsnvclass()</code>, <code><a href="reference/ggsamplevar.html">ggsamplevar()</a></code>, <code><a href="reference/ggtopgenes.html">ggtopgenes()</a></code>, <code><a href="reference/gggenecor.html">gggenecor()</a></code>, and <code><a href="reference/ggcomut.html">ggcomut()</a></code>.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/ggvarclass.html">ggvarclass</a></span><span class="op">(</span>mutation <span class="op">=</span> <span class="va">mut</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-5-1.png" width="100%"></p>
</div>
<div class="section level2">
<h2 id="summarize--analyze">Summarize &amp; Analyze<a class="anchor" aria-label="anchor" href="#summarize--analyze"></a>
</h2>
<p>You can tabulate summarize your genomic data frame using the <code><a href="reference/tbl_genomic.html">tbl_genomic()</a></code> function, a wrapper for <code><a href="https://www.danieldsjoberg.com/gtsummary/reference/tbl_summary.html" class="external-link">gtsummary::tbl_summary()</a></code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">gen_dat</span> <span class="op">&lt;-</span> <span class="va">gen_dat</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>trt_status <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span>x <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"pre-trt"</span>, <span class="st">"post-trt"</span><span class="op">)</span>,</span>
<span>       size <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/nrow.html" class="external-link">nrow</a></span><span class="op">(</span><span class="va">gen_dat</span><span class="op">)</span>, replace <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span> </span></code></pre></div>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">gene_tbl_trt</span> <span class="op">&lt;-</span>  <span class="va">gen_dat</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="reference/tbl_genomic.html">tbl_genomic</a></span><span class="op">(</span>freq_cutoff <span class="op">=</span> <span class="fl">.1</span>, by <span class="op">=</span> <span class="va">trt_status</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">gtsummary</span><span class="fu">::</span><span class="fu"><a href="https://www.danieldsjoberg.com/gtsummary/reference/add_p.html" class="external-link">add_p</a></span><span class="op">(</span><span class="op">)</span> </span></code></pre></div>
<p><img src="reference/figures/README-tbl_genomic_print.png" width="30%"></p>
<p>Additionally, you can analyze custom pathways, or a set of default gene pathways using <code><a href="reference/add_pathways.html">add_pathways()</a></code>:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">path_by_trt</span> <span class="op">&lt;-</span> <span class="va">gen_dat</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="reference/add_pathways.html">add_pathways</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">select</span><span class="op">(</span><span class="va">trt_status</span>, <span class="fu">contains</span><span class="op">(</span><span class="st">"pathway_"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="reference/tbl_genomic.html">tbl_genomic</a></span><span class="op">(</span>by <span class="op">=</span> <span class="va">trt_status</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">gtsummary</span><span class="fu">::</span><span class="fu"><a href="https://www.danieldsjoberg.com/gtsummary/reference/add_p.html" class="external-link">add_p</a></span><span class="op">(</span><span class="op">)</span> </span></code></pre></div>
<p><img src="reference/figures/README-path_by_trt.png" width="30%"></p>
</div>
<div class="section level2">
<h2 id="further-analytical-tools">Further analytical tools<a class="anchor" aria-label="anchor" href="#further-analytical-tools"></a>
</h2>
<p>Along with mutation, cna and fusion data, {gnomeR} also allows analysis and visualization of <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5027494/" class="external-link">FACETs</a> data. <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5027494/" class="external-link">FACETs</a> is an allele-specific copy number tool and open-source software with a broad application to whole genome, whole-exome, as well as targeted panel sequencing platforms. It is a fully integrated stand-alone pipeline that includes sequencing BAM file post-processing, joint segmentation of total- and allele-specific read counts, and integer copy number calls corrected for tumor purity, ploidy and clonal heterogeneity, with comprehensive output.</p>
<p>You can visualize this data using <code>facets_heatmap()</code></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">select_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html" class="external-link">sample</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">seg</span><span class="op">$</span><span class="va">ID</span><span class="op">)</span>,  <span class="fl">100</span><span class="op">)</span></span>
<span><span class="va">p.heat</span> <span class="op">&lt;-</span> <span class="fu">facets_heatmap</span><span class="op">(</span>seg <span class="op">=</span> <span class="va">seg</span>, samples <span class="op">=</span> <span class="va">select_samples</span>, min_purity <span class="op">=</span> <span class="fl">0</span><span class="op">)</span></span>
<span><span class="va">p.heat</span><span class="op">$</span><span class="va">p</span></span></code></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-10-1.png" width="100%"></p>
</div>
</div>
<div class="section level1">
<h1 id="contributing">Contributing<a class="anchor" aria-label="anchor" href="#contributing"></a>
</h1>
<p>Please note that the gnomeR project is released with a <a href="https://contributor-covenant.org/version/2/0/CODE_OF_CONDUCT.html" class="external-link">Contributor Code of Conduct</a>. By contributing to this project, you agree to abide by its terms.</p>
<p>Thank you to all contributors!</p>
<p><a href="https://github.com/akriti21" class="external-link">@akriti21</a>, <a href="https://github.com/arorarshi" class="external-link">@arorarshi</a>, <a href="https://github.com/AxelitoMartin" class="external-link">@AxelitoMartin</a>, <a href="https://github.com/carokos" class="external-link">@carokos</a>, <a href="https://github.com/ChristineZ-msk" class="external-link">@ChristineZ-msk</a>, <a href="https://github.com/edrill" class="external-link">@edrill</a>, <a href="https://github.com/hfuchs5" class="external-link">@hfuchs5</a>, <a href="https://github.com/jalavery" class="external-link">@jalavery</a>, <a href="https://github.com/jflynn264" class="external-link">@jflynn264</a>, <a href="https://github.com/karissawhiting" class="external-link">@karissawhiting</a>, <a href="https://github.com/michaelcurry1123" class="external-link">@michaelcurry1123</a>, <a href="https://github.com/mljaniczek" class="external-link">@mljaniczek</a>, and <a href="https://github.com/slb2240" class="external-link">@slb2240</a></p>
</div>

  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/MSKCC-Epi-Bio/gnomeR/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/MSKCC-Epi-Bio/gnomeR/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>

<div class="community">
<h2 data-toc-skip>Community</h2>
<ul class="list-unstyled">
<li><a href="CODE_OF_CONDUCT.html">Code of conduct</a></li>
</ul>
</div>

<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing gnomeR</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>
<a href="https://www.karissawhiting.com/" class="external-link">Karissa Whiting</a> <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-4683-1868" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Michael Curry <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-0261-4044" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>
<a href="https://github.com/hfuchs5/" class="external-link">Hannah Fuchs</a> <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0003-2426-0096" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Axel Martin <br><small class="roles"> Author </small> <a href="https://orcid.org/YOUR-ORCID-ID" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Arshi Arora <br><small class="roles"> Author </small>  </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/MSKCC-Epi-Bio/gnomeR/actions" class="external-link"><img src="https://github.com/MSKCC-Epi-Bio/gnomeR/workflows/R-CMD-check/badge.svg" alt="R-CMD-check"></a></li>
<li><a href="https://app.codecov.io/gh/MSKCC-Epi-Bio/gnomeR?branch=main" class="external-link"><img src="https://codecov.io/gh/MSKCC-Epi-Bio/gnomeR/branch/main/graph/badge.svg" alt="Codecov test coverage"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://www.karissawhiting.com/" class="external-link">Karissa Whiting</a>, Michael Curry, <a href="https://github.com/hfuchs5/" class="external-link">Hannah Fuchs</a>, Axel Martin, Arshi Arora.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
