<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Wrangle and analyze IMPACT and TCGA mutation data  • gnomeR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Wrangle and analyze IMPACT and TCGA mutation data ">
<meta property="og:description" content="Prepare raw IMPACT and TCGA mutation data for analysis. Annotate OncoKB, summarize maf files, visualize results and more. ">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">gnomeR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/Analizing-genomic-data.html">Analizing genomic data</a>
    </li>
    <li>
      <a href="articles/Data-processing.html">Data-processing</a>
    </li>
    <li>
      <a href="articles/Visualizations.html">Visualizations</a>
    </li>
    <li>
      <a href="articles/gnomeR-vignette.html">gnomeR for IMPACT data</a>
    </li>
  </ul>
</li>
<li>
  <a href="news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/AxelitoMartin/gnomeR/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div id="gnomer" class="section level1">
<div class="page-header"><h1 class="hasAnchor">
<a href="#gnomer" class="anchor"></a>gnomeR</h1></div>
<!-- badges: start -->

<p>the <code>gnomeR</code> package provides a consistent framework for genetic data processing, visualization and analysis. This is primarily targeted to IMPACT datasets but can also be applied to any genomic data provided by CbioPortal.</p>
<ul>
<li>
<a href="https://github.com/karissawhiting/cbioportalr"><strong>Dowloading and gathering data from CbioPortal</strong></a> through an integrated API using simply the sample IDs of the samples of interests or the name of the study to retrive all samples in that study. A separate package <code>cbioportalr</code> was developed independently.</li>
<li>
<a href="https://axelitomartin.github.io/gnomeR/articles/Data-processing.html"><strong>Processing genomic data</strong></a> retrieved for mutations (MAF file), fusions (MAF file) and copy-number alterations (and when available segmentation files) into an analysis ready format.</li>
<li>
<a href="https://axelitomartin.github.io/gnomeR/articles/Visualizations.html"><strong>Visualization of the processed data</strong></a> provided through MAF file summaries, OncoPrints and heatmaps.</li>
<li>
<a href="https://axelitomartin.github.io/gnomeR/articles/Analizing-genomic-data.html"><strong>Analyzing the processed data</strong></a> for association with binary, continuous and survival outcome. Including further visualiztion to improve understanding of the results.</li>
</ul>
<div id="installation" class="section level2">
<h2 class="hasAnchor">
<a href="#installation" class="anchor"></a>Installation</h2>
<p>You can install <code>gnomeR</code> from <a href="https://github.com/">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit">
<span class="co"># install.packages("devtools")</span>
<span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"AxelitoMartin/gnomeR"</span><span class="op">)</span></pre></div>
<p>Similarly for those who wish to explore the development version of <code>gnomeR</code>:</p>
<div class="sourceCode" id="cb2"><pre class="downlit">
<span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"AxelitoMartin/gnomeR"</span>, ref <span class="op">=</span> <span class="st">"development"</span><span class="op">)</span></pre></div>
<p>Along with its companion package for cbioPortal data download:</p>
<div class="sourceCode" id="cb3"><pre class="downlit">
<span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"karissawhiting/cbioportalr"</span><span class="op">)</span></pre></div>
</div>
<div id="examples" class="section level2">
<h2 class="hasAnchor">
<a href="#examples" class="anchor"></a>Examples</h2>
<div id="setting-up-the-api" class="section level3">
<h3 class="hasAnchor">
<a href="#setting-up-the-api" class="anchor"></a>Setting up the API</h3>
<p>In order to download the data from CbioPortal, one must first require a token from the website <a href="https://cbioportal.mskcc.org/">CbioPortal</a> wich will prompt a login page with your MSKCC credentials. Then navigate to “Web API” in the top bar menu, following this simply download a token and copy it after running the following command in R:</p>
<div class="sourceCode" id="cb4"><pre class="downlit">
<span class="fu">usethis</span><span class="fu">::</span><span class="fu">edit_r_environ</span><span class="op">(</span><span class="op">)</span></pre></div>
<p>And pasting the token you were given in the .Renviron file that was created and saving after pasting your token.</p>
<div class="sourceCode" id="cb5"><pre class="downlit">
<span class="va">CBIOPORTAL_TOKEN</span> <span class="op">=</span> <span class="st">'YOUR_TOKEN'</span></pre></div>
<p>You can test your connection using:</p>
<div class="sourceCode" id="cb6"><pre class="downlit">
<span class="fu">get_cbioportal_token</span><span class="op">(</span><span class="op">)</span></pre></div>
</div>
<div id="retrieving-data" class="section level3">
<h3 class="hasAnchor">
<a href="#retrieving-data" class="anchor"></a>Retrieving data</h3>
<p>Now that the Cbioportal API is set up in your environment, you must first specify the database of interest (IMPACT or TCGA are the two available options). Following this one can either sepcify the samples or study of interest:</p>
<div class="sourceCode" id="cb7"><pre class="downlit">
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/AxelitoMartin/gnomeR">gnomeR</a></span><span class="op">)</span>
<span class="kw"><a href="https://rdrr.io/r/base/library.html">library</a></span><span class="op">(</span><span class="va">cbioportalr</span><span class="op">)</span>
<span class="va">ids</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">mut</span><span class="op">$</span><span class="va">Tumor_Sample_Barcode</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">100</span><span class="op">]</span><span class="op">)</span>
<span class="va">df</span> <span class="op">&lt;-</span> <span class="fu">get_genetics</span><span class="op">(</span>sample_ids <span class="op">=</span> <span class="va">ids</span>,database <span class="op">=</span> <span class="st">"msk_impact"</span>,
                       mutations <span class="op">=</span> <span class="cn">TRUE</span>, fusions <span class="op">=</span> <span class="cn">TRUE</span>, cna <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></pre></div>
</div>
<div id="processing-the-downloaded-data" class="section level3">
<h3 class="hasAnchor">
<a href="#processing-the-downloaded-data" class="anchor"></a>Processing the downloaded data</h3>
<p>The <code><a href="reference/binmat.html">binmat()</a></code> function is the feature of the data processing of <code>gnomeR</code>. It takes genomic inputs from various sources of CbioPortal (mutation files, fusion files and copy number raw counts) to give out a clean binary matrix of n samples by all the events that were found in the files.</p>
<div class="sourceCode" id="cb8"><pre class="downlit">
<span class="va">df.clean</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/binmat.html">binmat</a></span><span class="op">(</span>maf <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">mut</span>, cna <span class="op">=</span> <span class="va">df</span><span class="op">$</span><span class="va">cna</span><span class="op">)</span></pre></div>
<p>We further included example datasets from the raw dowloaded files on CbioPortal (<code>mut</code>, <code>fusion</code>, <code>cna</code>) which we will use for the following examples.</p>
<div class="sourceCode" id="cb9"><pre class="downlit">
<span class="fu"><a href="https://rdrr.io/r/base/Random.html">set.seed</a></span><span class="op">(</span><span class="fl">123</span><span class="op">)</span>
<span class="va">patients</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">mut</span><span class="op">$</span><span class="va">Tumor_Sample_Barcode</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span><span class="op">(</span><span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span><span class="op">(</span><span class="va">mut</span><span class="op">$</span><span class="va">Tumor_Sample_Barcode</span><span class="op">)</span><span class="op">)</span>, <span class="fl">100</span>, replace<span class="op">=</span><span class="cn">FALSE</span><span class="op">)</span><span class="op">]</span>

<span class="va">gen.dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/binmat.html">binmat</a></span><span class="op">(</span>patients <span class="op">=</span> <span class="va">patients</span>, maf <span class="op">=</span> <span class="va">mut</span>, fusion <span class="op">=</span> <span class="va">fusion</span>, cna <span class="op">=</span> <span class="va">cna</span><span class="op">)</span>
<span class="fu">kable</span><span class="op">(</span><span class="va">gen.dat</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span>,row.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">TP53</th>
<th align="right">IGF1R</th>
<th align="right">KEAP1</th>
<th align="right">KDM5C</th>
<th align="right">KRAS</th>
<th align="right">TERT</th>
<th align="right">MAP2K1</th>
<th align="right">NCOR1</th>
<th align="right">DDR2</th>
<th align="right">FIP1L1</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">P-0010604-T01-IM5</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">P-0002651-T01-IM3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">P-0000270-T01-IM3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">P-0002915-T01-IM3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">P-0011099-T01-IM5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">P-0000080-T01-IM3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">P-0001741-T01-IM3</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">P-0003964-T01-IM3</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">1</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="odd">
<td align="left">P-0003842-T01-IM5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
<tr class="even">
<td align="left">P-0002597-T02-IM5</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
<td align="right">0</td>
</tr>
</tbody>
</table>
</div>
<div id="visualization" class="section level3">
<h3 class="hasAnchor">
<a href="#visualization" class="anchor"></a>Visualization</h3>
<div id="maf" class="section level4">
<h4 class="hasAnchor">
<a href="#maf" class="anchor"></a>MAF</h4>
<p>Before we move on to more complex visualizations, we integrate the <code><a href="reference/maf_viz.html">maf_viz()</a></code> function to give an overview of the distribution of the different mutations across the cohort of interest:</p>
<div class="sourceCode" id="cb10"><pre class="downlit">
<span class="va">sum.plots</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/maf_viz.html">maf_viz</a></span><span class="op">(</span>maf <span class="op">=</span> <span class="va">mut</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/stats/filter.html">filter</a></span><span class="op">(</span><span class="va">Tumor_Sample_Barcode</span> <span class="op">%in%</span> <span class="va">patients</span><span class="op">)</span><span class="op">)</span>
<span class="va">sum.plots</span><span class="op">$</span><span class="va">topgenes</span></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-8-1.png" width="100%"></p>
<div class="sourceCode" id="cb11"><pre class="downlit">
<span class="va">sum.plots</span><span class="op">$</span><span class="va">genecomut</span></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-8-2.png" width="100%"></p>
</div>
<div id="oncoprints" class="section level4">
<h4 class="hasAnchor">
<a href="#oncoprints" class="anchor"></a>OncoPrints</h4>
<p>OncoPrints are a convenient way to display the overall genomic profiles of samples in the cohort of interest. This is best used for a subset of genes that are under consideration.</p>
<div class="sourceCode" id="cb12"><pre class="downlit">
<span class="va">genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"TP53"</span>,<span class="st">"PIK3CA"</span>,<span class="st">"KRAS"</span>,<span class="st">"TERT"</span>,<span class="st">"EGFR"</span>,<span class="st">"FAT"</span>,<span class="st">"ALK"</span>,<span class="st">"CDKN2A"</span>,<span class="st">"CDKN2B"</span><span class="op">)</span>
<span class="fu"><a href="reference/plot_oncoPrint.html">plot_oncoPrint</a></span><span class="op">(</span>gen.dat <span class="op">=</span> <span class="va">gen.dat</span> <span class="op">%&gt;%</span> <span class="fu">select</span><span class="op">(</span><span class="fu">starts_with</span><span class="op">(</span><span class="va">genes</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-9-1.png" width="100%"></p>
</div>
<div id="facets" class="section level4">
<h4 class="hasAnchor">
<a href="#facets" class="anchor"></a>FACETs</h4>
<p><a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5027494/">FACETs</a> is an ASCN tool and open-source software with a broad application to whole genome, whole-exome, as well as targeted panel sequencing platforms. It is a fully integrated stand-alone pipeline that includes sequencing BAM file post-processing, joint segmentation of total- and allele-specific read counts, and integer copy number calls corrected for tumor purity, ploidy and clonal heterogeneity, with comprehensive output.</p>
<div class="sourceCode" id="cb13"><pre class="downlit">
<span class="va">p.heat</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/facets.heatmap.html">facets.heatmap</a></span><span class="op">(</span>seg <span class="op">=</span> <span class="va">seg</span>, patients <span class="op">=</span> <span class="va">patients</span>, min.purity <span class="op">=</span> <span class="fl">0</span><span class="op">)</span>
<span class="va">p.heat</span><span class="op">$</span><span class="va">p</span></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-10-1.png" width="100%"></p>
</div>
</div>
<div id="analysis" class="section level3">
<h3 class="hasAnchor">
<a href="#analysis" class="anchor"></a>Analysis</h3>
<p>In this section we will quickly overview the possible analysis in gnomeR.</p>
<div id="binary-and-continuous-outcomes" class="section level4">
<h4 class="hasAnchor">
<a href="#binary-and-continuous-outcomes" class="anchor"></a>Binary and continuous outcomes</h4>
<p>The <code><a href="reference/gen.summary.html">gen.summary()</a></code> function let’s the user perform a large scale association between the genomic features present in the <code><a href="reference/binmat.html">binmat()</a></code> function output and an outcome of choice:</p>
<ul>
<li>binary (unpaired test using Fisher’s exact test and paired test using McNemmar’s exact test)</li>
<li>continuous (using simple linear regression)</li>
</ul>
<!-- end list --><div class="sourceCode" id="cb14"><pre class="downlit">
<span class="va">outcome</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/factor.html">factor</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/Binomial.html">rbinom</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">patients</span><span class="op">)</span>,size <span class="op">=</span> <span class="fl">1</span>,prob <span class="op">=</span> <span class="fl">1</span><span class="op">/</span><span class="fl">2</span><span class="op">)</span>,levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"0"</span>,<span class="st">"1"</span><span class="op">)</span><span class="op">)</span>
<span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/gen.summary.html">gen.summary</a></span><span class="op">(</span>gen.dat <span class="op">=</span> <span class="va">gen.dat</span>,outcome <span class="op">=</span> <span class="va">outcome</span>,filter <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span>
<span class="fu">kable</span><span class="op">(</span><span class="va">out</span><span class="op">$</span><span class="va">fits</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="op">]</span>,row.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">Feature</th>
<th align="left">Overall</th>
<th align="left">0(N=50)</th>
<th align="left">1(N=50)</th>
<th align="left">OddsRatio</th>
<th align="left">Pvalue</th>
<th align="left">FDR</th>
<th align="left">Lower</th>
<th align="left">Upper</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">MYC.Amp</td>
<td align="left">MYC.Amp</td>
<td align="left">6%</td>
<td align="left">10%</td>
<td align="left">2%</td>
<td align="left">0.19</td>
<td align="left">2.04e-01</td>
<td align="left">1.00e+00</td>
<td align="left">0</td>
<td align="left">1.76</td>
</tr>
<tr class="even">
<td align="left">PIK3CA</td>
<td align="left">PIK3CA</td>
<td align="left">12%</td>
<td align="left">8%</td>
<td align="left">16%</td>
<td align="left">2.17</td>
<td align="left">3.57e-01</td>
<td align="left">1.00e+00</td>
<td align="left">0.53</td>
<td align="left">10.6</td>
</tr>
<tr class="odd">
<td align="left">EPHA5</td>
<td align="left">EPHA5</td>
<td align="left">5%</td>
<td align="left">8%</td>
<td align="left">2%</td>
<td align="left">0.24</td>
<td align="left">3.62e-01</td>
<td align="left">1.00e+00</td>
<td align="left">0</td>
<td align="left">2.52</td>
</tr>
<tr class="even">
<td align="left">FLT4</td>
<td align="left">FLT4</td>
<td align="left">5%</td>
<td align="left">2%</td>
<td align="left">8%</td>
<td align="left">4.21</td>
<td align="left">3.62e-01</td>
<td align="left">1.00e+00</td>
<td align="left">0.4</td>
<td align="left">213.8</td>
</tr>
<tr class="odd">
<td align="left">DOT1L</td>
<td align="left">DOT1L</td>
<td align="left">5%</td>
<td align="left">2%</td>
<td align="left">8%</td>
<td align="left">4.21</td>
<td align="left">3.62e-01</td>
<td align="left">1.00e+00</td>
<td align="left">0.4</td>
<td align="left">213.8</td>
</tr>
<tr class="even">
<td align="left">ERBB2.Amp</td>
<td align="left">ERBB2.Amp</td>
<td align="left">5%</td>
<td align="left">2%</td>
<td align="left">8%</td>
<td align="left">4.21</td>
<td align="left">3.62e-01</td>
<td align="left">1.00e+00</td>
<td align="left">0.4</td>
<td align="left">213.8</td>
</tr>
<tr class="odd">
<td align="left">STK11</td>
<td align="left">STK11</td>
<td align="left">7%</td>
<td align="left">10%</td>
<td align="left">4%</td>
<td align="left">0.38</td>
<td align="left">4.36e-01</td>
<td align="left">1.00e+00</td>
<td align="left">0.03</td>
<td align="left">2.46</td>
</tr>
<tr class="even">
<td align="left">APC</td>
<td align="left">APC</td>
<td align="left">7%</td>
<td align="left">4%</td>
<td align="left">10%</td>
<td align="left">2.64</td>
<td align="left">4.36e-01</td>
<td align="left">1.00e+00</td>
<td align="left">0.41</td>
<td align="left">29.07</td>
</tr>
<tr class="odd">
<td align="left">MLL</td>
<td align="left">MLL</td>
<td align="left">9%</td>
<td align="left">12%</td>
<td align="left">6%</td>
<td align="left">0.47</td>
<td align="left">4.87e-01</td>
<td align="left">1.00e+00</td>
<td align="left">0.07</td>
<td align="left">2.37</td>
</tr>
<tr class="even">
<td align="left">FAT1</td>
<td align="left">FAT1</td>
<td align="left">11%</td>
<td align="left">14%</td>
<td align="left">8%</td>
<td align="left">0.54</td>
<td align="left">5.25e-01</td>
<td align="left">1.00e+00</td>
<td align="left">0.11</td>
<td align="left">2.29</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb15"><pre class="downlit">
<span class="va">out</span><span class="op">$</span><span class="va">forest.plot</span></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-11-1.png" width="100%"></p>
</div>
<div id="survival-analysis" class="section level4">
<h4 class="hasAnchor">
<a href="#survival-analysis" class="anchor"></a>Survival analysis</h4>
<p>Similarly we include simple tools to perform univariate Cox’s proportional regression adjusted for false discovery rate in the <code><a href="reference/uni.cox.html">uni.cox()</a></code> function.</p>
<div class="sourceCode" id="cb16"><pre class="downlit">
<span class="va">time</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/Exponential.html">rexp</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html">length</a></span><span class="op">(</span><span class="va">patients</span><span class="op">)</span><span class="op">)</span>
<span class="va">status</span> <span class="op">&lt;-</span> <span class="va">outcome</span>
<span class="va">surv.dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/cbind.html">cbind</a></span><span class="op">(</span><span class="va">time</span>,<span class="va">status</span><span class="op">)</span><span class="op">)</span>
<span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/uni.cox.html">uni.cox</a></span><span class="op">(</span>X <span class="op">=</span> <span class="va">gen.dat</span>, surv.dat <span class="op">=</span> <span class="va">surv.dat</span>, surv.formula <span class="op">=</span> <span class="fu">Surv</span><span class="op">(</span><span class="va">time</span>,<span class="va">status</span><span class="op">)</span><span class="op">~</span><span class="va">.</span>,filter <span class="op">=</span> <span class="fl">0.05</span><span class="op">)</span>
<span class="fu">kable</span><span class="op">(</span><span class="va">out</span><span class="op">$</span><span class="va">tab</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>,<span class="op">]</span>,row.names <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">Feature</th>
<th align="right">Coefficient</th>
<th align="right">HR</th>
<th align="right">Pvalue</th>
<th align="right">FDR</th>
<th align="right">MutationFrequency</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">1</td>
<td align="left">MLL</td>
<td align="right">-1.480</td>
<td align="right">0.228</td>
<td align="right">0.0155</td>
<td align="right">0.6043112</td>
<td align="right">0.09</td>
</tr>
<tr class="even">
<td align="left">2</td>
<td align="left">STK11</td>
<td align="right">-1.460</td>
<td align="right">0.233</td>
<td align="right">0.0519</td>
<td align="right">0.9122695</td>
<td align="right">0.07</td>
</tr>
<tr class="odd">
<td align="left">3</td>
<td align="left">KEAP1</td>
<td align="right">-1.020</td>
<td align="right">0.362</td>
<td align="right">0.1670</td>
<td align="right">0.9122695</td>
<td align="right">0.05</td>
</tr>
<tr class="even">
<td align="left">4</td>
<td align="left">NOTCH1</td>
<td align="right">0.609</td>
<td align="right">1.840</td>
<td align="right">0.2070</td>
<td align="right">0.9122695</td>
<td align="right">0.08</td>
</tr>
<tr class="odd">
<td align="left">5</td>
<td align="left">DOT1L</td>
<td align="right">0.659</td>
<td align="right">1.930</td>
<td align="right">0.2110</td>
<td align="right">0.9122695</td>
<td align="right">0.05</td>
</tr>
<tr class="even">
<td align="left">6</td>
<td align="left">TSC1</td>
<td align="right">0.750</td>
<td align="right">2.120</td>
<td align="right">0.2130</td>
<td align="right">0.9122695</td>
<td align="right">0.05</td>
</tr>
<tr class="odd">
<td align="left">7</td>
<td align="left">MYC.Amp</td>
<td align="right">-1.200</td>
<td align="right">0.301</td>
<td align="right">0.2350</td>
<td align="right">0.9122695</td>
<td align="right">0.06</td>
</tr>
<tr class="even">
<td align="left">8</td>
<td align="left">CDH1</td>
<td align="right">0.605</td>
<td align="right">1.830</td>
<td align="right">0.2520</td>
<td align="right">0.9122695</td>
<td align="right">0.06</td>
</tr>
<tr class="odd">
<td align="left">9</td>
<td align="left">FGFR1.Amp</td>
<td align="right">-0.826</td>
<td align="right">0.438</td>
<td align="right">0.2550</td>
<td align="right">0.9122695</td>
<td align="right">0.06</td>
</tr>
<tr class="even">
<td align="left">10</td>
<td align="left">EPHA5</td>
<td align="right">-1.080</td>
<td align="right">0.340</td>
<td align="right">0.2870</td>
<td align="right">0.9122695</td>
<td align="right">0.05</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb17"><pre class="downlit">
<span class="va">out</span><span class="op">$</span><span class="va">KM</span><span class="op">[[</span><span class="fl">1</span><span class="op">]</span><span class="op">]</span></pre></div>
<p><img src="reference/figures/README-unnamed-chunk-12-1.png" width="100%"></p>
</div>
</div>
<div id="further-analytical-tools" class="section level3">
<h3 class="hasAnchor">
<a href="#further-analytical-tools" class="anchor"></a>Further analytical tools</h3>
<p>The primary goal of <code>gnomeR</code> not being in depth analysis of genomic data but rather reliable, modulable and reproducible framework for processing various types of genomic data. For users interested in large scale genomic analytical methods we compiled various packages developed by Memorial Sloan-Kettering Cancer Center employees under an umbrella R package, <a href="https://github.com/AxelitoMartin/genomeVerse">gnomeVerse</a>.</p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="links">
<h2>Links</h2>
<ul class="list-unstyled">
<li>Browse source code at <br><a href="https://github.com/AxelitoMartin/gnomeR/">https://​github.com/​AxelitoMartin/​gnomeR/​</a>
</li>
<li>Report a bug at <br><a href="https://github.com/AxelitoMartin/gnomeR/issues">https://​github.com/​AxelitoMartin/​gnomeR/​issues</a>
</li>
</ul>
</div>
<div class="license">
<h2>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>
<div class="developers">
<h2>Developers</h2>
<ul class="list-unstyled">
<li>Axel Martin <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/YOUR-ORCID-ID" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Arshi Arora <br><small class="roles"> Author </small>  </li>
<li>Karissa Whiting <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-4683-1868" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
<li>Margie Hannum <br><small class="roles"> Author </small> <a href="https://orcid.org/0000-0002-2953-0449" target="orcid.widget" aria-label="ORCID"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

  <div class="dev-status">
<h2>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://travis-ci.org/AxelitoMartin/gnomeR"><img src="https://travis-ci.com/AxelitoMartin/gnomeR.svg?branch=development" alt="Travis build status"></a></li>
<li><a href="https://codecov.io/gh/AxelitoMartin/gnomeR?branch=development"><img src="https://codecov.io/gh/AxelitoMartin/gnomeR/branch/development/graph/badge.svg" alt="Codecov test coverage"></a></li>
</ul>
</div>
</div>
</div>


      <footer><div class="copyright">
  <p>Developed by Axel Martin, Arshi Arora, Karissa Whiting, Margie Hannum.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
