---
title: "CNA_Vignette"
output: rmarkdown::html_vignette
author: Allison Reiner, Caroline Kostrzewa
vignette: >
  %\VignetteIndexEntry{CNA_Vignette}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

```{r setup, echo=F}
#library(gnomeR)
```


## What are copy number alterations
- CNAs= changes in copy number of a genomic segment
- Can be focal (0.5 kb – 85 Mb in length) or broad (entire chromosome arms) (Beroukhim et al., 2010)
- Oncogene activation often occurs due to amplification & TSG deactivation often due to deletions (Nabavi & Zare, 2022) 

## There are 2 forms of copy number data in cBioPortal
1. Whole number copy number
2. Segmentation data (seg.means)

## How to obtain each form
1. Whole number copy number
    - use get_cna_by_sample() from cbioportalR package
2. Segmentation data
    - Retrieve from cbioportal.mskcc.org by inputting desired patient or sample IDs
    - Select the "CN Segments" tab on the upper left hand side of the screen
    - Click the download button with the cloud symbol, above the Save SVG button
  
## Key Information about each form 
1. Whole number copy number:
    - Provides gene-level copy number alteration status
    - Amplification= 2
    - Deletion= -2
    <br>
**IMPORTANT**: Genes that do not have a CNA are not returned from the get_cna_by_sample function
- Use cbioportalR::get_panel_by_sample() to obtain which version of IMPACT panel on which your samples were run
  
  2. Segmentation data:
    - Loc.start = genomic location where segment starts
    - Loc.end = genomic location where segment ends
    - Num.mark = number of SNPs in the segment
    - Seg.mean = mean log2(copy number in tumor sample/ copy number in pooled normal sample) for a given segment

## How seg.means data gets translated to gene-level copy number data
- In cBioPortal, the threshold for a segment to qualify as a gene-level copy number alteration is |0.2|
- Gene level copy number alteration must be “signed out” by the clinical bioinformatics team/pathologist
- Can see sign out notes under the “clinical” tab in cBioPortal (column may be called "SO Comments")

## Steps to create a copy number heatmap across samples using segmentation data

![Copy Number Heatmap Example(Hieronymus et al.,2018)](H:/Biostatistics/ReinerA1/Projects/CNA_Info/Images/CNAheatmapex.jpg) 
<br>
Copy Number Heatmap Example (Hieronymus et al.,2018)
<br>

1. Obtain seg.means data

2. Use the CNregions function from iClusterPlus package to obtain the minimum set of segments that capture the CNA landscape
    - All samples must have segments on every chromosome
    
3. Use emcncf function from FACETS package to obtain purity, ploidy and clonal heterogeneity-adjusted copy number estimates

4. Plot using ComplexHeatmap package


  
  



