---
title: "Processing Data with {gnomeR}"
output: rmarkdown::html_vignette
author: Akriti Mishra, Karissa Whiting
vignette: >
  %\VignetteIndexEntry{test}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Introduction

This this vignette we will walk through a data example to show available {gnomeR} data processing,
visualization and analysis functions. We will also outline some common pitfalls and format inconsistencies you
may encounter when working with mutation, CNA or structural variant data, and show some {gnomeR} helper functions that can help
you manage these inconsistencies.

##  Setting up 

Make sure that gnomeR & cbioportalR packages are installed & loaded. We will use dplyr for the purposes of this vignette as well. 

```{r message = FALSE, warning=FALSE}

library(cbioportalR)
library(gnomeR)
library(dplyr)
```

For details on how to connect and access cbioportalR, you can use the link 
https://github.com/karissawhiting/cbioportalR or follow the steps below

For every new R session, you need to set your database URL. The set_cbioportal_db() function will set an environmental variable for your session that tells the package which database to point to for all API calls. 
You can set it to point to the public database with db = 'www.cbioportal.org' or db = 'public'


```{r }
set_cbioportal_db(db = "public")
```

You are now set up for the remainder of your session! API calls depend on your internet connection and possibly a VPN connection so you can use the following to check your connection at any time throughout your session:

```{r}
test_cbioportal_db()
```

This vignette will walk you through how to import data for a particular study from cbioportal and process it using some gnomeR functions.

Once connected to cbioportal, you can get the list of studies (limited to 5 here) by using - 

```{r } 
available_studies() %>% head(n = 5)
```

To get information on a particular study (eg - acbc_mskcc_2015) - 

```{r }
get_study_info("acbc_mskcc_2015") %>% t()
```



To get data from a specific study - 

```{r }
df <- get_genetics_by_study(study_id = "acbc_mskcc_2015")
```

It creates a list of all the available dataframes for the study. For instance here, the study has 3 - one for each of Mutation, CNA and Structural Variants. 

Depending on the study, the number of variables in a particular dataframe might be different. Here, the mutations data has 31 variables.

You can create one dataset each for the 3 types using-

```{r , include= TRUE}
study_mutation<- as.data.frame(df$mutation)

study_sv<- as.data.frame(df$structural_variant)

study_cna<- as.data.frame(df$cna)
```

## Data Formats

Mutation, CNA or fusion data may be formatting differently depending on where you source it. Below we outline some differences you may encounter when downloading data from the [cBioPortal website] (https://www.cbioportal.org/) versus pulling it via the [cBioPortal API](https://github.com/karissawhiting/cbioportalR), and review the important/required columns for each. Example data in this package was pulled using the API. 

See [cBioPortal documentation](https://docs.cbioportal.org/file-formats/) for more details on different file formats available supported on cBioPortal, and their data schema and coding. 

### Mutation Data

The most common mutation data format is the [Mutation Annotation Format (MAF)](https://docs.gdc.cancer.gov/Data/File_Formats/MAF_Format/) created as part of The Cancer Genome Atlas (TCGA) project. Each row in a maf file represents a mutation on a specific gene for a specific sample, therefore there are usually several rows per sample (one for each mutation). To use MAF files, you need at minimum a sample ID column and a hugo symbol column, though often additional information like mutation type or location are also often necessary. 

MAF formats are fairly consistent across sources, however if you download the raw data from a virtual study on the cBioPortal website using the download button and compare with the mutation data imported from the API, you might notice some differences in the variables names. For instance: 

    * `Tumor_Sample_Barcode` is called `sampleId` in API data
    * The `Hugo_symbol` is called `hugoGeneSymbol` 
    * `Variant_Classification` is called `mutationType`
    * `HGVSp_Short` is called `proteinChange`
    * `Chromosome` is called 'chr'

Some of the other variables are named differently as well but those differences are more intuitive to map the variables. You can
refer to `gnomeR::names_df` for more information on MAF variables. 

Luckily, most {gnomeR} functions use the data dictionary in `gnomeR::names_df` to automatically recognize the most common MAF variable names and turn them into clean, snakecase names in resulting dataframes. For example:

```{r}
gnomeR::mutations %>% names()
```

```{r}
rename_columns(gnomeR::mutations) %>% names()
```

### CNA data

The discrete copy number data from cBioPortal contains values that would be derived from copy-number analysis algorithms like GISTIC 2.0 or RAE. CNA data is often presented in a long or wide format:

    * Long format - Each row is a CNA event for a given gene and sample, therefore samples often have multiple rows (one per gene CNA event). This is most common format you will receive when downloading data using the API.
      
```{r }
gnomeR::cna[1:6, ]
```
  
    * Wide format - One column per sample, one row per gene . Each row contains data on a specific gene CNA for a given sample as shown below. One sample can have several rows. This is most common format you will receive when downloading data from the cBioPortal web browser.  

```{r}
gnomeR::cna_wide[1:6, 1:6]
```

{gnomeR} features two helper functions to help you pivot from wide to long data. 

```{r, eval=FALSE}
pivot_cna_wider(gnomeR::cna)
pivot_cna_longer(gnomeR::wide_cna)
```

There may also be differences in the way CNA events are coded. Most Events are coded in one of the following ways:

```{r, results='hide', echo=FALSE}
allowed_cna_levels <- tibble::tribble(
               ~detailed_coding, ~numeric_coding,   ~simplified_coding,
                      "neutral",             "0",       "neutral",
          "homozygous deletion",            "-2",      "deletion",
                          "loh",          "-1.5",      "deletion",
          "hemizygous deletion",            "-1",      "deletion",
                         "gain",             "1", "amplification",
      "high level amplification",            "2", "amplification")

allowed_cna_levels %>% knitr::kable()

```

{gnomeR} automatically checks CNA data labels and recodes as needed within functions. You can also use the `recode_cna()` function to do it yourself:

```{r, eval=FALSE}
x <- gnomeR::cna %>%
  mutate(alteration_recoded = recode_cna(alteration))
```


## Preparing Data For Analysis
 
### Process Data with `create_gene_binary()`

Often a first step to analyzing genomic data is organizing it in an event matrix. This matrix will have one row for each sample in your cohort and one column for each type of genomic event. Each cell will take value of `0` (no event on that gene/sample), `1` (event on that gene/sample) or `NA` (missing data or gene not tested on panel). The `create_gene_binary()` function helps you process your data into this format for use in downstream analysis. 

You can `create_gene_binary()` from any single type of data:

```{r, include = TRUE}
mut_only <- create_gene_binary(mutation = gnomeR::mutations)

head(mut_only)
```


Or from several types in one matrix. Here a dataset called `all_bin` created for ten samples that include mutation, CNA and structural variant events.

```{r, include = TRUE}

samples <- unique(study_mutation$sampleId)[1:10]

all_bin <- create_gene_binary(
    samples = samples,
    mutation = gnomeR::mutations,
    cna = gnomeR::cna,
    fusion = gnomeR::sv, 
    rm_empty = FALSE
)

all_bin[1:10, 1:10] 

```

Notes on some arguments:    

    * The arguments mutation, fusion and cna can be used to specify the datasets you want to transpose. 
    * If you are only interested in data for some sample IDs, you can mention those in the argument samples. 

### Collpase Data with `summarize_by_gene()`

You may notice some columns being repeated in the data when you create a combined dataset for the transposed datasets for Mutations, CNA and Fusion. 
To keep only the unique columns for genes, you can use the summarize_by_gene() function as shown below

```{r, include = TRUE}
 sum_all3 <- summarize_by_gene(all3)
```

The data all3 has 1855 columns. And the data sum_all3 has 1836 variables as it only keeps the unique columns across the datasets.

## Analyzing Data

### Summarize Alterations with `tbl_genomic()`

The function can be used to summarize data by gene. You can summarize data from one or more of the data types. 
The gene_binary argument expects the binary matrix data as generated by the create_gene_binary function. Example below shows the summary using gene data for ten samples. 

gene_subset argument can be used to specify the genes that you want to summarize the data for. 

freq_cutoff argument shows the summarized data for the genes that meet the cutoff. In the example below for tb1 - if the percent gene (of the total samples) is at least 5%, the gene is included in the summary table.

```{r, include = TRUE}
samples <- unique(study_mutation$sampleId)[1:10]

gene_binary <- create_gene_binary(
  samples = samples,
  mutation = study_mutation,
  cna = study_cna,
  mut_type = "somatic_only", snp_only = FALSE,
  specify_panel = "no"
)
 tb1 <- tbl_genomic(gene_binary = gene_binary, freq_cutoff = .05)
 tb2 <- tbl_genomic(gene_binary = gene_binary, gene_subset = c("BRCA1", "BRAF"))

```

### Annotate Gene Pathways with `add_pathways()` 

This function is helpful if you are interested in looking at the data for the Oncogenic Signaling Pathways. 

For the argument gene_binary - the function expects a binary matrix as obtained from the gene_binary() function. 

Using the argument pathways - you can merge a column for the pathway(s).  

The pathways available in gnomeR can be viewed using the code below.

The merged column(s) gives a flag (1/0) depending on pathway alteration status. 

```{r, include = TRUE}
names(gnomeR::pathways)
pathway_df <- add_pathways(gene_binary, pathways = "Notch")
pathway_df %>% select (pathway_Notch)

pathway_df <- add_pathways(gene_binary, pathways = c("Notch", "p53"))

```
 
### Data Visualizations


The mutation_viz functions allows you to visualize data for the variables related to variant classification, variant type, SNV class as well as top variant genes.

```{r, echo = TRUE, message = FALSE}
mutation_viz(study_mutation)
```


