---
title: "Processing Data with {gnomeR}"
output: rmarkdown::html_vignette
author: Akriti Mishra, Karissa Whiting
vignette: >
  %\VignetteIndexEntry{test}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
```

## Introduction

This this vignette we will walk through a data example to show available {gnomeR} data processing,
visualization and analysis functions. We will also outline some common pitfalls and format inconsistencies you
may encounter when working with mutation, CNA or structural variant data, and show some {gnomeR} helper functions that can help
you manage these inconsistencies.

##  Setting up 

Make sure {gnomeR} is installed & loaded. We will use dplyr for the purposes of this vignette as well. 

```{r message = FALSE, warning=FALSE}

library(gnomeR)
library(dplyr)
```

To demonstrate {gnomeR} functions, we will be using a random sample of 200 patients from a publicly available prostate cancer data pulled from cBioPortal. Data on mutations, CNA and structural variants for these patients are available in the gnomeR package (`gnomeR::mutations`, `gnomeR::cna`, `gnomeR::sv`). 

Note: To access data from cBioPortal, you can use the {cbioportalR} package: 
https://github.com/karissawhiting/cbioportalR. 


## Data Formats

Mutation, CNA or fusion data may be formatting differently depending on where you source it. Below we outline some differences you may encounter when downloading data from the [cBioPortal website] (https://www.cbioportal.org/) versus pulling it via the [cBioPortal API](https://github.com/karissawhiting/cbioportalR), and review the important/required columns for each. Example data in this package was pulled using the API. 

See [cBioPortal documentation](https://docs.cbioportal.org/file-formats/) for more details on different file formats available supported on cBioPortal, and their data schema and coding. 

### Mutation Data

The most common mutation data format is the [Mutation Annotation Format (MAF)](https://docs.gdc.cancer.gov/Data/File_Formats/MAF_Format/) created as part of The Cancer Genome Atlas (TCGA) project. Each row in a maf file represents a mutation on a specific gene for a specific sample, therefore there are usually several rows per sample (one for each mutation). To use MAF files, you need at minimum a sample ID column and a hugo symbol column, though often additional information like mutation type or location are also often necessary. 

MAF formats are fairly consistent across sources, however if you download the raw data from a virtual study on the cBioPortal website using the download button and compare with the mutation data imported from the API, you might notice some differences in the variables names. For instance: 

    * `Tumor_Sample_Barcode` is called `sampleId` in API data
    * The `Hugo_symbol` is called `hugoGeneSymbol` 
    * `Variant_Classification` is called `mutationType`
    * `HGVSp_Short` is called `proteinChange`
    * `Chromosome` is called 'chr'

Some of the other variables are named differently as well but those differences are more intuitive to map the variables. You can
refer to `gnomeR::names_df` for more information on MAF variables. 

Luckily, most {gnomeR} functions use the data dictionary in `gnomeR::names_df` to automatically recognize the most common MAF variable names and turn them into clean, snakecase names in resulting dataframes. For example:

```{r}
gnomeR::mutations %>% names()
```

```{r}
rename_columns(gnomeR::mutations) %>% names()
```

### CNA data

The discrete copy number data from cBioPortal contains values that would be derived from copy-number analysis algorithms like GISTIC 2.0 or RAE. CNA data is often presented in a long or wide format:

    * Long format - Each row is a CNA event for a given gene and sample, therefore samples often have multiple rows (one per gene CNA event). This is most common format you will receive when downloading data using the API.
      
```{r }
gnomeR::cna[1:6, ]
```
  
    * Wide format - One column per sample, one row per gene . Each row contains data on a specific gene CNA for a given sample as shown below. One sample can have several rows. This is most common format you will receive when downloading data from the cBioPortal web browser.  

```{r}
gnomeR::cna_wide[1:6, 1:6]
```

{gnomeR} features two helper functions to help you pivot from wide to long data. 

```{r, eval=FALSE}
pivot_cna_wider(gnomeR::cna)
pivot_cna_longer(gnomeR::wide_cna)
```

There may also be differences in the way CNA events are coded. Most Events are coded in one of the following ways:

```{r, results='hide', echo=FALSE}
allowed_cna_levels <- tibble::tribble(
               ~detailed_coding, ~numeric_coding,   ~simplified_coding,
                      "neutral",             "0",       "neutral",
          "homozygous deletion",            "-2",      "deletion",
                          "loh",          "-1.5",      "deletion",
          "hemizygous deletion",            "-1",      "deletion",
                         "gain",             "1", "amplification",
      "high level amplification",            "2", "amplification")

allowed_cna_levels %>% knitr::kable()

```

{gnomeR} automatically checks CNA data labels and recodes as needed within functions. You can also use the `recode_cna()` function to do it yourself:

```{r, eval=FALSE}
x <- gnomeR::cna %>%
  mutate(alteration_recoded = recode_cna(alteration))
```


## Preparing Data For Analysis
 
### Process Data with `create_gene_binary()`

Often a first step to analyzing genomic data is organizing it in an event matrix. This matrix will have one row for each sample in your cohort and one column for each type of genomic event. Each cell will take value of `0` (no event on that gene/sample), `1` (event on that gene/sample) or `NA` (missing data or gene not tested on panel). The `create_gene_binary()` function helps you process your data into this format for use in downstream analysis. 

You can `create_gene_binary()` from any single type of data (mutation, CNA or fusion):

```{r, include = TRUE}
mut_only <- create_gene_binary(mutation = gnomeR::mutations)

head(mut_only)
```

or you can process several types of alterations into a single matrix. Supported data types are:

- mutations
- copy number amplifications
- copy number deletions
- gene fusions

When processing multiple types of alteration data, by default there will be a separate column for each type of alteration on that gene. For example, TP53 mutation will result in a 1 in the `TP53` column, TP53 CNA amplification will results in a 1 in the `TP53.Amp` column etc. 

Note the use of the `samples` argument. This allows you to specify exactly which samples are in
your resulting data frame. Without specifying samples, if a sample has no alterations it will be excluded from the final matrix.

```{r, include = TRUE}

samples <- unique(gnomeR::mutations$sampleId)[1:10]

all_bin <- create_gene_binary(
    samples = samples,
    mutation = gnomeR::mutations,
    cna = gnomeR::cna,
    fusion = gnomeR::sv
)

all_bin[1:10, 1:10] 

```

**Notes on some helpful `create_gene_binary()` arguments:**

- `mut_type`- by default, any germline mutations will be omitted because data is often incomplete, but you can choose to leave them in if needed.
- `specify_panel`- If you are working across a set of samples that was sequenced on several different gene panels, this argument will insert NAs for the genes that weren't tested for any given sample. You can pass a string `"impact"` indicating automatically guessing panels and processsing IMPACT samples based on ID, or you can pass a dataframe of sample IDs and panel IDs for more fine grained control of NA annotation. 
- `recode_aliases` - Sometimes genes have several accepted names or change names over time. This can be an issue if genes are coded under multiple names in studies, or if you are working across studies. By default, this function will search for aliases for genes in your data set and resolved them to their current most common name. 

### Collapse Data with `summarize_by_gene()`

If you wish to combine all types of alterations for a given gene into 1 gene column you can use `summarize_by_gene()` 


```{r, include = TRUE}

# THIS DOESN"T WORK
# dim(all_bin)
# 
# all_bin %>% 
#   summarize_by_gene()
```


## Analyzing Data

Once you have processed the data into a binary matrix, you may want to visualize and summarize it with the following helper functions:

### Summarize Alterations with `tbl_genomic()`

This function can be used to display gene summary tables. It is built off the  {gtsummary} package and therefore you can use most customizations available int that package to customize the look of your tables. 

The `gene_binary` argument expects the binary matrix data as generated by the `create_gene_binary()` function. Example below shows the summary using gene data for ten samples. 

**Arguments to `tbl_genomic()` can help customize your tables:**

- `gene_subset` argument can be used to specify the genes that you want to summarize the data for. 

- `freq_cutoff` argument shows the summarized data for the genes that meet the cutoff. 

In the example below for `tb1`, if the percent gene (of the total samples) is at least 5%, the gene is included in the summary table.

```{r, include = TRUE}
samples <- unique(mutations$sampleId)[1:10]

gene_binary <- create_gene_binary(
  samples = samples,
  mutation = mutations,
  cna = cna,
  mut_type = "somatic_only", snp_only = FALSE,
  specify_panel = "no"
)
 tb1 <- tbl_genomic(gene_binary = gene_binary, freq_cutoff = .05)
 tb2 <- tbl_genomic(gene_binary = gene_binary, gene_subset = c("PARP1", "ALK"))

```

### Annotate Gene Pathways with `add_pathways()` 

The `add_pathways()` function allows you add columns to your gene binary matrix that annotate custom gene pathways, or oncogenic signaling pathways (add citation). 

The function expects a binary matrix as obtained from the `gene_binary()` function and will return a gene binary with additional columns added for specified pathways.

There are a set of default pathways available in the package that can be viewed using `gnomeR::pathways`


```{r, include = TRUE}
# available pathways
names(gnomeR::pathways)

pathway_df <- add_pathways(gene_binary, pathways = c("Notch", "p53"))

```
 
### Data Visualizations

The mutation_viz functions allows you to visualize data for the variables related to variant classification, variant type, SNV class as well as top variant genes.

```{r, echo = TRUE, message = FALSE}
mutation_viz(mutations)
```


