<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="gnomeR">
<title>GENIE-BPC-vignette • gnomeR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><link href="../deps/Atkinson_Hyperlegible-0.4.5/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="GENIE-BPC-vignette">
<meta property="og:description" content="gnomeR">
<meta name="robots" content="noindex">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">gnomeR</a>

    <small class="nav-text text-default me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="In-development version">1.2.0.9001</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/GENIE-vignette.html">GENIE-BPC-vignette</a>
    <a class="dropdown-item" href="../articles/data-processing-vignette.html">Processing Data with {gnomeR}</a>
    <a class="dropdown-item" href="../articles/qa-impact-data.html">How to QA Your IMPACT Data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MSKCC-Epi-Bio/gnomeR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>GENIE-BPC-vignette</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/MSKCC-Epi-Bio/gnomeR/blob/HEAD/vignettes/GENIE-vignette.Rmd" class="external-link"><code>vignettes/GENIE-vignette.Rmd</code></a></small>
      <div class="d-none name"><code>GENIE-vignette.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette will walk through how to apply {gnomeR} functions to
data from AACR Project Genomics Evidence Neoplasia Information Exchange
BioPharma Collaborative (GENIE BPC). A broad overview of AACR Project
GENIE BPC can be found <a href="https://www.aacr.org/about-the-aacr/newsroom/news-releases/aacr-project-genie-begins-five-year-collaborative-research-project-with-36-million-in-new-funding/" class="external-link">here</a>,
with details on the clinical data structure available on the <a href="https://genie-bpc.github.io/genieBPC/articles/clinical_data_structure_vignette.html" class="external-link">{genieBPC}
package website</a>.</p>
<p>For the purposes of this vignette, we will use the first publicly
available GENIE BPC data release of non-small cell lung cancer patients,
NSCLC v2.0-public.</p>
<p>Additional details regarding the GENIE BPC data and the {genieBPC} R
package are published in the following papers:</p>
<ul>
<li><p>Lavery, J. A., Brown, S., Curry, M. A., Martin, A., Sjoberg, D.
D., &amp; Whiting, K. (2023). <a href="https://pubmed.ncbi.nlm.nih.gov/36519837/" class="external-link">A data processing
pipeline for the AACR project GENIE biopharma collaborative data with
the {genieBPC} R package</a>. Bioinformatics (Oxford, England), 39(1),
btac796. <a href="https://doi.org/10.1093/bioinformatics/btac796" class="external-link uri">https://doi.org/10.1093/bioinformatics/btac796</a></p></li>
<li><p>Lavery, J. A., Lepisto, E. M., Brown, S., Rizvi, H., McCarthy,
C., LeNoue-Newton, M., Yu, C., Lee, J., Guo, X., Yu, T., Rudolph, J.,
Sweeney, S., AACR Project GENIE Consortium, Park, B. H., Warner, J. L.,
Bedard, P. L., Riely, G., Schrag, D., &amp; Panageas, K. S. (2022). <a href="https://pubmed.ncbi.nlm.nih.gov/35192403/" class="external-link">A Scalable Quality
Assurance Process for Curating Oncology Electronic Health Records: The
Project GENIE Biopharma Collaborative Approach</a>. JCO clinical cancer
informatics, 6, e2100105. <a href="https://doi.org/10.1200/CCI.21.00105" class="external-link uri">https://doi.org/10.1200/CCI.21.00105</a></p></li>
</ul>
<p>Technical details regarding proper analysis of this data can be found
in the following publication:</p>
<ul>
<li><p>Brown, S., Lavery, J. A., Shen, R., Martin, A. S., Kehl, K. L.,
Sweeney, S. M., Lepisto, E. M., Rizvi, H., McCarthy, C. G., Schultz, N.,
Warner, J. L., Park, B. H., Bedard, P. L., Riely, G. J., Schrag, D.,
Panageas, K. S., &amp; AACR Project GENIE Consortium (2022). <a href="https://pubmed.ncbi.nlm.nih.gov/34734967/" class="external-link">Implications of
Selection Bias Due to Delayed Study Entry in Clinical Genomic
Studies</a>. JAMA oncology, 8(2), 287–291. <a href="https://doi.org/10.1001/jamaoncol.2021.5153" class="external-link uri">https://doi.org/10.1001/jamaoncol.2021.5153</a></p></li>
<li><p>Kehl, K. L., Uno, H., Gusev, A., Groha, S., Brown, S., Lavery, J.
A., Schrag, D., &amp; Panageas, K. S. (2023). <a href="https://pubmed.ncbi.nlm.nih.gov/36626408/" class="external-link">Elucidating Analytic
Bias Due to Informative Cohort Entry in Cancer Clinico-genomic
Datasets</a>. Cancer epidemiology, biomarkers &amp; prevention: a
publication of the American Association for Cancer Research, cosponsored
by the American Society of Preventive Oncology, 32(3), 344–352. <a href="https://doi.org/10.1158/1055-9965.EPI-22-0875" class="external-link uri">https://doi.org/10.1158/1055-9965.EPI-22-0875</a></p></li>
<li><p>Kehl, K. L., Riely, G. J., Lepisto, E. M., Lavery, J. A., Warner,
J. L., LeNoue-Newton, M. L., Sweeney, S. M., Rudolph, J. E., Brown, S.,
Yu, C., Bedard, P. L., Schrag, D., Panageas, K. S., &amp; American
Association of Cancer Research (AACR) Project Genomics Evidence
Neoplasia Information Exchange (GENIE) Consortium (2021). <a href="https://pubmed.ncbi.nlm.nih.gov/34309669/" class="external-link">Correlation Between
Surrogate End Points and Overall Survival in a Multi-institutional
Clinicogenomic Cohort of Patients With Non-Small Cell Lung or Colorectal
Cancer</a>. JAMA network open, 4(7), e2117547. <a href="https://doi.org/10.1001/jamanetworkopen.2021.17547" class="external-link uri">https://doi.org/10.1001/jamanetworkopen.2021.17547</a></p></li>
</ul>
</div>
<div class="section level2">
<h2 id="data-access">Data Access<a class="anchor" aria-label="anchor" href="#data-access"></a>
</h2>
<p>To gain access to the GENIE BPC data, please follow the instructions
on the <a href="https://genie-bpc.github.io/genieBPC/articles/pull_data_synapse_vignette.html" class="external-link">{genieBPC}
<code>pull_data_synapse()</code> vignette</a> to register for a Synapse
account. Once your Synapse account is created and you authenticate
yourself using <code><a href="https://genie-bpc.github.io/genieBPC/reference/set_synapse_credentials.html" class="external-link">genieBPC::set_synapse_credentials()</a></code>, you’ll
be ready to pull the GENIE BPC clinical and genomic data from Synapse
into your local environment:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://genie-bpc.github.io/genieBPC/" class="external-link">genieBPC</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># if credentials are not stored in your R environment</span></span>
<span><span class="fu"><a href="https://genie-bpc.github.io/genieBPC/reference/set_synapse_credentials.html" class="external-link">set_synapse_credentials</a></span><span class="op">(</span>username <span class="op">=</span> <span class="st">"username"</span>, password <span class="op">=</span> <span class="st">"password"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># if credentials are stored in your R environment</span></span>
<span><span class="fu"><a href="https://genie-bpc.github.io/genieBPC/reference/set_synapse_credentials.html" class="external-link">set_synapse_credentials</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="obtain-genie-bpc-data">Obtain GENIE BPC Data<a class="anchor" aria-label="anchor" href="#obtain-genie-bpc-data"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># pull NSCLC v2.0-public data from Synapse into the R environment</span></span>
<span><span class="va">nsclc_2_0</span> <span class="op">=</span> <span class="fu"><a href="https://genie-bpc.github.io/genieBPC/reference/pull_data_synapse.html" class="external-link">pull_data_synapse</a></span><span class="op">(</span>cohort <span class="op">=</span> <span class="st">"NSCLC"</span>, version <span class="op">=</span> <span class="st">"v2.0-public"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">pt_char</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">ca_dx_index</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">ca_dx_non_index</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">ca_drugs</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">prissmm_imaging</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">prissmm_pathology</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">prissmm_md</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">cpt</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">mutations_extended</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">fusions</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">cna</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span></code></pre></div>
<p>The resulting <code>nsclc_2_0</code> object is a nested list of
datasets, including the mutations, fusions, and copy number data.</p>
<ul>
<li><code>nsclc_2_0$NSCLC_v2.0$mutations_extended</code></li>
<li><code>nsclc_2_0$NSCLC_v2.0$fusions</code></li>
<li><code>nsclc_2_0$NSCLC_v2.0$cna</code></li>
</ul>
<p>Note that while the GENIE BPC clinical data are only available via
Synapse, the genomic data can be accessed via both Synapse and
cBioPortal. Using the {cbioportalR} package, users can pull the GENIE
BPC genomic data directly from cBioPortal:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/karissawhiting/cbioportalR" class="external-link">cbioportalR</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># connect to the GENIE instance of cBioPortal</span></span>
<span><span class="fu">cbioportalR</span><span class="fu">::</span><span class="fu"><a href="https://karissawhiting.github.io/cbioportalR/reference/set_cbioportal_db.html" class="external-link">set_cbioportal_db</a></span><span class="op">(</span><span class="st">"https://genie.cbioportal.org/api"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># view list of available studies from this instance of the portal</span></span>
<span><span class="co"># NSCLC v2.0-public is: nsclc_public_genie_bpc</span></span>
<span><span class="fu"><a href="https://karissawhiting.github.io/cbioportalR/reference/available_studies.html" class="external-link">available_studies</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># obtain genomic data for GENIE BPC NSCLC v2.0-public</span></span>
<span><span class="va">mutations_extended_2_0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://karissawhiting.github.io/cbioportalR/reference/get_mutations_by_study.html" class="external-link">get_mutations_by_study</a></span><span class="op">(</span><span class="st">"nsclc_public_genie_bpc"</span><span class="op">)</span></span>
<span><span class="va">cna_2_0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://karissawhiting.github.io/cbioportalR/reference/get_cna_by_study.html" class="external-link">get_cna_by_study</a></span><span class="op">(</span><span class="st">"nsclc_public_genie_bpc"</span><span class="op">)</span></span>
<span><span class="va">fusions_2_0</span> <span class="op">&lt;-</span> <span class="va">get_fu</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data-formats">Data Formats<a class="anchor" aria-label="anchor" href="#data-formats"></a>
</h2>
<p>The genomic data for GENIE BPC are stored both on Synapse and in
cBioPortal. The data structure differs depending on where the genomic
data are downloaded from. Therefore, the remainder of this vignette will
proceed by outlining the process of annotating genomic data separately
for genomic data downloaded from Synapse and genomic data downloaded
from cBioPortal.</p>
<p>Additionally, due to filtering criteria applied by cBioPortal, the
genomic data from Synapse and cBioPortal do not align exactly.
Specifically, the mutation data variant classification data are filtered
on conditions described <a href="https://docs.cbioportal.org/file-formats/#mutation-data" class="external-link">here</a>.
The CNA data as downloaded from cBioPortal only includes high level CNA
(-2, 2), while the CNA data as downloaded from Synapse includes all
CNA.</p>
</div>
<div class="section level2">
<h2 id="selecting-a-cohort-for-analysis">Selecting a Cohort for Analysis<a class="anchor" aria-label="anchor" href="#selecting-a-cohort-for-analysis"></a>
</h2>
<p>The following code chunk uses the
<code><a href="https://genie-bpc.github.io/genieBPC/reference/create_analytic_cohort.html" class="external-link">genieBPC::create_analytic_cohort()</a></code> to create an analytic
cohort of patients diagnosed with stage IV NSCLC of adenocarcinoma
histology. Then, for patients with multiple genomic samples, the
<code><a href="https://genie-bpc.github.io/genieBPC/reference/select_unique_ngs.html" class="external-link">genieBPC::select_unique_ngs()</a></code> function chooses the genomic
sample with OncoTree code LUAD (if available). For patients with
multiple samples with OncoTree code LUAD, we will select the metastatic
genomic sample. If any patients have multiple metastatic samples with
OncoTree code LUAD, take the latest of the samples.</p>
<p>Note: for patients with exactly one genomic sample, that unique
genomic sample will be returned <em>regardless of whether it meets the
argument criteria specified below</em>.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># create analytic cohort of patients diagnosed with Stage IV adenocarcinoma</span></span>
<span><span class="va">nsclc_2_0_example</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://genie-bpc.github.io/genieBPC/reference/create_analytic_cohort.html" class="external-link">create_analytic_cohort</a></span><span class="op">(</span></span>
<span>  data_synapse <span class="op">=</span> <span class="va">nsclc_2_0</span><span class="op">$</span><span class="va">NSCLC_v2.0</span>,</span>
<span>  stage_dx <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Stage IV"</span><span class="op">)</span>,</span>
<span>  histology <span class="op">=</span> <span class="st">"Adenocarcinoma"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># select unique NGS samples for this analytic cohort</span></span>
<span><span class="va">nsclc_2_0_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://genie-bpc.github.io/genieBPC/reference/select_unique_ngs.html" class="external-link">select_unique_ngs</a></span><span class="op">(</span></span>
<span>  data_cohort <span class="op">=</span> <span class="va">nsclc_2_0_example</span><span class="op">$</span><span class="va">cohort_ngs</span>,</span>
<span>  oncotree_code <span class="op">=</span> <span class="st">"LUAD"</span>,</span>
<span>  sample_type <span class="op">=</span> <span class="st">"Metastasis"</span>,</span>
<span>  min_max_time <span class="op">=</span> <span class="st">"max"</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Create a dataframe of the corresponding panel and sample IDs:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># specify sample panels and IDs</span></span>
<span><span class="va">nsclc_2_0_sample_panels</span> <span class="op">&lt;-</span> <span class="va">nsclc_2_0_samples</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">cpt_seq_assay_id</span>, <span class="va">cpt_genie_sample_id</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>panel_id <span class="op">=</span> <span class="va">cpt_seq_assay_id</span>,</span>
<span>         sample_id <span class="op">=</span> <span class="va">cpt_genie_sample_id</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="process-data-with-create_gene_binary">Process Data with <code>create_gene_binary()</code><a class="anchor" aria-label="anchor" href="#process-data-with-create_gene_binary"></a>
</h2>
<p>The <code><a href="../reference/create_gene_binary.html">create_gene_binary()</a></code> function takes inputs of
mutations, fusions, and CNA data and returns a binary matrix with the
alteration status for each gene, annotating missingness when genes were
not included on a next generation sequencing panel.</p>
<p>It is critical to utilize the <code>specify_panel</code> argument of
<code><a href="../reference/create_gene_binary.html">create_gene_binary()</a></code>. Samples included in GENIE BPC were
sequenced across multiple sequencing platforms, with the genes included
varying across panels. Without the <code>specify_panel</code> argument,
missingness will not be correctly annotated, and genes that were not
tested will be incorrectly documented as not being altered.</p>
<p><strong>Using the genomic data from Synapse:</strong></p>
<p>The fusions and CNA data as downloaded from Synapse require some
modifications prior to being supplied to the
<code><a href="../reference/create_gene_binary.html">gnomeR::create_gene_binary()</a></code> function.</p>
<p>First, the CNA file can be transposed to match the expected input for
<code><a href="../reference/create_gene_binary.html">create_gene_binary()</a></code> using
<code><a href="../reference/pivot_cna_longer.html">pivot_cna_longer()</a></code>:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># transpose CNA data from Synapse</span></span>
<span><span class="va">cna_synapse_long</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pivot_cna_longer.html">pivot_cna_longer</a></span><span class="op">(</span><span class="va">nsclc_2_0</span><span class="op">$</span><span class="va">NSCLC_v2.0</span><span class="op">$</span><span class="va">cna</span><span class="op">)</span></span></code></pre></div>
<p><strong>PLACEHOLDER</strong>: Next, the fusions file can be
transposed to match the expected input for
<code><a href="../reference/create_gene_binary.html">create_gene_binary()</a></code></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># transpose fusions data from Synapse</span></span>
<span><span class="co"># fusions_synapse_long &lt;- pivot_fusions_longer(nsclc_2_0$NSCLC_v2.0$fusions)</span></span></code></pre></div>
<p>Finally, the reformatted genomic data can be supplied to
<code><a href="../reference/create_gene_binary.html">create_gene_binary()</a></code> to annotate genomic alterations for
patients in the analytic cohort of interest.</p>
<p>The CNA data as downloaded from cBioPortal only includes high level
CNA (-2, 2), so we will specify <code>high_level_cna_only = TRUE</code>
to be consistent with the results based on the genomic data as
downloaded from cBioPortal.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nsclc_2_0_gen_dat_synapse</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/create_gene_binary.html">create_gene_binary</a></span><span class="op">(</span></span>
<span>    mutation <span class="op">=</span> <span class="va">nsclc_2_0</span><span class="op">$</span><span class="va">NSCLC_v2.0</span><span class="op">$</span><span class="va">mutations_extended</span>,</span>
<span>    cna <span class="op">=</span> <span class="va">cna_synapse_long</span>,</span>
<span>    high_level_cna_only <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    fusions <span class="op">=</span> <span class="va">fusions_synapse_long</span>,</span>
<span>    samples <span class="op">=</span> <span class="va">nsclc_2_0_sample_panels</span><span class="op">$</span><span class="va">sample_id</span>,</span>
<span>    specify_panel <span class="op">=</span> <span class="va">nsclc_2_0_sample_panels</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><strong>Using the genomic data from cBioPortal:</strong></p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nsclc_2_0_gen_dat_cbio</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/create_gene_binary.html">create_gene_binary</a></span><span class="op">(</span></span>
<span>    mutation <span class="op">=</span> <span class="va">mutations_extended_2_0</span>,</span>
<span>    cna <span class="op">=</span> <span class="va">cna_2_0</span>,</span>
<span>    samples <span class="op">=</span> <span class="va">nsclc_2_0_sample_panels</span><span class="op">$</span><span class="va">sample_id</span>,</span>
<span>    specify_panel <span class="op">=</span> <span class="va">nsclc_2_0_sample_panels</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>We can compare the binary genomic matrices created using the genomic
data downloaded from Synapse versus the genomic data downloaded from
cBioPortal. We observe that they are exactly equal, and will proceed
using the <code>nsclc_2_0_gen_dat_cbio</code> object.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># compare binary genomic matrices</span></span>
<span><span class="fu">arsenal</span><span class="fu">::</span><span class="fu">comparedf</span><span class="op">(</span><span class="va">nsclc_2_0_gen_dat_synapse</span>, <span class="va">nsclc_2_0_gen_dat_cbio</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="collapse-data-with-summarize_by_gene">Collapse Data with <code>summarize_by_gene()</code><a class="anchor" aria-label="anchor" href="#collapse-data-with-summarize_by_gene"></a>
</h2>
<p>We can summarize the presence of any alteration event (mutation,
amplification, deletion, structural variant) with the
<code><a href="../reference/summarize_by_gene.html">summarize_by_gene()</a></code> function, such that each gene is a
column that captures the presence of any event regardless of alteration
type.</p>
<p>Summarizing the first 10 samples and first 10 genes:</p>
<p><strong>Using the genomic data from Synapse:</strong></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nsclc_2_0_gen_dat_synapse</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/summarize_by_gene.html">summarize_by_gene</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><strong>Using the genomic data from cBioPortal</strong></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nsclc_2_0_gen_dat_cbio</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/summarize_by_gene.html">summarize_by_gene</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="analyzing-data">Analyzing Data<a class="anchor" aria-label="anchor" href="#analyzing-data"></a>
</h2>
<p>After the data have been transformed into a binary format, we can
create summaries and visualizations to better understand the data.</p>
<div class="section level3">
<h3 id="summarize-data-with-tbl_genomic">Summarize Data with <code>tbl_genomic()</code><a class="anchor" aria-label="anchor" href="#summarize-data-with-tbl_genomic"></a>
</h3>
<p>The <code><a href="../reference/tbl_genomic.html">tbl_genomic()</a></code> function summarizes the frequency of
alteration events from the binary data returned from
<code><a href="../reference/create_gene_binary.html">create_gene_binary()</a></code> or <code><a href="../reference/summarize_by_gene.html">summarize_by_gene()</a></code>.
Users can specify particular genes (<code>gene_subset</code> argument)
or a a minimum frequency (<code>freq_cutoff</code> argument) to include
in the table.</p>
<p><strong>Using the genomic data from Synapse:</strong></p>
<p>Summarizing the frequencies of KEAP1, STK11, and SMARCA4 alteration
events:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/tbl_genomic.html">tbl_genomic</a></span><span class="op">(</span>gene_binary <span class="op">=</span> <span class="va">nsclc_2_0_gen_dat_synapse</span>, gene_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"KEAP1"</span>, <span class="st">"STK11"</span>, <span class="st">"SMARCA4"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Summarizing alteration events with &gt;=10% frequency:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/tbl_genomic.html">tbl_genomic</a></span><span class="op">(</span>gene_binary <span class="op">=</span> <span class="va">nsclc_2_0_gen_dat_synapse</span>, freq_cutoff <span class="op">=</span> <span class="fl">.10</span><span class="op">)</span></span></code></pre></div>
<p><strong>Using the genomic data from cBioPortal:</strong></p>
<p>Summarizing the frequencies of KEAP1, STK11, and SMARCA4 alteration
events:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/tbl_genomic.html">tbl_genomic</a></span><span class="op">(</span>gene_binary <span class="op">=</span> <span class="va">nsclc_2_0_gen_dat_cbio</span>, gene_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"KEAP1"</span>, <span class="st">"STK11"</span>, <span class="st">"SMARCA4"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>Summarizing alteration events with &gt;=10% frequency:</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/tbl_genomic.html">tbl_genomic</a></span><span class="op">(</span>gene_binary <span class="op">=</span> <span class="va">nsclc_2_0_gen_dat_cbio</span>, freq_cutoff <span class="op">=</span> <span class="fl">.10</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="data-visualizations">Data Visualizations<a class="anchor" aria-label="anchor" href="#data-visualizations"></a>
</h3>
<p>We can use the <code><a href="../reference/mutation_viz.html">mutation_viz()</a></code> function to visualize
several aspects of the mutation data, including variant classification,
variant type, SNV class and top variant genes.</p>
<p><strong>Using the genomic data from Synapse:</strong></p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mutation_viz_gen_dat_synapse</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mutation_viz.html">mutation_viz</a></span><span class="op">(</span><span class="va">nsclc_2_0</span><span class="op">$</span><span class="va">NSCLC_v2.0</span><span class="op">$</span><span class="va">mutations_extended</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mutation_viz_gen_dat_synapse</span></span></code></pre></div>
<p><strong>Using the genomic data from cBioPortal:</strong></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mutation_viz_gen_dat_cbio</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mutation_viz.html">mutation_viz</a></span><span class="op">(</span><span class="va">mutations_extended_2_0</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mutation_viz_gen_dat_cbio</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://www.karissawhiting.com/" class="external-link">Karissa Whiting</a>, Michael Curry, <a href="https://github.com/hfuchs5/" class="external-link">Hannah Fuchs</a>, Axel Martin, Arshi Arora.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
