<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="gnomeR">
<title>How to QA Your IMPACT Data • gnomeR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><link href="../deps/Atkinson_Hyperlegible-0.4.5/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="How to QA Your IMPACT Data">
<meta property="og:description" content="gnomeR">
<meta name="robots" content="noindex">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">gnomeR</a>

    <small class="nav-text text-default me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="In-development version">1.2.0.9002</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/data-processing-vignette.html">Processing Data with {gnomeR}</a>
    <a class="dropdown-item" href="../articles/genie-bpc-vignette.html">Analyzing GENIE BPC Data Using {gnomeR}</a>
    <a class="dropdown-item" href="../articles/qa-impact-data.html">How to QA Your IMPACT Data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MSKCC-Epi-Bio/gnomeR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>How to QA Your IMPACT Data</h1>
                        <h4 data-toc-skip class="author">Esther
Drill</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/MSKCC-Epi-Bio/gnomeR/blob/HEAD/vignettes/qa-impact-data.Rmd" class="external-link"><code>vignettes/qa-impact-data.Rmd</code></a></small>
      <div class="d-none name"><code>qa-impact-data.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>The purpose of this vignette is to outline best practices for
downloading, QA-ing and analyzing data generated from MSK IMPACT, a
targeted tumor-sequencing test that can detect more than 468 gene
mutations and other critical genetic changes in common and rare cancers.
Using a hepatocellular cancer case study, we demonstrate a data analysis
pipeline using {cbioportalR} functions that can help users generate
reproducible analyses using this data.</p>
</div>
<div class="section level2">
<h2 id="setting-up">Setting up<a class="anchor" aria-label="anchor" href="#setting-up"></a>
</h2>
<p>For this vignette, we will be using {cbioportalR}, a package to
download data from the cBioPortal website. We will also be using {dplyr}
and {tidyr} to clean and manipulate the data:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MSKCC-Epi-Bio/gnomeR" class="external-link">gnomeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/karissawhiting/cbioportalR" class="external-link">cbioportalR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span></code></pre></div>
<p>To access cBioPortal data using the <a href="https://www.karissawhiting.com/cbioportalR/" class="external-link">{cbioportalR}</a>
package, you must set the cBioPortal database using the
<code><a href="https://karissawhiting.github.io/cbioportalR/reference/set_cbioportal_db.html" class="external-link">set_cbioportal_db()</a></code> function. To access public data, set
this to <code>db = public</code>. If you are using a private version of
cBioPortal, you would set the <code>db</code> argument to your
institution’s cBioPortal URL.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://karissawhiting.github.io/cbioportalR/reference/set_cbioportal_db.html" class="external-link">set_cbioportal_db</a></span><span class="op">(</span>db <span class="op">=</span> <span class="st">"public"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> You are successfully connected!</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">base_url</span> for this R session is now set to <span style="color: #0000BB;">"www.cbioportal.org/api"</span></span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="case-study">Case Study<a class="anchor" aria-label="anchor" href="#case-study"></a>
</h2>
<p><strong>Scenario:</strong> You are a data analyst whose collaborator
has sent you a clinical file of a cohort of patients with hepatocellular
cancer that she is interested in for retrospective data analysis. In
particular, she wants to look at IMPACT sequencing data for the cohort
and investigate associations between genomic alterations and
pathological and clinical characteristics. She asks if you can get the
IMPACT data and do the analysis.</p>
<p>She gives you a clinical file with 80 sample IDs:
<code>clin_collab_df</code>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">clin_collab_df</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 3</span></span></span>
<span><span class="co">#&gt;   cbioportal_id     ctype          primary_mets</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> P-0066922-T02-IM7 Hepatocellular Primary     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> P-0009540-T01-IM5 Hepatocellular Primary     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> P-0000182-T01-IM3 Hepatocellular Metastasis  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> P-0000037-T02-IM3 Hepatocellular Primary     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> P-0005357-T01-IM5 Hepatocellular Primary     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> P-0007773-T01-IM5 Hepatocellular Metastasis</span></span></code></pre></div>
<p>The sample IDs are in the <code>cbioportal_id</code> column.</p>
<p>Before using cBioPortal to access the genomic data, you first want to
do some QA on the clinical data and make sure it matches up with the
clinical data in cBioPortal.</p>
<div class="section level3">
<h3 id="check-for-multiple-samples-per-patient">Check For Multiple Samples Per Patient<a class="anchor" aria-label="anchor" href="#check-for-multiple-samples-per-patient"></a>
</h3>
<p>One of the first things to check in your data is whether you have
multiple sample IDs from the same patient. Sometimes a clinical file
will have a patient_ID column as well; this one doesn’t, so you can make
your own. The patient ID is just the first 9 digits of the
<code>cbioportal_id</code>:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">clin_collab_df</span> <span class="op">&lt;-</span> <span class="va">clin_collab_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span></span>
<span>    patient_id <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/substr.html" class="external-link">substr</a></span><span class="op">(</span><span class="va">cbioportal_id</span>, <span class="fl">1</span>, <span class="fl">9</span><span class="op">)</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p>If there is only one sample per patient, there should be the same
number of samples as patients.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clin_collab_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarize</a></span><span class="op">(</span> patients <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">patient_id</span><span class="op">)</span><span class="op">)</span>,</span>
<span>            samples<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">cbioportal_id</span><span class="op">)</span><span class="op">)</span><span class="op">)</span> </span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 2</span></span></span>
<span><span class="co">#&gt;   patients samples</span></span>
<span><span class="co">#&gt;      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span>       78      80</span></span></code></pre></div>
<p>So it’s clear that we have multiple samples per patient. To find out
which patient/s, you can count the <code>patient_id</code> values and
filter for &gt;1.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">multiple_samps</span> <span class="op">&lt;-</span> <span class="va">clin_collab_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">patient_id</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">n</span> <span class="op">&gt;</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">multiple_samps</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 2 × 2</span></span></span>
<span><span class="co">#&gt;   patient_id     n</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> P-0004876      2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> P-0012198      2</span></span></code></pre></div>
<p>There are 2 patients who each have 2 samples in the collaborator’s
dataset. Filter the dataset to see the <code>cbioportal_id</code>’s in
question:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">clin_collab_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span></span>
<span>  <span class="va">patient_id</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> </span>
<span>    <span class="op">(</span><span class="va">multiple_samps</span><span class="op">$</span><span class="va">patient_id</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 4</span></span></span>
<span><span class="co">#&gt;   cbioportal_id     ctype          primary_mets patient_id</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> P-0004876-T01-IM5 Hepatocellular Primary      P-0004876 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> P-0012198-T02-IM5 Hepatocellular Primary      P-0012198 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> P-0004876-T02-IM5 Hepatocellular Primary      P-0004876 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> P-0012198-T01-IM5 Hepatocellular Primary      P-0012198</span></span></code></pre></div>
<p>These are patients and samples to ask your collaborator about: Does
using both samples make sense? Often times the answer is no. And if not,
which sample is the most appropriate one to include? (To get more info
for yourself, you can enter the patient ids into the <a href="https://www.cbioportal.org/" class="external-link">cBioPortal website</a>.)</p>
</div>
<div class="section level3">
<h3 id="check-that-all-cbioportal_ids-are-in-cbioportal-database">Check That All <code>cbioportal_ids</code> Are In cBioPortal
Database<a class="anchor" aria-label="anchor" href="#check-that-all-cbioportal_ids-are-in-cbioportal-database"></a>
</h3>
<p>To do this, you need to retrieve the clinical data from cBioPortal
using <a href="https://www.karissawhiting.com/cbioportalR/" class="external-link">{cbioportalR}</a>.
You can use the <code><a href="https://karissawhiting.github.io/cbioportalR/reference/get_clinical_by_sample.html" class="external-link">get_clinical_by_sample()</a></code> function from
{cbioportalR} to do this. Set the <code>sample_id</code> parameter to
the <code>cbioportal_ids</code> from the clinical collaborator’s
file.<br>
Store the sample data in a file called <code>clin_cbio</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">clin_cbio</span> <span class="op">=</span> <span class="fu"><a href="https://karissawhiting.github.io/cbioportalR/reference/get_clinical_by_sample.html" class="external-link">get_clinical_by_sample</a></span><span class="op">(</span>sample_id <span class="op">=</span> <span class="va">clin_collab_df</span><span class="op">$</span><span class="va">cbioportal_id</span><span class="op">)</span> </span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> No `clinical_attribute` passed. Defaulting to returning</span></span>
<span><span class="co">#&gt; all clinical attributes in <span style="color: #0000BB;">"msk_impact_2017"</span> study</span></span></code></pre></div>
<p>(You can disregard the warning message for now, though you may be
interested in specific clinical attributes later.)</p>
<p><em>Note: If you are using the public version of cBioPortal, this
function will only query the <code>msk_impact_2017</code>
study.</em></p>
<p>Notice that you now have 2 clinical files: one given to you by the
collaborator (<code>clin_collab_df</code>) and one you have retrieved
yourself from cBioPortal (<code>clin_cbio</code>).</p>
<p>Here’s the header of <code>clin_cbio</code>:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">clin_cbio</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;                                uniqueSampleKey</span></span>
<span><span class="co">#&gt; 1 UC0wMDAwMDM3LVQwMi1JTTM6bXNrX2ltcGFjdF8yMDE3</span></span>
<span><span class="co">#&gt; 2 UC0wMDAwMDM3LVQwMi1JTTM6bXNrX2ltcGFjdF8yMDE3</span></span>
<span><span class="co">#&gt; 3 UC0wMDAwMDM3LVQwMi1JTTM6bXNrX2ltcGFjdF8yMDE3</span></span>
<span><span class="co">#&gt; 4 UC0wMDAwMDM3LVQwMi1JTTM6bXNrX2ltcGFjdF8yMDE3</span></span>
<span><span class="co">#&gt; 5 UC0wMDAwMDM3LVQwMi1JTTM6bXNrX2ltcGFjdF8yMDE3</span></span>
<span><span class="co">#&gt; 6 UC0wMDAwMDM3LVQwMi1JTTM6bXNrX2ltcGFjdF8yMDE3</span></span>
<span><span class="co">#&gt;                     uniquePatientKey          sampleId patientId</span></span>
<span><span class="co">#&gt; 1 UC0wMDAwMDM3Om1za19pbXBhY3RfMjAxNw P-0000037-T02-IM3 P-0000037</span></span>
<span><span class="co">#&gt; 2 UC0wMDAwMDM3Om1za19pbXBhY3RfMjAxNw P-0000037-T02-IM3 P-0000037</span></span>
<span><span class="co">#&gt; 3 UC0wMDAwMDM3Om1za19pbXBhY3RfMjAxNw P-0000037-T02-IM3 P-0000037</span></span>
<span><span class="co">#&gt; 4 UC0wMDAwMDM3Om1za19pbXBhY3RfMjAxNw P-0000037-T02-IM3 P-0000037</span></span>
<span><span class="co">#&gt; 5 UC0wMDAwMDM3Om1za19pbXBhY3RfMjAxNw P-0000037-T02-IM3 P-0000037</span></span>
<span><span class="co">#&gt; 6 UC0wMDAwMDM3Om1za19pbXBhY3RfMjAxNw P-0000037-T02-IM3 P-0000037</span></span>
<span><span class="co">#&gt;           studyId     clinicalAttributeId                    value</span></span>
<span><span class="co">#&gt; 1 msk_impact_2017             CANCER_TYPE     Hepatobiliary Cancer</span></span>
<span><span class="co">#&gt; 2 msk_impact_2017    CANCER_TYPE_DETAILED Hepatocellular Carcinoma</span></span>
<span><span class="co">#&gt; 3 msk_impact_2017               DNA_INPUT                      250</span></span>
<span><span class="co">#&gt; 4 msk_impact_2017 FRACTION_GENOME_ALTERED                   0.2373</span></span>
<span><span class="co">#&gt; 5 msk_impact_2017          MATCHED_STATUS                  Matched</span></span>
<span><span class="co">#&gt; 6 msk_impact_2017          MUTATION_COUNT                        3</span></span></code></pre></div>
<p>The sample IDs here are in the <code>sampleId</code> column. You may
notice that this file is in “long” format and each sample has multiple
rows. Later we will convert this file to “wide” format to do QA checking
on attributes.</p>
<p>But the first thing you want to know is whether you are able to find
all of the <code>cbioportal_ids</code> from your
<code>clin_collab_df</code> file in the <code>clin_cbio</code>
file.<br>
To do this, use the <code><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff()</a></code> function:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="fu"><a href="https://generics.r-lib.org/reference/setops.html" class="external-link">setdiff</a></span><span class="op">(</span><span class="va">clin_collab_df</span><span class="op">$</span><span class="va">cbioportal_id</span>, <span class="va">clin_cbio</span><span class="op">$</span><span class="va">sampleId</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] "P-0066922-T02-IM7" "P-0070148-T01-IM5"</span></span></code></pre></div>
<p>So there are two sample ID’s from your clinical file
(<code>clin_collab_df</code>) that are currently not found in cBioPortal
(in your <code>clin_cbio</code> file). Include these in the list of
cBioPortal questions to ask your collaborator.</p>
<p>(Again, if you want to investigate a bit further, you could enter the
patient cBioPortal IDs as queries into the <a href="https://www.cbioportal.org/" class="external-link">cBioPortal website</a>.)</p>
</div>
<div class="section level3">
<h3 id="check-clinical-data-matches-cbioportal-database">Check Clinical Data Matches cBioPortal Database<a class="anchor" aria-label="anchor" href="#check-clinical-data-matches-cbioportal-database"></a>
</h3>
<p>Now we need to check whether clinical information in collaborator’s
file (<code>clin_collab_df</code>) matches clinical information in
cBioPortal (in your <code>clin_cbio</code> file).</p>
<p>Look at the <code>clin_collab_df</code> again:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">clin_collab_df</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 4</span></span></span>
<span><span class="co">#&gt;   cbioportal_id     ctype          primary_mets patient_id</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> P-0066922-T02-IM7 Hepatocellular Primary      P-0066922 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> P-0009540-T01-IM5 Hepatocellular Primary      P-0009540 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> P-0000182-T01-IM3 Hepatocellular Metastasis   P-0000182 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> P-0000037-T02-IM3 Hepatocellular Primary      P-0000037 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> P-0005357-T01-IM5 Hepatocellular Primary      P-0005357 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> P-0007773-T01-IM5 Hepatocellular Metastasis   P-0007773</span></span></code></pre></div>
<p>Aside from <code>cbioportal_id</code>, you have cancer type
(<code>ctype</code>) and sample type (<code>primary_mets</code>)
variables. Because it’s a hepatocellular cancer study, all of the
<code>ctype</code> values will be the same. To double check that, count
<code>ctype</code>:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clin_collab_df</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">ctype</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 2</span></span></span>
<span><span class="co">#&gt;   ctype              n</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Hepatocellular    80</span></span></code></pre></div>
<p>So the only variable you can check in this example is the
<code>primary_mets</code>. To see if the <code>clin_cbio</code> file has
an analogous variable to check, first see the attributes that are
available in it.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clin_cbio</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">clinicalAttributeId</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 18 × 2</span></span></span>
<span><span class="co">#&gt;    clinicalAttributeId            n</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> CANCER_TYPE                   78</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> CANCER_TYPE_DETAILED          78</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> DNA_INPUT                     78</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> FRACTION_GENOME_ALTERED       78</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> MATCHED_STATUS                78</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> METASTATIC_SITE                7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> MUTATION_COUNT                78</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> ONCOTREE_CODE                 78</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> PRIMARY_SITE                  78</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> SAMPLE_CLASS                  78</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">11</span> SAMPLE_COLLECTION_SOURCE      78</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">12</span> SAMPLE_COVERAGE               78</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">13</span> SAMPLE_TYPE                   78</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">14</span> SOMATIC_STATUS                78</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">15</span> SPECIMEN_PRESERVATION_TYPE    78</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">16</span> SPECIMEN_TYPE                 78</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">17</span> TMB_NONSYNONYMOUS             78</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">18</span> TUMOR_PURITY                  77</span></span></code></pre></div>
<p>To quickly see values associated with a particular attribute, filter
by the attribute and count the values. For example:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clin_cbio</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">clinicalAttributeId</span><span class="op">==</span><span class="st">"SAMPLE_TYPE"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/count.html" class="external-link">count</a></span><span class="op">(</span><span class="va">value</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 2 × 2</span></span></span>
<span><span class="co">#&gt;   value          n</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Metastasis     7</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Primary       71</span></span></code></pre></div>
<p>The attribute <code>SAMPLE_TYPE</code> looks like the appropriate
variable to check <code>primary_mets</code> against. To do this, we will
convert <code>clin_cbio</code> to “wide” form (only for the
<code>SAMPLE_TYPE</code> variable for now), merge it with
<code>clin_collab_df</code> and then cross-tabulate the 2 variables.</p>
<p>To convert <code>clin_cbio</code> to “wide” form:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">clin_cbio_wide</span> <span class="op">=</span> <span class="va">clin_cbio</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span> <span class="va">sampleId</span>, <span class="va">clinicalAttributeId</span>, <span class="va">value</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span> <span class="va">clinicalAttributeId</span> <span class="op">==</span> <span class="st">"SAMPLE_TYPE"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_wider.html" class="external-link">pivot_wider</a></span><span class="op">(</span>names_from <span class="op">=</span> <span class="va">clinicalAttributeId</span>, values_from <span class="op">=</span> <span class="va">value</span><span class="op">)</span></span></code></pre></div>
<p>Take a look at the “wide” file:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">clin_cbio_wide</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;            sampleId SAMPLE_TYPE</span></span>
<span><span class="co">#&gt; 1 P-0000037-T02-IM3     Primary</span></span>
<span><span class="co">#&gt; 2 P-0000182-T01-IM3  Metastasis</span></span>
<span><span class="co">#&gt; 3 P-0000218-T01-IM3  Metastasis</span></span>
<span><span class="co">#&gt; 4 P-0000228-T03-IM5     Primary</span></span>
<span><span class="co">#&gt; 5 P-0000587-T01-IM3  Metastasis</span></span>
<span><span class="co">#&gt; 6 P-0000829-T01-IM3     Primary</span></span></code></pre></div>
<p>Now to check the <code>primary_mets</code> variable from
<code>clin_collab_df</code> against the <code>SAMPLE_TYPE</code>
variable from <code>clin_cbio_wide</code>, merge the files and tabulate
the variables.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clin_merged</span> <span class="op">&lt;-</span> <span class="va">clin_cbio_wide</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">clin_collab_df</span>, by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"sampleId"</span> <span class="op">=</span> <span class="st">"cbioportal_id"</span><span class="op">)</span><span class="op">)</span> </span>
<span><span class="va">clin_merged</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">primary_mets</span>, <span class="va">SAMPLE_TYPE</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/table.html" class="external-link">table</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;             SAMPLE_TYPE</span></span>
<span><span class="co">#&gt; primary_mets Metastasis Primary</span></span>
<span><span class="co">#&gt;   Metastasis          7       1</span></span>
<span><span class="co">#&gt;   Primary             0      70</span></span></code></pre></div>
<p>There is 1 sample that has a value of “Metastasis” for the
<code>primary_mets</code> variable but “Primary” for the
<code>SAMPLE_TYPE</code> variable. To find the sample ID, filter:</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clin_merged</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">primary_mets</span> <span class="op">==</span> <span class="st">"Metastasis"</span> <span class="op">&amp;</span> <span class="va">SAMPLE_TYPE</span> <span class="op">==</span> <span class="st">"Primary"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 1 × 5</span></span></span>
<span><span class="co">#&gt;   sampleId          SAMPLE_TYPE ctype          primary_mets patient_id</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>          <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> P-0001324-T01-IM3 Primary     Hepatocellular Metastasis   P-0001324</span></span></code></pre></div>
<p>Include this sample in the list of questions for your collaborator.
Either she will need to update her clinical file with the correct value
or you/she will have to notify cBioPortal to update their database.</p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://www.karissawhiting.com/" class="external-link">Karissa Whiting</a>, Michael Curry, <a href="https://github.com/hfuchs5/" class="external-link">Hannah Fuchs</a>, Axel Martin, Arshi Arora.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
