<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Enables creation of a binary matrix from a mutation file with
a predefined list of samples (rows are samples and columns are genes) — create_gene_binary • gnomeR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Enables creation of a binary matrix from a mutation file with
a predefined list of samples (rows are samples and columns are genes) — create_gene_binary"><meta property="og:description" content="Enables creation of a binary matrix from a mutation file with
a predefined list of samples (rows are samples and columns are genes)"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gnomeR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/AxelitoMartin/gnomeR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Enables creation of a binary matrix from a mutation file with
a predefined list of samples (rows are samples and columns are genes)</h1>
    <small class="dont-index">Source: <a href="https://github.com/AxelitoMartin/gnomeR/blob/HEAD/R/binary_matrix.R" class="external-link"><code>R/binary_matrix.R</code></a></small>
    <div class="hidden name"><code>create_gene_binary.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Enables creation of a binary matrix from a mutation file with
a predefined list of samples (rows are samples and columns are genes)</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="fu">create_gene_binary</span><span class="op">(</span>
  samples <span class="op">=</span> <span class="cn">NULL</span>,
  mutation <span class="op">=</span> <span class="cn">NULL</span>,
  mut_type <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"omit_germline"</span>, <span class="st">"somatic_only"</span>, <span class="st">"germline_only"</span>, <span class="st">"all"</span><span class="op">)</span>,
  snp_only <span class="op">=</span> <span class="cn">FALSE</span>,
  include_silent <span class="op">=</span> <span class="cn">FALSE</span>,
  fusion <span class="op">=</span> <span class="cn">NULL</span>,
  cna <span class="op">=</span> <span class="cn">NULL</span>,
  cna_binary <span class="op">=</span> <span class="cn">TRUE</span>,
  cna_relax <span class="op">=</span> <span class="cn">FALSE</span>,
  specify_panel <span class="op">=</span> <span class="st">"no"</span>,
  rm_empty <span class="op">=</span> <span class="cn">FALSE</span>,
  recode_aliases <span class="op">=</span> <span class="cn">TRUE</span>,
  <span class="va">...</span>
<span class="op">)</span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>samples</dt>
<dd><p>a character vector specifying which samples should be included in the resulting data frame.
Default is NULL is which case all samples with an alteration in the mutation, cna or fusions file will be used. If you specify
a vector of samples that contain samples not in any of the passed genomic data frames, 0's (or NAs when appropriate if specifying a panel) will be
returned for every column of that patient row.</p></dd>
<dt>mutation</dt>
<dd><p>A data frame of mutations in the format of a maf file.</p></dd>
<dt>mut_type</dt>
<dd><p>The mutation type to be used. Options are "omit_germline", "somatic_only", "germline_only" or "all". Note "all" will
keep all mutations regardless of status (not recommended). Default is omit_germline which includes all somatic mutations, as well as any unknown mutation types (most of which are usually somatic)</p></dd>
<dt>snp_only</dt>
<dd><p>Boolean to rather the genetics events to be kept only to be SNPs (insertions and deletions will be removed).
Default is FALSE.</p></dd>
<dt>include_silent</dt>
<dd><p>Boolean to keep or remove all silent mutations. TRUE keeps, FALSE removes. Default is FALSE.</p></dd>
<dt>fusion</dt>
<dd><p>A data frame of fusions. If not NULL the outcome will be added to the matrix with columns ending in ".fus".
Default is NULL.</p></dd>
<dt>cna</dt>
<dd><p>A data frame of copy number alterations. If inputed the outcome will be added to the matrix with columns ending in ".del" and ".amp".
Default is NULL.</p></dd>
<dt>cna_binary</dt>
<dd><p>A boolean argument specifying if the cna events should be enforced as binary. In which case separate columns for
amplifications and deletions will be created.</p></dd>
<dt>cna_relax</dt>
<dd><p>By default this argument is set to FALSE, where only deep deletions (-2) and amplifications (2) will be annotated as events. When set to TRUE all deletions (-1 shallow and -2 deep) are counted as an event same for all gains (1 gain, 2 amplification) as an event.</p></dd>
<dt>specify_panel</dt>
<dd><p>a character vector of length 1 with panel id (see gnomeR::gene_panels for available panels), "impact", or "no", or a
data frame of <code>sample_id</code>-<code>panel_id</code> pairs specifying panels for which to insert NAs indicating that gene was not tested.
If a single panel id is passed, all genes that are not in that panel (see gnomeR::gene_panels) will be set to NA in results.
If <code>"impact"</code> passed, impact panel version will be inferred based on each sample_id and NA's annotated accordingly for each sample
If specific panel IDs passed, genes not tested in that panel will be should be considered. Default When TRUE NAs will fill the cells for genes
Default is "no" which returns data as is with no NA annotation.
If you wish to specify different panels for each sample, pass a data frame (with all samples included) with columns: <code>sample_id</code>, and <code>panel_id</code>. Each sample will be
annotated with NAs according to that specific panel.</p></dd>
<dt>rm_empty</dt>
<dd><p>boolean specifying if alteration columns with no events founds should be removed. Default is FALSE</p></dd>
<dt>recode_aliases</dt>
<dd><p>boolean specifying if automated gene name alias matching should be done. Default is TRUE. When TRUE
the function will check for genes that may have more than 1 name in your data using the aliases im gnomeR::impact_gene_info alias column</p></dd>
<dt>...</dt>
<dd><p>Further arguments passed to the oncokb() function such a token</p></dd>
</dl></div>
    <div id="value">
    <h2>Value</h2>
    <p>a data frame with sample_id and alteration binary columns with values of 0/1</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span class="va">mut.only</span> <span class="op">&lt;-</span> <span class="fu">create_gene_binary</span><span class="op">(</span>mutation <span class="op">=</span> <span class="va">mut</span><span class="op">)</span></span>
<span class="r-msg co"><span class="r-pr">#&gt;</span> <span style="color: #00BBBB;">ℹ</span> `samples` argument is `NULL`. We will infer your cohort inclusion and resulting data frame will include all samples with at least one alteration in <span style="color: #00BB00;">mutation</span>, <span style="color: #00BB00;">fusion</span> or <span style="color: #00BB00;">cna</span> data frames</span>
<span class="r-in"></span>
<span class="r-in"><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/character.html" class="external-link">as.character</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">mut</span><span class="op">$</span><span class="va">Tumor_Sample_Barcode</span><span class="op">)</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">200</span><span class="op">]</span></span>
<span class="r-in"></span>
<span class="r-in"><span class="va">bin.mut</span> <span class="op">&lt;-</span> <span class="fu">create_gene_binary</span><span class="op">(</span>samples <span class="op">=</span> <span class="va">samples</span>, mutation <span class="op">=</span> <span class="va">mut</span>,</span>
<span class="r-in">mut_type <span class="op">=</span> <span class="st">"omit_germline"</span> ,snp_only <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span class="r-in">include_silent <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
<span class="r-in"><span class="va">bin.mut</span> <span class="op">&lt;-</span> <span class="fu">create_gene_binary</span><span class="op">(</span>samples <span class="op">=</span> <span class="va">samples</span>, mutation <span class="op">=</span> <span class="va">mut</span>,</span>
<span class="r-in">mut_type <span class="op">=</span> <span class="st">"omit_germline"</span>, snp_only <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span class="r-in">include_silent <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span class="r-in">cna_relax <span class="op">=</span> <span class="cn">TRUE</span>, specify_panel <span class="op">=</span> <span class="st">"no"</span>, rm_empty <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Axel Martin, Arshi Arora, Karissa Whiting, Margie Hannum.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.2.</p>
</div>

      </footer></div>

  


  

  </body></html>

