<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Function reference â€¢ gnomeR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-index">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gnomeR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.1.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/MSKCC-Epi-Bio/gnomeR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index"><colgroup><col class="alias"><col class="title"></colgroup><tbody><tr><th colspan="2">
          <h2 id="all-functions">All functions <a href="#all-functions" class="anchor" aria-hidden="true"></a></h2>
          <p class="section-desc"></p>
        </th>
      </tr></tbody><tbody><tr><td>
          <p><code><a href="CNregions.mod.html">CNregions.mod()</a></code> </p>
        </td>
        <td><p>CNregions.mod
Modified CNregions function from the facets package to handle small sample sizes.</p></td>
      </tr><tr><td>
          <p><code><a href="add.perc.html">add.perc()</a></code> </p>
        </td>
        <td><p>Add a percentage to counts</p></td>
      </tr><tr><td>
          <p><code><a href="add_pathways.html">add_pathways()</a></code> </p>
        </td>
        <td><p>Pathway Alterations</p></td>
      </tr><tr><td>
          <p><code><a href="annotate_any_panel.html">annotate_any_panel()</a></code> </p>
        </td>
        <td><p>Annotate Missing Gene Values According to Specific Panels</p></td>
      </tr><tr><td>
          <p><code><a href="annotate_specific_panel.html">annotate_specific_panel()</a></code> </p>
        </td>
        <td><p>Utility function  to insert NA's According to Panel</p></td>
      </tr><tr><td>
          <p><code><a href="clin.patients.html">clin.patients</a></code> </p>
        </td>
        <td><p>An example of clinical patient level information file from
IMPACT cbioPortal dataset.</p></td>
      </tr><tr><td>
          <p><code><a href="clin.sample.html">clin.sample</a></code> </p>
        </td>
        <td><p>An example of clinical sample level information file from IMPACT cbioPortal dataset.</p></td>
      </tr><tr><td>
          <p><code><a href="cna.html">cna</a></code> </p>
        </td>
        <td><p>An example IMPACT cBioPortal mutation data set in API format</p></td>
      </tr><tr><td>
          <p><code><a href="consequence_map.html">consequence_map</a></code> </p>
        </td>
        <td><p>Consequence Map</p></td>
      </tr><tr><td>
          <p><code><a href="create_gene_binary.html">create_gene_binary()</a></code> </p>
        </td>
        <td><p>Enables creation of a binary matrix from a mutation file with
a predefined list of samples (rows are samples and columns are genes)</p></td>
      </tr><tr><td>
          <p><code><a href="curated_genes.html">curated_genes</a></code> </p>
        </td>
        <td><p>List of curated genes for IMPACT oncoKB annotation.</p></td>
      </tr><tr><td>
          <p><code><a href="dot-cna_gene_binary.html">.cna_gene_binary()</a></code> </p>
        </td>
        <td><p>Make Binary Matrix From CNA data frame</p></td>
      </tr><tr><td>
          <p><code><a href="dot-fusions_gene_binary.html">.fusions_gene_binary()</a></code> </p>
        </td>
        <td><p>Make Binary Matrix From Fusion data frame</p></td>
      </tr><tr><td>
          <p><code><a href="dot-mutations_gene_binary.html">.mutations_gene_binary()</a></code> </p>
        </td>
        <td><p>Make Binary Matrix From Mutation data frame</p></td>
      </tr><tr><td>
          <p><code><a href="dot-sum_alts_in_pathway.html">.sum_alts_in_pathway()</a></code> </p>
        </td>
        <td><p>Sum Alterations in a Pathway</p></td>
      </tr><tr><td>
          <p><code><a href="facets_dat.html">facets_dat()</a></code> </p>
        </td>
        <td><p>facets_dat</p></td>
      </tr><tr><td>
          <p><code><a href="facets_heatmap.html">facets_heatmap()</a></code> </p>
        </td>
        <td><p>facets_heatmap</p></td>
      </tr><tr><td>
          <p><code><a href="gene_panels.html">gene_panels</a></code> </p>
        </td>
        <td><p>Public Gene Panels on cBioPortal</p></td>
      </tr><tr><td>
          <p><code><a href="genie_gene_info.html">genie_gene_info</a></code> </p>
        </td>
        <td><p>GENIE consortium Gene Meta Data</p></td>
      </tr><tr><td>
          <p><code><a href="ggcomut.html">ggcomut()</a></code> </p>
        </td>
        <td><p>Comutation Heatmap of the Top Altered Genes</p></td>
      </tr><tr><td>
          <p><code><a href="gggenecor.html">gggenecor()</a></code> </p>
        </td>
        <td><p>Correlation Heatmap of the Top Altered Genes</p></td>
      </tr><tr><td>
          <p><code><a href="ggsamplevar.html">ggsamplevar()</a></code> </p>
        </td>
        <td><p>Histogram of Variants Per Sample Colored By Variant Classification</p></td>
      </tr><tr><td>
          <p><code><a href="ggtopgenes.html">ggtopgenes()</a></code> </p>
        </td>
        <td><p>Barplot of Most Frequently Altered Genes</p></td>
      </tr><tr><td>
          <p><code><a href="ggvarclass.html">ggvarclass()</a></code> </p>
        </td>
        <td><p>Barplot of Variant Classification Counts</p></td>
      </tr><tr><td>
          <p><code><a href="ggvartype.html">ggvartype()</a></code> </p>
        </td>
        <td><p>Barplot of Variant Type Counts</p></td>
      </tr><tr><td>
          <p><code><a href="gnomer_colors.html">gnomer_colors</a></code> </p>
        </td>
        <td><p>List of suggested color palettes for when you need a large palette</p></td>
      </tr><tr><td>
          <p><code><a href="gnomer_cols.html">gnomer_cols()</a></code> </p>
        </td>
        <td><p>Function to extract colors from <code>gnomer_colors</code> as hex codes</p></td>
      </tr><tr><td>
          <p><code><a href="gnomer_pal.html">gnomer_pal()</a></code> </p>
        </td>
        <td><p>Return function to interpolate a gnomeR color palette</p></td>
      </tr><tr><td>
          <p><code><a href="gnomer_palette.html">gnomer_palette()</a></code> </p>
        </td>
        <td><p>Access the colors in a gnomeR color palette</p></td>
      </tr><tr><td>
          <p><code><a href="gnomer_palettes.html">gnomer_palettes</a></code> </p>
        </td>
        <td><p>Complete list of gnomeR color palettes</p></td>
      </tr><tr><td>
          <p><code><a href="impact_alias_table.html">impact_alias_table</a></code> </p>
        </td>
        <td><p>IMPACT Alias Tables</p></td>
      </tr><tr><td>
          <p><code><a href="impact_gene_info.html">impact_gene_info</a></code> </p>
        </td>
        <td><p>IMPACT Gene Meta Data</p></td>
      </tr><tr><td>
          <p><code><a href="mutation_viz.html">mutation_viz()</a></code> </p>
        </td>
        <td><p>Creates a set of plot summarising a mutation file.</p></td>
      </tr><tr><td>
          <p><code><a href="mutations.html">mutations</a></code> </p>
        </td>
        <td><p>An example IMPACT cBioPortal mutation data set in API format</p></td>
      </tr><tr><td>
          <p><code><a href="names_df.html">names_df</a></code> </p>
        </td>
        <td><p>Data Frame of Column Names</p></td>
      </tr><tr><td>
          <p><code><a href="panel_names.html">panel_names</a></code> </p>
        </td>
        <td><p>GENIE panel names as found in synapse data</p></td>
      </tr><tr><td>
          <p><code><a href="pathways.html">pathways</a></code> </p>
        </td>
        <td><p>IMPACT Gene Pathways</p></td>
      </tr><tr><td>
          <p><code><a href="pivot_cna_longer.html">pivot_cna_longer()</a></code> </p>
        </td>
        <td><p>Reformat Wide CNA Data to Long</p></td>
      </tr><tr><td>
          <p><code><a href="recode_alias.html">recode_alias()</a></code> </p>
        </td>
        <td><p>Recode Hugo Symbol Column</p></td>
      </tr><tr><td>
          <p><code><a href="reformat_cna.html">reformat_cna()</a></code> </p>
        </td>
        <td><p>Reformat CNA from maf (long) version to wide version</p></td>
      </tr><tr><td>
          <p><code><a href="rename_columns.html">rename_columns()</a></code> </p>
        </td>
        <td><p>Rename columns from API results to work with gnomeR functions</p></td>
      </tr><tr><td>
          <p><code><a href="resolve_alias.html">resolve_alias()</a></code> </p>
        </td>
        <td><p>Resolve Hugo Symbol Names with Aliases</p></td>
      </tr><tr><td>
          <p><code><a href="sanitize_cna_input.html">sanitize_cna_input()</a></code> </p>
        </td>
        <td><p>Check CNA data frame to ensure columns are correct</p></td>
      </tr><tr><td>
          <p><code><a href="sanitize_fusion_input.html">sanitize_fusion_input()</a></code> </p>
        </td>
        <td><p>Check fusion data frame to ensure columns are correct</p></td>
      </tr><tr><td>
          <p><code><a href="sanitize_mutation_input.html">sanitize_mutation_input()</a></code> </p>
        </td>
        <td><p>Checks MAF input to ensure column names are correct and renamed genes are corrected</p></td>
      </tr><tr><td>
          <p><code><a href="scale_color_pancan.html">scale_color_pancan()</a></code> </p>
        </td>
        <td><p>Color scale creator to add gnomeR colors in ggplot</p></td>
      </tr><tr><td>
          <p><code><a href="scale_fill_pancan.html">scale_fill_pancan()</a></code> </p>
        </td>
        <td><p>Fill scale creator to add gnomeR colors in ggplot</p></td>
      </tr><tr><td>
          <p><code><a href="seg.html">seg</a></code> </p>
        </td>
        <td><p>A segmentation file from the cbioPortal datasets</p></td>
      </tr><tr><td>
          <p><code><a href="specify_impact_panels.html">specify_impact_panels()</a></code> </p>
        </td>
        <td><p>IMPACT Panel Annotation of NA's</p></td>
      </tr><tr><td>
          <p><code><a href="substrRight.html">substrRight()</a></code> </p>
        </td>
        <td><p>Utility Function to Extract SNV</p></td>
      </tr><tr><td>
          <p><code><a href="summarize_by_gene.html">summarize_by_gene()</a></code> </p>
        </td>
        <td><p>Simplify binary matrix to one column per gene that counts any alteration type as 1</p></td>
      </tr><tr><td>
          <p><code><a href="sv.html">sv</a></code> </p>
        </td>
        <td><p>An example IMPACT cBioPortal mutation data set in API format</p></td>
      </tr><tr><td>
          <p><code><a href="tbl_genomic.html">tbl_genomic()</a></code> </p>
        </td>
        <td><p>tbl_genomic</p></td>
      </tr><tr><td>
          <p><code><a href="tcga_genes.html">tcga_genes</a></code> </p>
        </td>
        <td><p>A vector of 19441 hugo symbols in TCGA</p></td>
      </tr><tr><td>
          <p><code><a href="tcga_samples.html">tcga_samples</a></code> </p>
        </td>
        <td><p>Data frame of all TCGA sample ids and their corresponding cancer type</p></td>
      </tr><tr><td>
          <p><code><a href="ti_341.html">ti_341</a></code> </p>
        </td>
        <td><p>Intervals sequenced in 341 panel</p></td>
      </tr><tr><td>
          <p><code><a href="ti_410.html">ti_410</a></code> </p>
        </td>
        <td><p>Intervals sequenced in 410 panel</p></td>
      </tr><tr><td>
          <p><code><a href="ti_468.html">ti_468</a></code> </p>
        </td>
        <td><p>Intervals sequenced in 468 panel</p></td>
      </tr><tr><td>
          <p><code><a href="which_impact_panel.html">which_impact_panel()</a></code> </p>
        </td>
        <td><p>provide a list of impact panels a provided gene is found within</p></td>
      </tr></tbody></table></div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Axel Martin, Karissa Whiting, Arshi Arora.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

