<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Function reference â€¢ gnomeR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">gnomeR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.1.9001</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/data-processing-vignette.html">Processing Data with {gnomeR}</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"><li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MSKCC-Epi-Bio/gnomeR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div>

    
  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Function reference</h1>
    </div>

    <div class="section level2">
      
      <h3 id="processing-data">Processing Data<a class="anchor" aria-label="anchor" href="#processing-data"></a></h3>
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="create_gene_binary.html">create_gene_binary()</a></code> 
        </dt>
        <dd>Enables creation of a binary matrix from a mutation, fusion or CNA file with
a predefined list of samples (rows are samples and columns are genes)</dd>
      </dl><dl><dt>
          
          <code><a href="summarize_by_gene.html">summarize_by_gene()</a></code> 
        </dt>
        <dd>Simplify binary matrix to one column per gene that counts any alteration type as 1</dd>
      </dl><dl><dt>
          
          <code><a href="pivot_cna_wider.html">pivot_cna_wider()</a></code> 
        </dt>
        <dd>Pivot CNA from maf (long) version to wide version</dd>
      </dl><dl><dt>
          
          <code><a href="pivot_cna_longer.html">pivot_cna_longer()</a></code> 
        </dt>
        <dd>Reformat Wide CNA Data to Long</dd>
      </dl><dl><dt>
          
          <code><a href="add_pathways.html">add_pathways()</a></code> 
        </dt>
        <dd>Pathway Alterations</dd>
      </dl><dl><dt>
          
          <code><a href="recode_alias.html">recode_alias()</a></code> 
        </dt>
        <dd>Recode Hugo Symbol Column</dd>
      </dl></div><div class="section level2">
      
      <h3 id="analyzing-data">Analyzing Data<a class="anchor" aria-label="anchor" href="#analyzing-data"></a></h3>
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="tbl_genomic.html">tbl_genomic()</a></code> 
        </dt>
        <dd>tbl_genomic</dd>
      </dl><dl><dt>
          
          <code><a href="mutation_viz.html">mutation_viz()</a></code> 
        </dt>
        <dd>Creates a set of plot summarising a mutation file.</dd>
      </dl><dl><dt>
          
          <code><a href="ggvarclass.html">ggvarclass()</a></code> 
        </dt>
        <dd>Barplot of Variant Classification Counts</dd>
      </dl><dl><dt>
          
          <code><a href="ggvartype.html">ggvartype()</a></code> 
        </dt>
        <dd>Barplot of Variant Type Counts</dd>
      </dl><dl><dt>
          
          <code><a href="ggsamplevar.html">ggsamplevar()</a></code> 
        </dt>
        <dd>Histogram of Variants Per Sample Colored By Variant Classification</dd>
      </dl><dl><dt>
          
          <code><a href="ggtopgenes.html">ggtopgenes()</a></code> 
        </dt>
        <dd>Barplot of Most Frequently Altered Genes</dd>
      </dl><dl><dt>
          
          <code><a href="gggenecor.html">gggenecor()</a></code> 
        </dt>
        <dd>Correlation Heatmap of the Top Altered Genes</dd>
      </dl><dl><dt>
          
          <code><a href="ggcomut.html">ggcomut()</a></code> 
        </dt>
        <dd>Comutation Heatmap of the Top Altered Genes</dd>
      </dl></div><div class="section level2">
      
      <h3 id="helper-functions">Helper Functions<a class="anchor" aria-label="anchor" href="#helper-functions"></a></h3>
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="which_impact_panel.html">which_impact_panel()</a></code> 
        </dt>
        <dd>provide a list of impact panels a provided gene is found within</dd>
      </dl><dl><dt>
          
          <code><a href="recode_cna.html">recode_cna()</a></code> 
        </dt>
        <dd>Internal function to recode numeric CNA alteration values to factor values</dd>
      </dl><dl><dt>
          
          <code><a href="rename_columns.html">rename_columns()</a></code> 
        </dt>
        <dd>Rename columns from API results to work with gnomeR functions</dd>
      </dl><dl><dt>
          
          <code><a href="resolve_alias.html">resolve_alias()</a></code> 
        </dt>
        <dd>Resolve Hugo Symbol Names with Aliases</dd>
      </dl></div><div class="section level2">
      
      <h3 id="color-palette">Color Palette<a class="anchor" aria-label="anchor" href="#color-palette"></a></h3>
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="gnomer_colors.html">gnomer_colors</a></code> 
        </dt>
        <dd>List of suggested color palettes for when you need a large palette</dd>
      </dl><dl><dt>
          
          <code><a href="gnomer_cols.html">gnomer_cols()</a></code> 
        </dt>
        <dd>Function to extract colors from <code>gnomer_colors</code> as hex codes</dd>
      </dl><dl><dt>
          
          <code><a href="gnomer_palettes.html">gnomer_palettes</a></code> 
        </dt>
        <dd>Complete list of gnomeR color palettes</dd>
      </dl><dl><dt>
          
          <code><a href="gnomer_palette.html">gnomer_palette()</a></code> 
        </dt>
        <dd>Access the colors in a gnomeR color palette</dd>
      </dl><dl><dt>
          
          <code><a href="gnomer_pal.html">gnomer_pal()</a></code> 
        </dt>
        <dd>Return function to interpolate a gnomeR color palette</dd>
      </dl><dl><dt>
          
          <code><a href="scale_color_pancan.html">scale_color_pancan()</a></code> 
        </dt>
        <dd>Color scale creator to add gnomeR colors in ggplot</dd>
      </dl><dl><dt>
          
          <code><a href="scale_fill_pancan.html">scale_fill_pancan()</a></code> 
        </dt>
        <dd>Fill scale creator to add gnomeR colors in ggplot</dd>
      </dl></div><div class="section level2">
      
      <h3 id="example-data-sets">Example Data Sets<a class="anchor" aria-label="anchor" href="#example-data-sets"></a></h3>
      

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="mutations.html">mutations</a></code> 
        </dt>
        <dd>An example IMPACT cBioPortal mutation data set in API format</dd>
      </dl><dl><dt>
          
          <code><a href="cna.html">cna</a></code> 
        </dt>
        <dd>An example IMPACT cBioPortal mutation data set in API format</dd>
      </dl><dl><dt>
          
          <code><a href="sv.html">sv</a></code> 
        </dt>
        <dd>An example IMPACT cBioPortal mutation data set in API format</dd>
      </dl><dl><dt>
          
          <code><a href="seg.html">seg</a></code> 
        </dt>
        <dd>A segmentation file from the cbioPortal datasets</dd>
      </dl><dl><dt>
          
          <code><a href="cna_wide.html">cna_wide</a></code> 
        </dt>
        <dd>An example IMPACT cBioPortal CNA in wide format</dd>
      </dl><dl><dt>
          
          <code><a href="consequence_map.html">consequence_map</a></code> 
        </dt>
        <dd>Consequence Map</dd>
      </dl><dl><dt>
          
          <code><a href="gene_panels.html">gene_panels</a></code> 
        </dt>
        <dd>Public Gene Panels on cBioPortal</dd>
      </dl><dl><dt>
          
          <code><a href="panel_names.html">panel_names</a></code> 
        </dt>
        <dd>GENIE panel names as found in synapse data</dd>
      </dl><dl><dt>
          
          <code><a href="genie_gene_info.html">genie_gene_info</a></code> 
        </dt>
        <dd>GENIE consortium Gene Meta Data</dd>
      </dl><dl><dt>
          
          <code><a href="names_df.html">names_df</a></code> 
        </dt>
        <dd>Data Frame of Column Names</dd>
      </dl><dl><dt>
          
          <code><a href="pathways.html">pathways</a></code> 
        </dt>
        <dd>IMPACT Gene Pathways</dd>
      </dl><dl><dt>
          
          <code><a href="impact_alias_table.html">impact_alias_table</a></code> 
        </dt>
        <dd>IMPACT Alias Tables</dd>
      </dl><dl><dt>
          
          <code><a href="impact_gene_info.html">impact_gene_info</a></code> 
        </dt>
        <dd>IMPACT Gene Meta Data</dd>
      </dl><dl><dt>
          
          <code><a href="ti_341.html">ti_341</a></code> 
        </dt>
        <dd>Intervals sequenced in 341 panel</dd>
      </dl><dl><dt>
          
          <code><a href="ti_410.html">ti_410</a></code> 
        </dt>
        <dd>Intervals sequenced in 410 panel</dd>
      </dl><dl><dt>
          
          <code><a href="ti_468.html">ti_468</a></code> 
        </dt>
        <dd>Intervals sequenced in 468 panel</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by <a href="https://www.karissawhiting.com/" class="external-link">Karissa Whiting</a>, Michael Curry, <a href="https://github.com/hfuchs5/" class="external-link">Hannah Fuchs</a>, Axel Martin, Arshi Arora.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

