<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>An example IMPACT cBioPortal mutation data set in API format — sv • gnomeR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="An example IMPACT cBioPortal mutation data set in API format — sv"><meta property="og:description" content="This set was created from a random sample of 200 patients from
publicly available prostate cancer data from cBioPortal. The file
is in API format."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gnomeR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.1.1.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/MSKCC-Epi-Bio/gnomeR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>An example IMPACT cBioPortal mutation data set in API format</h1>
    <small class="dont-index">Source: <a href="https://github.com/MSKCC-Epi-Bio/gnomeR/blob/HEAD/R/data.R" class="external-link"><code>R/data.R</code></a></small>
    <div class="hidden name"><code>sv.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>This set was created from a random sample of 200 patients from
publicly available prostate cancer data from cBioPortal. The file
is in API format.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="va">sv</span></span></code></pre></div>
    </div>

    <div id="format">
    <h2>Format</h2>
    <p>A data frame with structural variants from Abida et al. JCO Precis Oncol 2017.
Retrieved from cBioPortal.There are 94 observations and 29 variables.</p>
<p>A data frame with 94 rows and 44 variables:</p><dl><dt><code>uniqueSampleKey</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>uniquePatientKey</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>molecularProfileId</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>sampleId</code></dt>
<dd><p>MSKCC Sample ID</p></dd>

<dt><code>patientId</code></dt>
<dd><p>Patient ID</p></dd>

<dt><code>studyId</code></dt>
<dd><p>Indicator for Abida et al. 2017 study</p></dd>

<dt><code>site1EntrezGeneId</code></dt>
<dd><p>integer COLUMN_DESCRIPTION</p></dd>

<dt><code>site1HugoSymbol</code></dt>
<dd><p>Character w/ 31 levels,
Column containing HUGO symbols genes for first site of fusion</p></dd>

<dt><code>site1EnsemblTranscriptId</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>site1Chromosome</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>site1Position</code></dt>
<dd><p>integer COLUMN_DESCRIPTION</p></dd>

<dt><code>site1Contig</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>site1Region</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>site1RegionNumber</code></dt>
<dd><p>integer COLUMN_DESCRIPTION</p></dd>

<dt><code>site1Description</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>site2EntrezGeneId</code></dt>
<dd><p>integer COLUMN_DESCRIPTION</p></dd>

<dt><code>site2HugoSymbol</code></dt>
<dd><p>Character w/ 21 levels,
Column containing all HUGO symbols genes for second site of fusion</p></dd>

<dt><code>site2EnsemblTranscriptId</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>site2Chromosome</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>site2Position</code></dt>
<dd><p>integer COLUMN_DESCRIPTION</p></dd>

<dt><code>site2Contig</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>site2Region</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>site2RegionNumber</code></dt>
<dd><p>integer COLUMN_DESCRIPTION</p></dd>

<dt><code>site2Description</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>site2EffectOnFrame</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>ncbiBuild</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>dnaSupport</code></dt>
<dd><p>Factor, all are <code>yes</code> in this data</p></dd>

<dt><code>rnaSupport</code></dt>
<dd><p>Factor, all are <code>unknown</code> in this data</p></dd>

<dt><code>normalReadCount</code></dt>
<dd><p>integer COLUMN_DESCRIPTION</p></dd>

<dt><code>tumorReadCount</code></dt>
<dd><p>integer COLUMN_DESCRIPTION</p></dd>

<dt><code>normalVariantCount</code></dt>
<dd><p>integer COLUMN_DESCRIPTION</p></dd>

<dt><code>tumorVariantCount</code></dt>
<dd><p>integer COLUMN_DESCRIPTION</p></dd>

<dt><code>normalPairedEndReadCount</code></dt>
<dd><p>integer COLUMN_DESCRIPTION</p></dd>

<dt><code>tumorPairedEndReadCount</code></dt>
<dd><p>integer COLUMN_DESCRIPTION</p></dd>

<dt><code>normalSplitReadCount</code></dt>
<dd><p>integer COLUMN_DESCRIPTION</p></dd>

<dt><code>tumorSplitReadCount</code></dt>
<dd><p>integer COLUMN_DESCRIPTION</p></dd>

<dt><code>annotation</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>breakpointType</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>connectionType</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>eventInfo</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>variantClass</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>length</code></dt>
<dd><p>integer COLUMN_DESCRIPTION</p></dd>

<dt><code>comments</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>

<dt><code>svStatus</code></dt>
<dd><p>character COLUMN_DESCRIPTION</p></dd>


</dl></div>
    <div id="source">
    <h2>Source</h2>
    <p><a href="https://www.cbioportal.org/study/summary?id=prad_mskcc_2017" class="external-link">https://www.cbioportal.org/study/summary?id=prad_mskcc_2017</a></p>
    </div>

  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Karissa Whiting, Michael Curry, Hannah Fuchs, Axel Martin, Arshi Arora.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

