<!-- Generated by pkgdown: do not edit by hand -->
<!DOCTYPE html>
<html lang="en">
  <head>
  <meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">

<title>Function reference â€¢ gnomeR</title>


<!-- jquery -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
<!-- Bootstrap -->

<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous" />

<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script>

<!-- bootstrap-toc -->
<link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script>

<!-- Font Awesome icons -->
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous" />
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous" />

<!-- clipboard.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>

<!-- headroom.js -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script>

<!-- pkgdown -->
<link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script>




<meta property="og:title" content="Function reference" />




<!-- mathjax -->
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script>
<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script>

<!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->



  </head>

  <body data-spy="scroll" data-target="#toc">
    <div class="container template-reference-index">
      <header>
      <div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gnomeR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="../articles/API-tutorial.html">API-tutorial</a>
    </li>
    <li>
      <a href="../articles/Analizing-genomic-data.html">Analizing genomic data</a>
    </li>
    <li>
      <a href="../articles/Data-processing.html">Data-processing</a>
    </li>
    <li>
      <a href="../articles/Visualizations.html">Visualizations</a>
    </li>
    <li>
      <a href="../articles/gnomeR-vignette.html">gnomeR for IMPACT data</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/AxelitoMartin/gnomeR/">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
      
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header>

<div class="row">
  <div class="contents col-md-9">
    <div class="page-header">
      <h1>Reference</h1>
    </div>

    <table class="ref-index">

    <colgroup>
      
      <col class="alias" />
      <col class="title" />
    </colgroup>

    <tbody>
      <tr>
        <th colspan="2">
          <h2 id="section-all-functions" class="hasAnchor"><a href="#section-all-functions" class="anchor"></a>All functions</h2>
          <p class="section-desc"></p>
        </th>
      </tr>
      
      
    </tbody><tbody>
      
      
      <tr>
        
        <td>
          <p><code><a href="CNregions.mod.html">CNregions.mod()</a></code> </p>
        </td>
        <td><p>CNregions.mod
Modified CNregions function from the facets package to handle small sample sizes.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="binmat.html">binmat()</a></code> </p>
        </td>
        <td><p>binmat
Enables creation of a binary matrix from a maf file with
a predefined list of patients (rows are patients and columns are genes)</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="check_maf_column.html">check_maf_column()</a></code> </p>
        </td>
        <td><p>Check Columns of MAF</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="check_maf_input.html">check_maf_input()</a></code> </p>
        </td>
        <td><p>Checks MAF input to ensure column names are correct and renamed genes are corrected</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="clin.patients.html">clin.patients</a></code> </p>
        </td>
        <td><p>An example of clinical patient level information file from
IMPACT cbioPortal dataset.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="clin.sample.html">clin.sample</a></code> </p>
        </td>
        <td><p>An example of clinical sample level information file from IMPACT cbioPortal dataset.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="cna.html">cna</a></code> </p>
        </td>
        <td><p>An example copy number alteration raw calls file from IMPACT cbioPortal dataset</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="curated_genes.html">curated_genes</a></code> </p>
        </td>
        <td><p>List of curated genes for IMPACT oncoKB annotation.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="custom_pathway.html">custom_pathway()</a></code> </p>
        </td>
        <td><p>custom_pathway
Enables creation of a custom pathway binary matrix from a binmat() `object`</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="dat.oncoPrint.html">dat.oncoPrint()</a></code> </p>
        </td>
        <td><p>dat.oncoPrint
Enables creation of a matrix used to generate an OncoPrint heatmap.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="facets.dat.html">facets.dat()</a></code> </p>
        </td>
        <td><p>facets.dat</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="facets.heatmap.html">facets.heatmap()</a></code> </p>
        </td>
        <td><p>facets.heatmap</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="fusion.html">fusion</a></code> </p>
        </td>
        <td><p>An example of fusions calls file from IMPACT cbioPortal dataset</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="g.impact.html">g.impact</a></code> </p>
        </td>
        <td><p>List of curated genes for IMPACT</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="gen.summary.html">gen.summary()</a></code> </p>
        </td>
        <td><p>gen.summary</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ggcomut.html">ggcomut()</a></code> </p>
        </td>
        <td><p>Co-mutation Heatmap of the Top Altered Genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="gggenecor.html">gggenecor()</a></code> </p>
        </td>
        <td><p>Correlation Heatmap of the Top Altered Genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ggheatmap.html">ggheatmap()</a></code> </p>
        </td>
        <td><p>Heatmap of all events after binmat</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ggsamplevar.html">ggsamplevar()</a></code> </p>
        </td>
        <td><p>Histogram of Variants Per Sample Colored By Variant Classification</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ggsnvclass.html">ggsnvclass()</a></code> </p>
        </td>
        <td><p>Histogram of SNV class Counts</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ggtopgenes.html">ggtopgenes()</a></code> </p>
        </td>
        <td><p>Barplot of Most Frequently Altered Genes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ggvarclass.html">ggvarclass()</a></code> </p>
        </td>
        <td><p>Barplot of Variant Classification Counts</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="ggvartype.html">ggvartype()</a></code> </p>
        </td>
        <td><p>Barplot of Variant Type Counts</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="gnomer_colors.html">gnomer_colors</a></code> </p>
        </td>
        <td><p>List of suggested color palettes for when you need a large palette</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="gnomer_cols.html">gnomer_cols()</a></code> </p>
        </td>
        <td><p>Function to extract colors from <code>gnomer_colors</code> as hex codes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="gnomer_pal.html">gnomer_pal()</a></code> </p>
        </td>
        <td><p>Return function to interpolate a gnomeR color palette</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="gnomer_palette.html">gnomer_palette()</a></code> </p>
        </td>
        <td><p>Access the colors in a gnomeR color palette</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="gnomer_palettes.html">gnomer_palettes</a></code> </p>
        </td>
        <td><p>Complete list of gnomeR color palettes</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="impact_gene_info.html">impact_gene_info</a></code> </p>
        </td>
        <td><p>IMPACT Gene Meta Data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="impact_genes.html">impact_genes</a></code> </p>
        </td>
        <td><p>List of genes entrezID for the API</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="maf_viz.html">maf_viz()</a></code> </p>
        </td>
        <td><p>Creates a set of plot summarising a maf file.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="mut.html">mut</a></code> </p>
        </td>
        <td><p>An example maf file from IMPACT cbioPortal dataset</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="oncoKB_annotate.html">oncoKB_annotate()</a></code> </p>
        </td>
        <td><p>oncoKB_annotate
OncoKB annotator for MAF files.</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="oncokb.html">oncokb()</a></code> </p>
        </td>
        <td><p>OncoKB annotate</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="plot_oncoPrint.html">plot_oncoPrint()</a></code> </p>
        </td>
        <td><p>plot_oncoPrint
Creates the OncoPrint corresponding to the inputted genetic data</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="scale_color_pancan.html">scale_color_pancan()</a></code> </p>
        </td>
        <td><p>Color scale creator to add gnomeR colors in ggplot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="scale_fill_pancan.html">scale_fill_pancan()</a></code> </p>
        </td>
        <td><p>Fill scale creator to add gnomeR colors in ggplot</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="seg.html">seg</a></code> </p>
        </td>
        <td><p>A segmentation file from the cbioPortal datasets</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="substrRight.html">substrRight()</a></code> </p>
        </td>
        <td><p>Utility Function to Extract SNV</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="tcga_genes.html">tcga_genes</a></code> </p>
        </td>
        <td><p>A vector of 19441 hugo symbols in TCGA</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="tcga_samples.html">tcga_samples</a></code> </p>
        </td>
        <td><p>Data frame of all TCGA sample ids and their corresponding cancer type</p></td>
      </tr><tr>
        
        <td>
          <p><code><a href="uni.cox.html">uni.cox()</a></code> </p>
        </td>
        <td><p>uni.cox
Performs univariate cox proportional hazard model on every feature</p></td>
      </tr>
    </tbody>
    </table>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top">
      <h2 data-toc-skip>Contents</h2>
    </nav>
  </div>
</div>


      <footer>
      <div class="copyright">
  <p>Developed by Axel Martin, Arshi Arora, Karissa Whiting, Margie Hannum.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
   </div>

  


  </body>
</html>


