<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="gnomeR">
<title>Processing Data with {gnomeR} • gnomeR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><link href="../deps/Atkinson_Hyperlegible-0.4.3/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Processing Data with {gnomeR}">
<meta property="og:description" content="gnomeR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">gnomeR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.1.9001</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/data-processing-vignette.html">Processing Data with {gnomeR}</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MSKCC-Epi-Bio/gnomeR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Processing Data with {gnomeR}</h1>
                        <h4 data-toc-skip class="author">Akriti Mishra,
Karissa Whiting</h4>
            
      
      <small class="dont-index">Source: <a href="https://github.com/MSKCC-Epi-Bio/gnomeR/blob/HEAD/vignettes/data-processing-vignette.Rmd" class="external-link"><code>vignettes/data-processing-vignette.Rmd</code></a></small>
      <div class="d-none name"><code>data-processing-vignette.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This this vignette we will walk through a data example to show
available {gnomeR} data processing, visualization and analysis
functions. We will also outline some common pitfalls and format
inconsistencies you may encounter when working with mutation, CNA or
structural variant data, and show some {gnomeR} helper functions that
can help you manage these inconsistencies.</p>
</div>
<div class="section level2">
<h2 id="setting-up">Setting up<a class="anchor" aria-label="anchor" href="#setting-up"></a>
</h2>
<p>Make sure {gnomeR} is installed &amp; loaded. We will use dplyr for
the purposes of this vignette as well.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/MSKCC-Epi-Bio/gnomeR" class="external-link">gnomeR</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span></code></pre></div>
<p>To demonstrate {gnomeR} functions, we will be using a random sample
of 200 patients from a publicly available prostate cancer data pulled
from cBioPortal. Data on mutations, CNA and structural variants for
these patients are available in the gnomeR package
(<code><a href="../reference/mutations.html">gnomeR::mutations</a></code>, <code><a href="../reference/cna.html">gnomeR::cna</a></code>,
<code><a href="../reference/sv.html">gnomeR::sv</a></code>).</p>
<p>Note: To access data from cBioPortal, you can use the {cbioportalR}
package: <a href="https://github.com/karissawhiting/cbioportalR" class="external-link uri">https://github.com/karissawhiting/cbioportalR</a>.</p>
</div>
<div class="section level2">
<h2 id="data-formats">Data Formats<a class="anchor" aria-label="anchor" href="#data-formats"></a>
</h2>
<p>Mutation, CNA or fusion data may be formatting differently depending
on where you source it. Below we outline some differences you may
encounter when downloading data from the [cBioPortal website] (<a href="https://www.cbioportal.org/" class="external-link uri">https://www.cbioportal.org/</a>) versus pulling it via the
<a href="https://github.com/karissawhiting/cbioportalR" class="external-link">cBioPortal
API</a>, and review the important/required columns for each. Example
data in this package was pulled using the API.</p>
<p>See <a href="https://docs.cbioportal.org/file-formats/" class="external-link">cBioPortal
documentation</a> for more details on different file formats available
supported on cBioPortal, and their data schema and coding.</p>
<div class="section level3">
<h3 id="mutation-data">Mutation Data<a class="anchor" aria-label="anchor" href="#mutation-data"></a>
</h3>
<p>The most common mutation data format is the <a href="https://docs.gdc.cancer.gov/Data/File_Formats/MAF_Format/" class="external-link">Mutation
Annotation Format (MAF)</a> created as part of The Cancer Genome Atlas
(TCGA) project. Each row in a maf file represents a mutation on a
specific gene for a specific sample, therefore there are usually several
rows per sample (one for each mutation). To use MAF files, you need at
minimum a sample ID column and a hugo symbol column, though often
additional information like mutation type or location are also often
necessary.</p>
<p>MAF formats are fairly consistent across sources, however if you
download the raw data from a virtual study on the cBioPortal website
using the download button and compare with the mutation data imported
from the API, you might notice some differences in the variables names.
For instance:</p>
<pre><code>* `Tumor_Sample_Barcode` is called `sampleId` in API data
* The `Hugo_symbol` is called `hugoGeneSymbol` 
* `Variant_Classification` is called `mutationType`
* `HGVSp_Short` is called `proteinChange`
* `Chromosome` is called 'chr'</code></pre>
<p>Some of the other variables are named differently as well but those
differences are more intuitive to map the variables. You can refer to
<code><a href="../reference/names_df.html">gnomeR::names_df</a></code> for more information on MAF variables.</p>
<p>Luckily, most {gnomeR} functions use the data dictionary in
<code><a href="../reference/names_df.html">gnomeR::names_df</a></code> to automatically recognize the most common
MAF variable names and turn them into clean, snakecase names in
resulting dataframes. For example:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">gnomeR</span><span class="fu">::</span><span class="va"><a href="../reference/mutations.html">mutations</a></span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "hugoGeneSymbol"        "entrezGeneId"          "uniqueSampleKey"      </span></span>
<span><span class="co">#&gt;  [4] "uniquePatientKey"      "molecularProfileId"    "sampleId"             </span></span>
<span><span class="co">#&gt;  [7] "patientId"             "studyId"               "center"               </span></span>
<span><span class="co">#&gt; [10] "mutationStatus"        "validationStatus"      "startPosition"        </span></span>
<span><span class="co">#&gt; [13] "endPosition"           "referenceAllele"       "proteinChange"        </span></span>
<span><span class="co">#&gt; [16] "mutationType"          "functionalImpactScore" "fisValue"             </span></span>
<span><span class="co">#&gt; [19] "linkXvar"              "linkPdb"               "linkMsa"              </span></span>
<span><span class="co">#&gt; [22] "ncbiBuild"             "variantType"           "keyword"              </span></span>
<span><span class="co">#&gt; [25] "chr"                   "variantAllele"         "refseqMrnaId"         </span></span>
<span><span class="co">#&gt; [28] "proteinPosStart"       "proteinPosEnd"</span></span></code></pre></div>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/rename_columns.html">rename_columns</a></span><span class="op">(</span><span class="fu">gnomeR</span><span class="fu">::</span><span class="va"><a href="../reference/mutations.html">mutations</a></span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "hugo_symbol"            "entrez_gene_id"         "uniqueSampleKey"       </span></span>
<span><span class="co">#&gt;  [4] "uniquePatientKey"       "molecular_profile_id"   "sample_id"             </span></span>
<span><span class="co">#&gt;  [7] "patient_id"             "study_id"               "center"                </span></span>
<span><span class="co">#&gt; [10] "mutation_status"        "validation_status"      "start_position"        </span></span>
<span><span class="co">#&gt; [13] "end_position"           "reference_allele"       "hgv_sp_short"          </span></span>
<span><span class="co">#&gt; [16] "variant_classification" "functionalImpactScore"  "fisValue"              </span></span>
<span><span class="co">#&gt; [19] "linkXvar"               "linkPdb"                "linkMsa"               </span></span>
<span><span class="co">#&gt; [22] "ncbi_build"             "variant_type"           "keyword"               </span></span>
<span><span class="co">#&gt; [25] "chromosome"             "allele"                 "refseqMrnaId"          </span></span>
<span><span class="co">#&gt; [28] "protein_pos_start"      "protein_pos_end"</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="cna-data">CNA data<a class="anchor" aria-label="anchor" href="#cna-data"></a>
</h3>
<p>The discrete copy number data from cBioPortal contains values that
would be derived from copy-number analysis algorithms like GISTIC 2.0 or
RAE. CNA data is often presented in a long or wide format:</p>
<pre><code>* Long format - Each row is a CNA event for a given gene and sample, therefore samples often have multiple rows (one per gene CNA event). This is most common format you will receive when downloading data using the API.
  </code></pre>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">gnomeR</span><span class="fu">::</span><span class="va"><a href="../reference/cna.html">cna</a></span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, <span class="op">]</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 9</span></span></span>
<span><span class="co">#&gt;   hugoGeneSymbol entre…¹ uniqu…² uniqu…³ molec…⁴ sampl…⁵ patie…⁶ studyId alter…⁷</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> PTPRS             <span style="text-decoration: underline;">5</span>802 UC0wMD… UC0wMD… prad_m… P-0001… P-0001… prad_m…      -<span style="color: #BB0000;">2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> AR                 367 UC0wMD… UC0wMD… prad_m… P-0001… P-0001… prad_m…       2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> PIK3R1            <span style="text-decoration: underline;">5</span>295 UC0wMD… UC0wMD… prad_m… P-0002… P-0002… prad_m…      -<span style="color: #BB0000;">2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> AR                 367 UC0wMD… UC0wMD… prad_m… P-0003… P-0003… prad_m…       2</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> PTEN              <span style="text-decoration: underline;">5</span>728 UC0wMD… UC0wMD… prad_m… P-0000… P-0000… prad_m…      -<span style="color: #BB0000;">2</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> B2M                567 UC0wMD… UC0wMD… prad_m… P-0000… P-0000… prad_m…      -<span style="color: #BB0000;">2</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with abbreviated variable names ¹​entrezGeneId, ²​uniqueSampleKey,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ³​uniquePatientKey, ⁴​molecularProfileId, ⁵​sampleId, ⁶​patientId, ⁷​alteration</span></span></span></code></pre></div>
<pre><code>* Wide format - One column per sample, one row per gene . Each row contains data on a specific gene CNA for a given sample as shown below. One sample can have several rows. This is most common format you will receive when downloading data from the cBioPortal web browser.  </code></pre>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">gnomeR</span><span class="fu">::</span><span class="va"><a href="../reference/cna_wide.html">cna_wide</a></span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">6</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">6</span><span class="op">]</span></span>
<span><span class="co">#&gt;   Hugo_Symbol P-0070637-T01-IM7 P-0042589-T01-IM6 P-0026544-T01-IM6</span></span>
<span><span class="co">#&gt; 1       SMAD4                 0                 0                -2</span></span>
<span><span class="co">#&gt; 2       CCND1                 0                 0                 0</span></span>
<span><span class="co">#&gt; 3         MYC                 0                 0                 0</span></span>
<span><span class="co">#&gt; 4        FGF4                 0                 0                 0</span></span>
<span><span class="co">#&gt; 5        FGF3                 0                 0                 0</span></span>
<span><span class="co">#&gt; 6       FGF19                 0                 0                 0</span></span>
<span><span class="co">#&gt;   P-0032011-T01-IM6 P-0049337-T01-IM6</span></span>
<span><span class="co">#&gt; 1                 0                 0</span></span>
<span><span class="co">#&gt; 2                 0                 2</span></span>
<span><span class="co">#&gt; 3                 0                 2</span></span>
<span><span class="co">#&gt; 4                 0                 2</span></span>
<span><span class="co">#&gt; 5                 0                 2</span></span>
<span><span class="co">#&gt; 6                 0                 2</span></span></code></pre></div>
<p>{gnomeR} features two helper functions to help you pivot from wide to
long data.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/pivot_cna_wider.html">pivot_cna_wider</a></span><span class="op">(</span><span class="fu">gnomeR</span><span class="fu">::</span><span class="va"><a href="../reference/cna.html">cna</a></span><span class="op">)</span></span>
<span><span class="fu"><a href="../reference/pivot_cna_longer.html">pivot_cna_longer</a></span><span class="op">(</span><span class="fu">gnomeR</span><span class="fu">::</span><span class="va">wide_cna</span><span class="op">)</span></span></code></pre></div>
<p>There may also be differences in the way CNA events are coded. Most
Events are coded in one of the following ways:</p>
<p>{gnomeR} automatically checks CNA data labels and recodes as needed
within functions. You can also use the <code><a href="../reference/recode_cna.html">recode_cna()</a></code>
function to do it yourself:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">x</span> <span class="op">&lt;-</span> <span class="fu">gnomeR</span><span class="fu">::</span><span class="va"><a href="../reference/cna.html">cna</a></span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>alteration_recoded <span class="op">=</span> <span class="fu"><a href="../reference/recode_cna.html">recode_cna</a></span><span class="op">(</span><span class="va">alteration</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="preparing-data-for-analysis">Preparing Data For Analysis<a class="anchor" aria-label="anchor" href="#preparing-data-for-analysis"></a>
</h2>
<div class="section level3">
<h3 id="process-data-with-create_gene_binary">Process Data with <code>create_gene_binary()</code><a class="anchor" aria-label="anchor" href="#process-data-with-create_gene_binary"></a>
</h3>
<p>Often a first step to analyzing genomic data is organizing it in an
event matrix. This matrix will have one row for each sample in your
cohort and one column for each type of genomic event. Each cell will
take value of <code>0</code> (no event on that gene/sample),
<code>1</code> (event on that gene/sample) or <code>NA</code> (missing
data or gene not tested on panel). The <code><a href="../reference/create_gene_binary.html">create_gene_binary()</a></code>
function helps you process your data into this format for use in
downstream analysis.</p>
<p>You can <code><a href="../reference/create_gene_binary.html">create_gene_binary()</a></code> from any single type of
data (mutation, CNA or fusion):</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mut_only</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_gene_binary.html">create_gene_binary</a></span><span class="op">(</span>mutation <span class="op">=</span> <span class="fu">gnomeR</span><span class="fu">::</span><span class="va"><a href="../reference/mutations.html">mutations</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> `samples` argument is `NULL`. We will infer your cohort inclusion and resulting data frame will include all samples with at least one alteration in <span style="color: #00BB00;">mutation</span>, <span style="color: #00BB00;">fusion</span> or <span style="color: #00BB00;">cna</span> data frames</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mut_only</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 228</span></span></span>
<span><span class="co">#&gt;   sample_id    PARP1  AKT1   ALK   APC    AR  ARAF  RHOA ZFHX3   ATM   ATR  ATRX</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>        <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> P-0001128-T…     1     1     0     0     0     0     0     0     0     0     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> P-0001859-T…     1     0     0     1     0     0     0     0     0     0     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> P-0001895-T…     1     0     1     0     0     0     0     0     0     0     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> P-0001845-T…     0     1     0     0     0     0     0     0     0     0     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> P-0005570-T…     0     1     0     0     0     0     0     0     0     0     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> P-0001768-T…     0     0     1     0     0     0     0     0     0     0     0</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># … with 216 more variables: BARD1 &lt;dbl&gt;, CCND1 &lt;dbl&gt;, BCL6 &lt;dbl&gt;, PRDM1 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   BLM &lt;dbl&gt;, BMPR1A &lt;dbl&gt;, FOXL2 &lt;dbl&gt;, BRCA1 &lt;dbl&gt;, BRAF &lt;dbl&gt;, BRCA2 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   RUNX1 &lt;dbl&gt;, CBL &lt;dbl&gt;, CD79B &lt;dbl&gt;, CDH1 &lt;dbl&gt;, CDK4 &lt;dbl&gt;, CDK8 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   CDKN1A &lt;dbl&gt;, CDKN2C &lt;dbl&gt;, CREBBP &lt;dbl&gt;, CSF1R &lt;dbl&gt;, CSF3R &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   CTLA4 &lt;dbl&gt;, CTNNB1 &lt;dbl&gt;, DAXX &lt;dbl&gt;, DNMT1 &lt;dbl&gt;, EGFR &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   EP300 &lt;dbl&gt;, EPHA3 &lt;dbl&gt;, EPHA5 &lt;dbl&gt;, EPHB1 &lt;dbl&gt;, ERBB2 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   ERBB3 &lt;dbl&gt;, ERBB4 &lt;dbl&gt;, ERCC2 &lt;dbl&gt;, ERCC3 &lt;dbl&gt;, ERCC5 &lt;dbl&gt;, …</span></span></span></code></pre></div>
<p>or you can process several types of alterations into a single matrix.
Supported data types are:</p>
<ul>
<li>mutations</li>
<li>copy number amplifications</li>
<li>copy number deletions</li>
<li>gene fusions</li>
</ul>
<p>When processing multiple types of alteration data, by default there
will be a separate column for each type of alteration on that gene. For
example, TP53 mutation will result in a 1 in the <code>TP53</code>
column, TP53 CNA amplification will results in a 1 in the
<code>TP53.Amp</code> column etc.</p>
<p>Note the use of the <code>samples</code> argument. This allows you to
specify exactly which samples are in your resulting data frame. Without
specifying samples, if a sample has no alterations it will be excluded
from the final matrix.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="fu">gnomeR</span><span class="fu">::</span><span class="va"><a href="../reference/mutations.html">mutations</a></span><span class="op">$</span><span class="va">sampleId</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span>
<span></span>
<span><span class="va">all_bin</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_gene_binary.html">create_gene_binary</a></span><span class="op">(</span></span>
<span>    samples <span class="op">=</span> <span class="va">samples</span>,</span>
<span>    mutation <span class="op">=</span> <span class="fu">gnomeR</span><span class="fu">::</span><span class="va"><a href="../reference/mutations.html">mutations</a></span>,</span>
<span>    cna <span class="op">=</span> <span class="fu">gnomeR</span><span class="fu">::</span><span class="va"><a href="../reference/cna.html">cna</a></span>,</span>
<span>    fusion <span class="op">=</span> <span class="fu">gnomeR</span><span class="fu">::</span><span class="va"><a href="../reference/sv.html">sv</a></span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">all_bin</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span> </span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 10</span></span></span>
<span><span class="co">#&gt;    sample_id         PARP1  AKT1   ALK   APC BRCA2 CTNNB1 EPHB1  FAT1  JAK1</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>             <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> P-0001128-T01-IM3     1     1     0     0     1      1     0     0     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> P-0001859-T01-IM3     1     0     0     1     0      0     1     0     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> P-0001895-T01-IM3     1     0     1     0     0      0     0     0     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> P-0001845-T01-IM3     0     1     0     0     0      0     0     0     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> P-0005570-T01-IM5     0     1     0     0     0      0     0     0     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> P-0001768-T01-IM3     0     0     1     0     0      0     0     0     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> P-0004508-T01-IM5     0     0     1     0     0      0     0     0     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> P-0002984-T01-IM3     0     0     1     0     0      0     0     0     1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> P-0000964-T02-IM3     0     0     0     1     1      0     0     1     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> P-0000964-T01-IM3     0     0     0     1     1      0     0     1     0</span></span></code></pre></div>
<p><strong>Notes on some helpful <code><a href="../reference/create_gene_binary.html">create_gene_binary()</a></code>
arguments:</strong></p>
<ul>
<li>
<code>mut_type</code>- by default, any germline mutations will be
omitted because data is often incomplete, but you can choose to leave
them in if needed.</li>
<li>
<code>specify_panel</code>- If you are working across a set of
samples that was sequenced on several different gene panels, this
argument will insert NAs for the genes that weren’t tested for any given
sample. You can pass a string <code>"impact"</code> indicating
automatically guessing panels and processsing IMPACT samples based on
ID, or you can pass a dataframe of sample IDs and panel IDs for more
fine grained control of NA annotation.</li>
<li>
<code>recode_aliases</code> - Sometimes genes have several accepted
names or change names over time. This can be an issue if genes are coded
under multiple names in studies, or if you are working across studies.
By default, this function will search for aliases for genes in your data
set and resolved them to their current most common name.</li>
</ul>
</div>
<div class="section level3">
<h3 id="collapse-data-with-summarize_by_gene">Collapse Data with <code>summarize_by_gene()</code><a class="anchor" aria-label="anchor" href="#collapse-data-with-summarize_by_gene"></a>
</h3>
<p>If you wish to combine all types of alterations for a given gene into
1 gene column you can use <code><a href="../reference/summarize_by_gene.html">summarize_by_gene()</a></code></p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># THIS DOESN"T WORK</span></span>
<span><span class="co"># dim(all_bin)</span></span>
<span><span class="co"># </span></span>
<span><span class="co"># all_bin %&gt;% </span></span>
<span><span class="co">#   summarize_by_gene()</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="analyzing-data">Analyzing Data<a class="anchor" aria-label="anchor" href="#analyzing-data"></a>
</h2>
<p>Once you have processed the data into a binary matrix, you may want
to visualize and summarize it with the following helper functions:</p>
<div class="section level3">
<h3 id="summarize-alterations-with-tbl_genomic">Summarize Alterations with <code>tbl_genomic()</code><a class="anchor" aria-label="anchor" href="#summarize-alterations-with-tbl_genomic"></a>
</h3>
<p>This function can be used to display gene summary tables. It is built
off the {gtsummary} package and therefore you can use most
customizations available int that package to customize the look of your
tables.</p>
<p>The <code>gene_binary</code> argument expects the binary matrix data
as generated by the <code><a href="../reference/create_gene_binary.html">create_gene_binary()</a></code> function. Example
below shows the summary using gene data for ten samples.</p>
<p><strong>Arguments to <code><a href="../reference/tbl_genomic.html">tbl_genomic()</a></code> can help customize
your tables:</strong></p>
<ul>
<li><p><code>gene_subset</code> argument can be used to specify the
genes that you want to summarize the data for.</p></li>
<li><p><code>freq_cutoff</code> argument shows the summarized data for
the genes that meet the cutoff.</p></li>
</ul>
<p>In the example below for <code>tb1</code>, if the percent gene (of
the total samples) is at least 5%, the gene is included in the summary
table.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">mutations</span><span class="op">$</span><span class="va">sampleId</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span><span class="op">]</span></span>
<span></span>
<span><span class="va">gene_binary</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_gene_binary.html">create_gene_binary</a></span><span class="op">(</span></span>
<span>  samples <span class="op">=</span> <span class="va">samples</span>,</span>
<span>  mutation <span class="op">=</span> <span class="va">mutations</span>,</span>
<span>  cna <span class="op">=</span> <span class="va">cna</span>,</span>
<span>  mut_type <span class="op">=</span> <span class="st">"somatic_only"</span>, snp_only <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  specify_panel <span class="op">=</span> <span class="st">"no"</span></span>
<span><span class="op">)</span></span>
<span> <span class="va">tb1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tbl_genomic.html">tbl_genomic</a></span><span class="op">(</span>gene_binary <span class="op">=</span> <span class="va">gene_binary</span>, freq_cutoff <span class="op">=</span> <span class="fl">.05</span><span class="op">)</span></span>
<span> <span class="va">tb2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/tbl_genomic.html">tbl_genomic</a></span><span class="op">(</span>gene_binary <span class="op">=</span> <span class="va">gene_binary</span>, gene_subset <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"PARP1"</span>, <span class="st">"ALK"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="annotate-gene-pathways-with-add_pathways">Annotate Gene Pathways with <code>add_pathways()</code><a class="anchor" aria-label="anchor" href="#annotate-gene-pathways-with-add_pathways"></a>
</h3>
<p>The <code><a href="../reference/add_pathways.html">add_pathways()</a></code> function allows you add columns to
your gene binary matrix that annotate custom gene pathways, or oncogenic
signaling pathways (add citation).</p>
<p>The function expects a binary matrix as obtained from the
<code>gene_binary()</code> function and will return a gene binary with
additional columns added for specified pathways.</p>
<p>There are a set of default pathways available in the package that can
be viewed using <code><a href="../reference/pathways.html">gnomeR::pathways</a></code></p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># available pathways</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="fu">gnomeR</span><span class="fu">::</span><span class="va"><a href="../reference/pathways.html">pathways</a></span><span class="op">)</span></span>
<span><span class="co">#&gt;  [1] "RTK/RAS"    "Nrf2"       "PI3K"       "TGFB"       "p53"       </span></span>
<span><span class="co">#&gt;  [6] "Wnt"        "Myc"        "Cell cycle" "Hippo"      "Notch"</span></span>
<span></span>
<span><span class="va">pathway_df</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/add_pathways.html">add_pathways</a></span><span class="op">(</span><span class="va">gene_binary</span>, pathways <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Notch"</span>, <span class="st">"p53"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="data-visualizations">Data Visualizations<a class="anchor" aria-label="anchor" href="#data-visualizations"></a>
</h3>
<p>The mutation_viz functions allows you to visualize data for the
variables related to variant classification, variant type, SNV class as
well as top variant genes.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="../reference/mutation_viz.html">mutation_viz</a></span><span class="op">(</span><span class="va">mutations</span><span class="op">)</span></span>
<span><span class="co">#&gt; $varclass</span></span></code></pre></div>
<p><img src="data-processing-vignette_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $vartype</span></span></code></pre>
<p><img src="data-processing-vignette_files/figure-html/unnamed-chunk-15-2.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $samplevar</span></span></code></pre>
<p><img src="data-processing-vignette_files/figure-html/unnamed-chunk-15-3.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $topgenes</span></span></code></pre>
<p><img src="data-processing-vignette_files/figure-html/unnamed-chunk-15-4.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $genecor</span></span></code></pre>
<p><img src="data-processing-vignette_files/figure-html/unnamed-chunk-15-5.png" width="700"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://www.karissawhiting.com/" class="external-link">Karissa Whiting</a>, Michael Curry, <a href="https://github.com/hfuchs5/" class="external-link">Hannah Fuchs</a>, Axel Martin, Arshi Arora.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
