<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="gnomeR">
<title>Analyzing GENIE BPC Data Using {gnomeR} • gnomeR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><link href="../deps/Atkinson_Hyperlegible-0.4.7/font.css" rel="stylesheet">
<!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Analyzing GENIE BPC Data Using {gnomeR}">
<meta property="og:description" content="gnomeR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-dark navbar-expand-lg bg-primary"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">gnomeR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.3.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/data-processing-vignette.html">Processing Data with {gnomeR}</a>
    <a class="dropdown-item" href="../articles/genie-bpc-vignette.html">Analyzing GENIE BPC Data Using {gnomeR}</a>
    <a class="dropdown-item" href="../articles/qa-impact-data.html">How to QA Your IMPACT Data</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../news/index.html">Changelog</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav">
<li class="nav-item">
  <a class="external-link nav-link" href="https://github.com/MSKCC-Epi-Bio/gnomeR/" aria-label="github">
    <span class="fab fa fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Analyzing GENIE BPC Data Using {gnomeR}</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/MSKCC-Epi-Bio/gnomeR/blob/HEAD/vignettes/genie-bpc-vignette.Rmd" class="external-link"><code>vignettes/genie-bpc-vignette.Rmd</code></a></small>
      <div class="d-none name"><code>genie-bpc-vignette.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p>This vignette will walk through how to apply {gnomeR} functions to
data from AACR Project Genomics Evidence Neoplasia Information Exchange
BioPharma Collaborative (GENIE BPC). A broad overview of AACR Project
GENIE BPC can be found <a href="https://www.aacr.org/about-the-aacr/newsroom/news-releases/aacr-project-genie-begins-five-year-collaborative-research-project-with-36-million-in-new-funding/" class="external-link">here</a>,
with details on the clinical data structure available on the <a href="https://genie-bpc.github.io/genieBPC/articles/clinical_data_structure_vignette.html" class="external-link">{genieBPC}
package website</a>.</p>
<p>For the purposes of this vignette, we will use the first publicly
available GENIE BPC data release of non-small cell lung cancer patients,
NSCLC v2.0-public.</p>
<p>Note that the GENIE BPC genomic data are unique in a few particular
ways:</p>
<ul>
<li>These data are heterogeneous such that they come from multiple
genomic panels from multiple institutions. Gene aliases across panels
may vary.</li>
<li>The GENIE BPC fusions and copy number alterations data may differ
from common data formats. Differences are described in the Data Formats
subsection below.</li>
</ul>
</div>
<div class="section level3">
<h3 id="data-access">Data Access<a class="anchor" aria-label="anchor" href="#data-access"></a>
</h3>
<p>To gain access to the GENIE BPC data, please follow the instructions
on the <a href="https://genie-bpc.github.io/genieBPC/articles/pull_data_synapse_vignette.html" class="external-link">{genieBPC}
<code>pull_data_synapse()</code> vignette</a> to register for a Synapse
account. Once your Synapse account is created and you authenticate
yourself using <code><a href="https://genie-bpc.github.io/genieBPC/reference/set_synapse_credentials.html" class="external-link">genieBPC::set_synapse_credentials()</a></code>, you’ll
be ready to pull the GENIE BPC clinical and genomic data from Synapse
into your local environment:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://genie-bpc.github.io/genieBPC/" class="external-link">genieBPC</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># if credentials are not stored in your R environment</span></span>
<span><span class="fu"><a href="https://genie-bpc.github.io/genieBPC/reference/set_synapse_credentials.html" class="external-link">set_synapse_credentials</a></span><span class="op">(</span>username <span class="op">=</span> <span class="st">"username"</span>, password <span class="op">=</span> <span class="st">"password"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="co"># if credentials are stored in your R environment</span></span>
<span><span class="fu"><a href="https://genie-bpc.github.io/genieBPC/reference/set_synapse_credentials.html" class="external-link">set_synapse_credentials</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> You are now connected to 'Synapse' as</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">bstgeniebpc@mskcc.org</span> for this R session!</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="obtain-genie-bpc-data">Obtain GENIE BPC Data<a class="anchor" aria-label="anchor" href="#obtain-genie-bpc-data"></a>
</h3>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># pull NSCLC v2.0-public data from Synapse into the R environment</span></span>
<span><span class="va">nsclc_2_0</span> <span class="op">=</span> <span class="fu"><a href="https://genie-bpc.github.io/genieBPC/reference/pull_data_synapse.html" class="external-link">pull_data_synapse</a></span><span class="op">(</span>cohort <span class="op">=</span> <span class="st">"NSCLC"</span>,</span>
<span>                              version <span class="op">=</span> <span class="st">"v2.0-public"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">pt_char</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">ca_dx_index</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">ca_dx_non_index</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">ca_drugs</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">prissmm_imaging</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">prissmm_pathology</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">prissmm_md</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">cpt</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">mutations_extended</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">fusions</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">cna</span> has been imported for <span style="color: #0000BB;">"NSCLC_v2.0"</span></span></span></code></pre></div>
<p>The resulting <code>nsclc_2_0</code> object is a nested list of
datasets, including the mutations, fusions, and copy number data.</p>
<ul>
<li><code>nsclc_2_0$NSCLC_v2.0$mutations_extended</code></li>
<li><code>nsclc_2_0$NSCLC_v2.0$fusions</code></li>
<li><code>nsclc_2_0$NSCLC_v2.0$cna</code></li>
</ul>
<p>Note that while the GENIE BPC clinical data are only available via
Synapse, the genomic data can be accessed via both Synapse and
cBioPortal. Using the {cbioportalR} package, users can pull the GENIE
BPC genomic data directly from cBioPortal:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/karissawhiting/cbioportalR" class="external-link">cbioportalR</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># connect to the GENIE instance of cBioPortal</span></span>
<span><span class="fu">cbioportalR</span><span class="fu">::</span><span class="fu"><a href="https://karissawhiting.github.io/cbioportalR/reference/set_cbioportal_db.html" class="external-link">set_cbioportal_db</a></span><span class="op">(</span><span class="st">"https://genie.cbioportal.org/api"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> You are successfully connected!</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">base_url</span> for this R session is now set to <span style="color: #0000BB;">"genie.cbioportal.org/api"</span></span></span>
<span></span>
<span><span class="co"># view list of available studies from this instance of the portal</span></span>
<span><span class="co"># NSCLC v2.0-public is: nsclc_public_genie_bpc</span></span>
<span><span class="fu"><a href="https://karissawhiting.github.io/cbioportalR/reference/available_studies.html" class="external-link">available_studies</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 7 × 13</span></span></span>
<span><span class="co">#&gt;   studyId  name  description publicStudy groups status importDate allSampleCount</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;lgl&gt;</span>       <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   <span style="color: #949494; font-style: italic;">&lt;int&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>               <span style="color: #949494; font-style: italic;">&lt;int&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> brca_ak… AACR… <span style="color: #949494;">"</span>Targeted … TRUE        GENIE…      0 2022-10-0…            428</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> glioma_… DFCI… <span style="color: #949494;">"</span>1,535 gli… TRUE        GENIE…      0 2023-04-1…           <span style="text-decoration: underline;">1</span>535</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> mbc_gen… Meta… <span style="color: #949494;">"</span>Breast tu… TRUE        GENIE…      0 2023-04-1…            856</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> genie_p… GENI… <span style="color: #949494;">"</span>GENIE v14… TRUE        GENIE…      0 2023-10-1…         <span style="text-decoration: underline;">183</span>302</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> crc_pub… GENI… <span style="color: #949494;">"</span>CRC cohor… TRUE        GENIE…      0 2023-10-1…           <span style="text-decoration: underline;">1</span>551</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> erbb2_g… ERBB… <span style="color: #949494;">"</span>ERBB2 coh… TRUE        GENIE…      0 2023-10-1…            315</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">7</span> nsclc_p… GENI… <span style="color: #949494;">"</span>NSCLC coh… TRUE        GENIE…      0 2023-10-1…           <span style="text-decoration: underline;">2</span>004</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 5 more variables: readPermission &lt;lgl&gt;, cancerTypeId &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   referenceGenome &lt;chr&gt;, pmid &lt;chr&gt;, citation &lt;chr&gt;</span></span></span></code></pre></div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># obtain genomic data for GENIE BPC NSCLC v2.0-public</span></span>
<span><span class="va">mutations_extended_2_0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://karissawhiting.github.io/cbioportalR/reference/get_mutations_by_study.html" class="external-link">get_mutations_by_study</a></span><span class="op">(</span><span class="st">"nsclc_public_genie_bpc"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"nsclc_public_genie_bpc_mutations"</span> molecular profile in the <span style="color: #0000BB;">"nsclc_public_genie_bpc"</span> study</span></span>
<span><span class="va">cna_2_0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://karissawhiting.github.io/cbioportalR/reference/get_cna_by_study.html" class="external-link">get_cna_by_study</a></span><span class="op">(</span><span class="st">"nsclc_public_genie_bpc"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"nsclc_public_genie_bpc_cna"</span> molecular profile in the <span style="color: #0000BB;">"nsclc_public_genie_bpc"</span> study</span></span>
<span><span class="va">fusions_2_0</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://karissawhiting.github.io/cbioportalR/reference/get_fusions_by_study.html" class="external-link">get_fusions_by_study</a></span><span class="op">(</span><span class="st">"nsclc_public_genie_bpc"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Returning all data for the <span style="color: #0000BB;">"nsclc_public_genie_bpc_structural_variants"</span> molecular profile in the <span style="color: #0000BB;">"nsclc_public_genie_bpc"</span> study</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="data-formats">Data Formats<a class="anchor" aria-label="anchor" href="#data-formats"></a>
</h3>
<p>The genomic data for GENIE BPC are stored both on Synapse and in
cBioPortal. The data structure differs depending on where the genomic
data are downloaded from. Therefore, the remainder of this vignette will
proceed by outlining the process of annotating genomic data separately
for genomic data downloaded from Synapse and genomic data downloaded
from cBioPortal.</p>
<div class="section level4">
<h4 id="differences-between-synapse-and-cbioportal-genomic-data">Differences Between Synapse and cBioPortal Genomic Data<a class="anchor" aria-label="anchor" href="#differences-between-synapse-and-cbioportal-genomic-data"></a>
</h4>
<p>Please note that pulling genomic GENIE data from Synapse using
<code><a href="https://genie-bpc.github.io/genieBPC/reference/pull_data_synapse.html" class="external-link">pull_data_synapse()</a></code> and pulling GENIE data from cBioPortal
may result in small differences in the data due to systematic
differences in the processing pipelines employed by Synapse and
cBioPortal. These differences may include:</p>
<ul>
<li><p>Data formatting - Some data sets (e.g. CNA files) may appear in
wide format in Synapse data versus long format in cBioPortal data, or
column attributes and names may appear sightly different (e.g. fusions
files).</p></li>
<li><p>Default filtering - By default, cBioPortal filters out Silent,
Intron, IGR, 3’UTR, 5’UTR, 3’Flank and 5’Flank, except for the promoter
mutations of the TERT gene. See <a href="https://docs.cbioportal.org/file-formats/#mutation-data" class="external-link">cBioPortal
documentation</a> for more details. These are retained in Synapse
processing pipelines.</p></li>
<li><p>Hugo Symbols - Some genes have more than one accepted Hugo Symbol
and may be referred to differently between data sources
(e.g. <code>NSD3</code> is an alias for <code>WHSC1L1</code>). Some
tools exist to help you resolve gene aliases across genomic data
sources. See <code><a href="../reference/recode_alias.html">gnomeR::recode_alias()</a></code>,
<code>cbioportal::get_alias()</code> and vignettes from the <a href="https://mskcc-epi-bio.github.io/gnomeR/">{gnomeR}</a> and <a href="https://www.karissawhiting.com/cbioportalR/" class="external-link">{cbioportalR}</a> for
more information on how to use these functions and work with gene
aliases.</p></li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="selecting-a-cohort-for-analysis">Selecting a Cohort for Analysis<a class="anchor" aria-label="anchor" href="#selecting-a-cohort-for-analysis"></a>
</h3>
<p>The following code chunk uses the
<code><a href="https://genie-bpc.github.io/genieBPC/reference/create_analytic_cohort.html" class="external-link">genieBPC::create_analytic_cohort()</a></code> to create an analytic
cohort of patients diagnosed with stage IV NSCLC of adenocarcinoma
histology. Then, for patients with multiple genomic samples, the
<code><a href="https://genie-bpc.github.io/genieBPC/reference/select_unique_ngs.html" class="external-link">genieBPC::select_unique_ngs()</a></code> function chooses the genomic
sample with OncoTree code LUAD (if available). For patients with
multiple samples with OncoTree code LUAD, we will select the metastatic
genomic sample. If any patients have multiple metastatic samples with
OncoTree code LUAD, take the latest of the samples.</p>
<p>Note: for patients with exactly one genomic sample, that unique
genomic sample will be returned <em>regardless of whether it meets the
argument criteria specified below</em>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># create analytic cohort of patients diagnosed with Stage IV adenocarcinoma</span></span>
<span><span class="va">nsclc_2_0_example</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://genie-bpc.github.io/genieBPC/reference/create_analytic_cohort.html" class="external-link">create_analytic_cohort</a></span><span class="op">(</span></span>
<span>  data_synapse <span class="op">=</span> <span class="va">nsclc_2_0</span><span class="op">$</span><span class="va">NSCLC_v2.0</span>,</span>
<span>  stage_dx <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"Stage IV"</span><span class="op">)</span>,</span>
<span>  histology <span class="op">=</span> <span class="st">"Adenocarcinoma"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># select unique NGS samples for this analytic cohort</span></span>
<span><span class="va">nsclc_2_0_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://genie-bpc.github.io/genieBPC/reference/select_unique_ngs.html" class="external-link">select_unique_ngs</a></span><span class="op">(</span></span>
<span>  data_cohort <span class="op">=</span> <span class="va">nsclc_2_0_example</span><span class="op">$</span><span class="va">cohort_ngs</span>,</span>
<span>  oncotree_code <span class="op">=</span> <span class="st">"LUAD"</span>,</span>
<span>  sample_type <span class="op">=</span> <span class="st">"Metastasis"</span>,</span>
<span>  min_max_time <span class="op">=</span> <span class="st">"max"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #00BB00;">63</span> patients with &gt; 1 next generation sequencing reports were identified</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> <span style="color: #0000BB;">62</span> of <span style="color: #00BB00;">63</span> had a unique NGS report selected based on given criteria.</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> <span style="color: #0000BB;">1</span> of <span style="color: #00BB00;">63</span> did not have a unique NGS report selected based on the selected criteria or by having the largest panel, so a NGS report was selected at random (be sure to set a seed for reproducbility!) See `attributes(&lt;your-results&gt;)$random_samples` to view these sample IDs.</span></span></code></pre></div>
<p>Create a dataframe of the corresponding panel and sample IDs:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># specify sample panels and IDs</span></span>
<span><span class="va">nsclc_2_0_sample_panels</span> <span class="op">&lt;-</span> <span class="va">nsclc_2_0_samples</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">cpt_seq_assay_id</span>, <span class="va">cpt_genie_sample_id</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>panel_id <span class="op">=</span> <span class="va">cpt_seq_assay_id</span>,</span>
<span>         sample_id <span class="op">=</span> <span class="va">cpt_genie_sample_id</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">panel_id</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="process-data-with-create_gene_binary">Process Data with <code>create_gene_binary()</code><a class="anchor" aria-label="anchor" href="#process-data-with-create_gene_binary"></a>
</h3>
<p>The <code><a href="../reference/create_gene_binary.html">create_gene_binary()</a></code> function takes inputs of
mutations, fusions, and CNA data and returns a binary matrix with the
alteration status for each gene, annotating missingness when genes were
not included on a next generation sequencing panel.</p>
<p>It is critical to utilize the <code>specify_panel</code> argument of
<code><a href="../reference/create_gene_binary.html">create_gene_binary()</a></code>. Samples included in GENIE BPC were
sequenced across multiple sequencing platforms, with the genes included
varying across panels. Without the <code>specify_panel</code> argument,
missingness will not be correctly annotated, and genes that were not
tested will be incorrectly documented as not being altered.</p>
<p>Note: you can optionally check and recode any older gene names to
their newer Hugo Symbol in your data set by passing the
<code>genie</code> option to
<code>create_gene_binary(recode_aliases=)</code>.</p>
<p><strong>Using the genomic data from Synapse:</strong></p>
<p>The fusions and CNA data as downloaded from Synapse require some
modifications prior to being supplied to the
<code><a href="../reference/create_gene_binary.html">gnomeR::create_gene_binary()</a></code> function.</p>
<p>First, the CNA file can be transposed to match the expected input for
<code><a href="../reference/create_gene_binary.html">create_gene_binary()</a></code> using
<code><a href="../reference/pivot_cna_longer.html">pivot_cna_longer()</a></code>:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># transpose CNA data from Synapse</span></span>
<span><span class="va">cna_synapse_long</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pivot_cna_longer.html">pivot_cna_longer</a></span><span class="op">(</span><span class="va">nsclc_2_0</span><span class="op">$</span><span class="va">NSCLC_v2.0</span><span class="op">$</span><span class="va">cna</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> Replacing all `.` to `-` in <span style="color: #00BB00;">sample_id</span> field (e.g. `P.0001930.T01.IM3` -&gt; `P-0001930-T01-IM3`).</span></span>
<span><span class="co">#&gt; To prevent this, use argument `clean_sample_ids = FALSE`</span></span></code></pre></div>
<p>Next, the fusions file can be transposed to match the expected input
for <code><a href="../reference/create_gene_binary.html">create_gene_binary()</a></code></p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># transpose fusions data from Synapse</span></span>
<span><span class="va">fusions_synapse_long</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/reformat_fusion.html">reformat_fusion</a></span><span class="op">(</span><span class="va">nsclc_2_0</span><span class="op">$</span><span class="va">NSCLC_v2.0</span><span class="op">$</span><span class="va">fusions</span><span class="op">)</span></span></code></pre></div>
<p>Finally, the reformatted genomic data can be supplied to
<code><a href="../reference/create_gene_binary.html">create_gene_binary()</a></code> to annotate genomic alterations for
patients in the analytic cohort of interest.</p>
<p>The CNA data as downloaded from cBioPortal only includes high level
CNA (-2, 2), so we will specify <code>high_level_cna_only = TRUE</code>
to be consistent with the results based on the genomic data as
downloaded from cBioPortal.</p>
<p>Additionally, we will use the built in
’genie<code>option to check gene aliases (see</code>?create_gene_binary`
for more info).</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nsclc_2_0_gen_dat_synapse</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/create_gene_binary.html">create_gene_binary</a></span><span class="op">(</span></span>
<span>    mutation <span class="op">=</span> <span class="va">nsclc_2_0</span><span class="op">$</span><span class="va">NSCLC_v2.0</span><span class="op">$</span><span class="va">mutations_extended</span>,</span>
<span>    cna <span class="op">=</span> <span class="va">cna_synapse_long</span>,</span>
<span>    high_level_cna_only <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>    fusion <span class="op">=</span> <span class="va">fusions_synapse_long</span>,</span>
<span>    samples <span class="op">=</span> <span class="va">nsclc_2_0_sample_panels</span><span class="op">$</span><span class="va">sample_id</span>,</span>
<span>    specify_panel <span class="op">=</span> <span class="va">nsclc_2_0_sample_panels</span>, </span>
<span>    recode_aliases <span class="op">=</span> <span class="st">"genie"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">#&gt; Warning: To ensure gene with multiple names/aliases are correctly grouped together, the</span></span>
<span><span class="co">#&gt; following genes in your dataframe have been recoded (if you are running</span></span>
<span><span class="co">#&gt; `create_gene_binary()` you can prevent this with `alias_table = FALSE`):</span></span>
<span><span class="co">#&gt; PARK2 recoded to PRKN</span></span>
<span><span class="co">#&gt; MRE11A recoded to MRE11</span></span>
<span><span class="co">#&gt; RFWD2 recoded to COP1</span></span>
<span><span class="co">#&gt; GNB2L1 recoded to RACK1</span></span>
<span><span class="co">#&gt; FAM46C recoded to TENT5C</span></span>
<span><span class="co">#&gt; WHSC1L1 recoded to NSD3</span></span>
<span><span class="co">#&gt; C17orf70 recoded to FAAP100</span></span>
<span><span class="co">#&gt; PVRL4 recoded to NECTIN4</span></span>
<span><span class="co">#&gt; HIST1H3C recoded to H3C3</span></span>
<span><span class="co">#&gt; BRE recoded to BABAM2</span></span>
<span><span class="co">#&gt; C1orf86 recoded to FAAP20</span></span>
<span><span class="co">#&gt; WHSC1 recoded to NSD2</span></span>
<span><span class="co">#&gt; HIST1H3B recoded to H3C2</span></span>
<span><span class="co">#&gt; TCEB1 recoded to ELOC</span></span>
<span><span class="co">#&gt; H3F3A recoded to H3-3A</span></span>
<span><span class="co">#&gt; HIST1H3G recoded to H3C8</span></span>
<span><span class="co">#&gt; HIST1H3D recoded to H3C4</span></span>
<span><span class="co">#&gt; HIST1H1C recoded to H1-2</span></span>
<span><span class="co">#&gt; HIST1H3I recoded to H3C11</span></span>
<span><span class="co">#&gt; PAK7 recoded to PAK5</span></span>
<span><span class="co">#&gt; HIST1H3E recoded to H3C6</span></span>
<span><span class="co">#&gt; HIST1H3H recoded to H3C10</span></span>
<span><span class="co">#&gt; HIST1H3F recoded to H3C7</span></span>
<span><span class="co">#&gt; FAM175A recoded to ABRAXAS1</span></span>
<span><span class="co">#&gt; HIST1H2BD recoded to H2BC5</span></span>
<span><span class="co">#&gt; HIST1H3J recoded to H3C12</span></span>
<span><span class="co">#&gt; HIST3H3 recoded to H3-4</span></span>
<span><span class="co">#&gt; H3F3C recoded to H3-5</span></span>
<span><span class="co">#&gt; FAM58A recoded to CCNQ</span></span>
<span><span class="co">#&gt; H3F3B recoded to H3-3B</span></span>
<span><span class="co">#&gt; SETD8 recoded to KMT5A</span></span>
<span><span class="co">#&gt; GPR124 recoded to ADGRA2</span></span>
<span><span class="co">#&gt; C11orf30 recoded to EMSY</span></span>
<span><span class="co">#&gt; WISP3 recoded to CCN6</span></span>
<span><span class="co">#&gt; HIST2H3C recoded to H3C14</span></span>
<span><span class="co">#&gt; HIST2H3D recoded to H3C13</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> 64 columns have all missing values. This may occur when there are genes in your data that are not in the specified panels (see `specify_panel` argument)</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> 7 `samples` had no alterations found in data sets (See `attr(&lt;your_df&gt;, 'zero_alteration_samples')` to view). These were retained in results as having 0 alterations.</span></span></code></pre></div>
<p><strong>Using the genomic data from cBioPortal:</strong></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nsclc_2_0_gen_dat_cbio</span> <span class="op">&lt;-</span></span>
<span>  <span class="fu"><a href="../reference/create_gene_binary.html">create_gene_binary</a></span><span class="op">(</span></span>
<span>    mutation <span class="op">=</span> <span class="va">mutations_extended_2_0</span>,</span>
<span>    cna <span class="op">=</span> <span class="va">cna_2_0</span>,</span>
<span>    fusion <span class="op">=</span> <span class="va">fusions_2_0</span>,</span>
<span>    samples <span class="op">=</span> <span class="va">nsclc_2_0_sample_panels</span><span class="op">$</span><span class="va">sample_id</span>,</span>
<span>    specify_panel <span class="op">=</span> <span class="va">nsclc_2_0_sample_panels</span>, </span>
<span>    recode_aliases <span class="op">=</span> <span class="st">"genie"</span></span>
<span>  <span class="op">)</span></span>
<span><span class="co">#&gt; Warning: To ensure gene with multiple names/aliases are correctly grouped together, the</span></span>
<span><span class="co">#&gt; following genes in your dataframe have been recoded (if you are running</span></span>
<span><span class="co">#&gt; `create_gene_binary()` you can prevent this with `alias_table = FALSE`):</span></span>
<span><span class="co">#&gt; HIST3H3 recoded to H3-4</span></span>
<span><span class="co">#&gt; HIST1H3B recoded to H3C2</span></span>
<span><span class="co">#&gt; HIST1H3I recoded to H3C11</span></span>
<span><span class="co">#&gt; HIST1H3D recoded to H3C4</span></span>
<span><span class="co">#&gt; HIST1H3F recoded to H3C7</span></span>
<span><span class="co">#&gt; HIST1H3C recoded to H3C3</span></span>
<span><span class="co">#&gt; HIST1H3G recoded to H3C8</span></span>
<span><span class="co">#&gt; H3F3C recoded to H3-5</span></span>
<span><span class="co">#&gt; HIST1H3E recoded to H3C6</span></span>
<span><span class="co">#&gt; HIST1H1C recoded to H1-2</span></span>
<span><span class="co">#&gt; H3F3B recoded to H3-3B</span></span>
<span><span class="co">#&gt; HIST1H3J recoded to H3C12</span></span>
<span><span class="co">#&gt; HIST1H2BD recoded to H2BC5</span></span>
<span><span class="co">#&gt; H3F3A recoded to H3-3A</span></span>
<span><span class="co">#&gt; HIST1H3H recoded to H3C10</span></span>
<span><span class="co">#&gt; HIST2H3D recoded to H3C13</span></span>
<span><span class="co">#&gt; HIST2H3C recoded to H3C14</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> 57 columns have all missing values. This may occur when there are genes in your data that are not in the specified panels (see `specify_panel` argument)</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> 7 `samples` had no alterations found in data sets (See `attr(&lt;your_df&gt;, 'zero_alteration_samples')` to view). These were retained in results as having 0 alterations.</span></span></code></pre></div>
<p>Binary genomic matrices created using the genomic data downloaded
from Synapse and cBioPortal should be equal. We will proceed using the
<code>nsclc_2_0_gen_dat_cbio</code> object.</p>
</div>
<div class="section level3">
<h3 id="collapse-data-with-summarize_by_gene">Collapse Data with <code>summarize_by_gene()</code><a class="anchor" aria-label="anchor" href="#collapse-data-with-summarize_by_gene"></a>
</h3>
<p>We can summarize the presence of any alteration event (mutation,
amplification, deletion, structural variant) with the
<code><a href="../reference/summarize_by_gene.html">summarize_by_gene()</a></code> function, such that each gene is a
column that captures the presence of any event regardless of alteration
type.</p>
<p>Summarizing the first 10 samples for KRAS alterations:</p>
<p><strong>Using the genomic data from Synapse:</strong></p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">nsclc_2_0_gen_dat_synapse</span><span class="op">[</span><span class="fl">1</span><span class="op">:</span><span class="fl">10</span>, <span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">KRAS</span>, <span class="va">KRAS.Amp</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/summarize_by_gene.html">summarize_by_gene</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 10 × 2</span></span></span>
<span><span class="co">#&gt;    sample_id                KRAS</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> GENIE-DFCI-005843-1704      0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> GENIE-DFCI-004022-1313      0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> GENIE-DFCI-000013-8840      1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> GENIE-DFCI-000136-6004      1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> GENIE-DFCI-000215-8010      0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> GENIE-DFCI-000260-11571     0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> GENIE-DFCI-000381-9526      0</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> GENIE-DFCI-000410-10003     1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> GENIE-DFCI-000454-5667      1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> GENIE-DFCI-000583-11175     0</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="analyzing-data">Analyzing Data<a class="anchor" aria-label="anchor" href="#analyzing-data"></a>
</h3>
<p>After the data have been transformed into a binary format, we can
create summaries and visualizations to better understand the data.</p>
<div class="section level4">
<h4 id="summarize-data-with-tbl_genomic">Summarize Data with <code>tbl_genomic()</code><a class="anchor" aria-label="anchor" href="#summarize-data-with-tbl_genomic"></a>
</h4>
<p>The <code><a href="../reference/tbl_genomic.html">tbl_genomic()</a></code> function summarizes the frequency of
alteration events from the binary data returned from
<code><a href="../reference/create_gene_binary.html">create_gene_binary()</a></code> or
<code><a href="../reference/summarize_by_gene.html">summarize_by_gene()</a></code>.</p>
<p><strong>Using the genomic data from Synapse:</strong></p>
<p>Summarizing the frequencies of KEAP1, STK11, and SMARCA4 alteration
events:</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">nsclc_2_0_gen_dat_synapse</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">KEAP1</span>, <span class="va">STK11</span>, <span class="va">SMARCA4</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/tbl_genomic.html">tbl_genomic</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div id="iaffeqqxcx" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#iaffeqqxcx table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#iaffeqqxcx thead, #iaffeqqxcx tbody, #iaffeqqxcx tfoot, #iaffeqqxcx tr, #iaffeqqxcx td, #iaffeqqxcx th {
  border-style: none;
}

#iaffeqqxcx p {
  margin: 0;
  padding: 0;
}

#iaffeqqxcx .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#iaffeqqxcx .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#iaffeqqxcx .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#iaffeqqxcx .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#iaffeqqxcx .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#iaffeqqxcx .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#iaffeqqxcx .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#iaffeqqxcx .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#iaffeqqxcx .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#iaffeqqxcx .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#iaffeqqxcx .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#iaffeqqxcx .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#iaffeqqxcx .gt_spanner_row {
  border-bottom-style: hidden;
}

#iaffeqqxcx .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#iaffeqqxcx .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#iaffeqqxcx .gt_from_md > :first-child {
  margin-top: 0;
}

#iaffeqqxcx .gt_from_md > :last-child {
  margin-bottom: 0;
}

#iaffeqqxcx .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#iaffeqqxcx .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#iaffeqqxcx .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#iaffeqqxcx .gt_row_group_first td {
  border-top-width: 2px;
}

#iaffeqqxcx .gt_row_group_first th {
  border-top-width: 2px;
}

#iaffeqqxcx .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#iaffeqqxcx .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#iaffeqqxcx .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#iaffeqqxcx .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#iaffeqqxcx .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#iaffeqqxcx .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#iaffeqqxcx .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#iaffeqqxcx .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#iaffeqqxcx .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#iaffeqqxcx .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#iaffeqqxcx .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#iaffeqqxcx .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#iaffeqqxcx .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#iaffeqqxcx .gt_left {
  text-align: left;
}

#iaffeqqxcx .gt_center {
  text-align: center;
}

#iaffeqqxcx .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#iaffeqqxcx .gt_font_normal {
  font-weight: normal;
}

#iaffeqqxcx .gt_font_bold {
  font-weight: bold;
}

#iaffeqqxcx .gt_font_italic {
  font-style: italic;
}

#iaffeqqxcx .gt_super {
  font-size: 65%;
}

#iaffeqqxcx .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#iaffeqqxcx .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#iaffeqqxcx .gt_indent_1 {
  text-indent: 5px;
}

#iaffeqqxcx .gt_indent_2 {
  text-indent: 10px;
}

#iaffeqqxcx .gt_indent_3 {
  text-indent: 15px;
}

#iaffeqqxcx .gt_indent_4 {
  text-indent: 20px;
}

#iaffeqqxcx .gt_indent_5 {
  text-indent: 25px;
}
</style>
<table class="table gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead><tr class="gt_col_headings">
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="&lt;strong&gt;Characteristic&lt;/strong&gt;"><strong>Characteristic</strong></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id='&lt;strong&gt;N = 547&lt;/strong&gt;&lt;span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;"&gt;&lt;sup&gt;1&lt;/sup&gt;&lt;/span&gt;'>
<strong>N = 547</strong><span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;"><sup>1</sup></span>
</th>
    </tr></thead>
<tbody class="gt_table_body">
<tr>
<td headers="label" class="gt_row gt_left">STK11</td>
<td headers="stat_0" class="gt_row gt_center">71 (15%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">67</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">KEAP1</td>
<td headers="stat_0" class="gt_row gt_center">65 (14%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">69</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">SMARCA4</td>
<td headers="stat_0" class="gt_row gt_center">49 (10%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">67</td>
</tr>
</tbody>
<tfoot class="gt_footnotes"><tr>
<td class="gt_footnote" colspan="2">
<span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;"><sup>1</sup></span> n (%)</td>
    </tr></tfoot>
</table>
</div>
<p>Users can subset their data set to only include genes above a certain
prevalence frequency threshold before passing to the function using the
<code><a href="../reference/subset_by_frequency.html">subset_by_frequency()</a></code> function.</p>
<p>Below, we summarize alteration events with &gt;=10% frequency:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">nsclc_2_0_gen_dat_synapse</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/subset_by_frequency.html">subset_by_frequency</a></span><span class="op">(</span>t <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/tbl_genomic.html">tbl_genomic</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div id="gdxfywihya" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#gdxfywihya table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#gdxfywihya thead, #gdxfywihya tbody, #gdxfywihya tfoot, #gdxfywihya tr, #gdxfywihya td, #gdxfywihya th {
  border-style: none;
}

#gdxfywihya p {
  margin: 0;
  padding: 0;
}

#gdxfywihya .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#gdxfywihya .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#gdxfywihya .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#gdxfywihya .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#gdxfywihya .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#gdxfywihya .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gdxfywihya .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#gdxfywihya .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#gdxfywihya .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#gdxfywihya .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#gdxfywihya .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#gdxfywihya .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#gdxfywihya .gt_spanner_row {
  border-bottom-style: hidden;
}

#gdxfywihya .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#gdxfywihya .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#gdxfywihya .gt_from_md > :first-child {
  margin-top: 0;
}

#gdxfywihya .gt_from_md > :last-child {
  margin-bottom: 0;
}

#gdxfywihya .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#gdxfywihya .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#gdxfywihya .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#gdxfywihya .gt_row_group_first td {
  border-top-width: 2px;
}

#gdxfywihya .gt_row_group_first th {
  border-top-width: 2px;
}

#gdxfywihya .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#gdxfywihya .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#gdxfywihya .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#gdxfywihya .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gdxfywihya .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#gdxfywihya .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#gdxfywihya .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#gdxfywihya .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#gdxfywihya .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#gdxfywihya .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#gdxfywihya .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#gdxfywihya .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#gdxfywihya .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#gdxfywihya .gt_left {
  text-align: left;
}

#gdxfywihya .gt_center {
  text-align: center;
}

#gdxfywihya .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#gdxfywihya .gt_font_normal {
  font-weight: normal;
}

#gdxfywihya .gt_font_bold {
  font-weight: bold;
}

#gdxfywihya .gt_font_italic {
  font-style: italic;
}

#gdxfywihya .gt_super {
  font-size: 65%;
}

#gdxfywihya .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#gdxfywihya .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#gdxfywihya .gt_indent_1 {
  text-indent: 5px;
}

#gdxfywihya .gt_indent_2 {
  text-indent: 10px;
}

#gdxfywihya .gt_indent_3 {
  text-indent: 15px;
}

#gdxfywihya .gt_indent_4 {
  text-indent: 20px;
}

#gdxfywihya .gt_indent_5 {
  text-indent: 25px;
}
</style>
<table class="table gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead><tr class="gt_col_headings">
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="&lt;strong&gt;Characteristic&lt;/strong&gt;"><strong>Characteristic</strong></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id='&lt;strong&gt;N = 547&lt;/strong&gt;&lt;span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;"&gt;&lt;sup&gt;1&lt;/sup&gt;&lt;/span&gt;'>
<strong>N = 547</strong><span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;"><sup>1</sup></span>
</th>
    </tr></thead>
<tbody class="gt_table_body">
<tr>
<td headers="label" class="gt_row gt_left">TP53</td>
<td headers="stat_0" class="gt_row gt_center">302 (55%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">EGFR</td>
<td headers="stat_0" class="gt_row gt_center">168 (31%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">KRAS</td>
<td headers="stat_0" class="gt_row gt_center">167 (31%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">LRP1B</td>
<td headers="stat_0" class="gt_row gt_center">5 (20%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">522</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">STK11</td>
<td headers="stat_0" class="gt_row gt_center">71 (15%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">67</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">KEAP1</td>
<td headers="stat_0" class="gt_row gt_center">65 (14%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">69</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">GRM3</td>
<td headers="stat_0" class="gt_row gt_center">3 (13%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">523</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">RANBP2</td>
<td headers="stat_0" class="gt_row gt_center">3 (13%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">523</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">SLIT2</td>
<td headers="stat_0" class="gt_row gt_center">3 (13%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">523</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">SPTA1</td>
<td headers="stat_0" class="gt_row gt_center">3 (13%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">523</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">GLI3</td>
<td headers="stat_0" class="gt_row gt_center">12 (12%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">444</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">KMT2D</td>
<td headers="stat_0" class="gt_row gt_center">53 (11%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">67</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">CDKN2A.Del</td>
<td headers="stat_0" class="gt_row gt_center">52 (11%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">67</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">PRKDC</td>
<td headers="stat_0" class="gt_row gt_center">22 (11%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">339</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">SMARCA4</td>
<td headers="stat_0" class="gt_row gt_center">49 (10%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">67</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">COL7A1</td>
<td headers="stat_0" class="gt_row gt_center">8 (10%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">467</td>
</tr>
</tbody>
<tfoot class="gt_footnotes"><tr>
<td class="gt_footnote" colspan="2">
<span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;"><sup>1</sup></span> n (%)</td>
    </tr></tfoot>
</table>
</div>
<p><strong>Using the genomic data from cBioPortal:</strong></p>
<p>Summarizing the frequencies of KEAP1, STK11, and SMARCA4 alteration
events:</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">nsclc_2_0_gen_dat_cbio</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">sample_id</span>, <span class="va">KEAP1</span>, <span class="va">STK11</span>, <span class="va">SMARCA4</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/tbl_genomic.html">tbl_genomic</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div id="jdiaydxzfc" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#jdiaydxzfc table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#jdiaydxzfc thead, #jdiaydxzfc tbody, #jdiaydxzfc tfoot, #jdiaydxzfc tr, #jdiaydxzfc td, #jdiaydxzfc th {
  border-style: none;
}

#jdiaydxzfc p {
  margin: 0;
  padding: 0;
}

#jdiaydxzfc .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#jdiaydxzfc .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#jdiaydxzfc .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#jdiaydxzfc .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#jdiaydxzfc .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#jdiaydxzfc .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jdiaydxzfc .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#jdiaydxzfc .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#jdiaydxzfc .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#jdiaydxzfc .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#jdiaydxzfc .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#jdiaydxzfc .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#jdiaydxzfc .gt_spanner_row {
  border-bottom-style: hidden;
}

#jdiaydxzfc .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#jdiaydxzfc .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#jdiaydxzfc .gt_from_md > :first-child {
  margin-top: 0;
}

#jdiaydxzfc .gt_from_md > :last-child {
  margin-bottom: 0;
}

#jdiaydxzfc .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#jdiaydxzfc .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#jdiaydxzfc .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#jdiaydxzfc .gt_row_group_first td {
  border-top-width: 2px;
}

#jdiaydxzfc .gt_row_group_first th {
  border-top-width: 2px;
}

#jdiaydxzfc .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#jdiaydxzfc .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#jdiaydxzfc .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#jdiaydxzfc .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jdiaydxzfc .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#jdiaydxzfc .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#jdiaydxzfc .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#jdiaydxzfc .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#jdiaydxzfc .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#jdiaydxzfc .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#jdiaydxzfc .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#jdiaydxzfc .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#jdiaydxzfc .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#jdiaydxzfc .gt_left {
  text-align: left;
}

#jdiaydxzfc .gt_center {
  text-align: center;
}

#jdiaydxzfc .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#jdiaydxzfc .gt_font_normal {
  font-weight: normal;
}

#jdiaydxzfc .gt_font_bold {
  font-weight: bold;
}

#jdiaydxzfc .gt_font_italic {
  font-style: italic;
}

#jdiaydxzfc .gt_super {
  font-size: 65%;
}

#jdiaydxzfc .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#jdiaydxzfc .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#jdiaydxzfc .gt_indent_1 {
  text-indent: 5px;
}

#jdiaydxzfc .gt_indent_2 {
  text-indent: 10px;
}

#jdiaydxzfc .gt_indent_3 {
  text-indent: 15px;
}

#jdiaydxzfc .gt_indent_4 {
  text-indent: 20px;
}

#jdiaydxzfc .gt_indent_5 {
  text-indent: 25px;
}
</style>
<table class="table gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead><tr class="gt_col_headings">
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="&lt;strong&gt;Characteristic&lt;/strong&gt;"><strong>Characteristic</strong></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id='&lt;strong&gt;N = 547&lt;/strong&gt;&lt;span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;"&gt;&lt;sup&gt;1&lt;/sup&gt;&lt;/span&gt;'>
<strong>N = 547</strong><span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;"><sup>1</sup></span>
</th>
    </tr></thead>
<tbody class="gt_table_body">
<tr>
<td headers="label" class="gt_row gt_left">STK11</td>
<td headers="stat_0" class="gt_row gt_center">71 (15%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">67</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">KEAP1</td>
<td headers="stat_0" class="gt_row gt_center">65 (14%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">69</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">SMARCA4</td>
<td headers="stat_0" class="gt_row gt_center">48 (10%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">67</td>
</tr>
</tbody>
<tfoot class="gt_footnotes"><tr>
<td class="gt_footnote" colspan="2">
<span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;"><sup>1</sup></span> n (%)</td>
    </tr></tfoot>
</table>
</div>
<p>Summarizing alteration events with &gt;=10% frequency:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">nsclc_2_0_gen_dat_cbio</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/subset_by_frequency.html">subset_by_frequency</a></span><span class="op">(</span>t <span class="op">=</span> <span class="fl">0.1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/tbl_genomic.html">tbl_genomic</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<div id="qtjzfccopy" style="padding-left:0px;padding-right:0px;padding-top:10px;padding-bottom:10px;overflow-x:auto;overflow-y:auto;width:auto;height:auto;">
<style>#qtjzfccopy table {
  font-family: system-ui, 'Segoe UI', Roboto, Helvetica, Arial, sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol', 'Noto Color Emoji';
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

#qtjzfccopy thead, #qtjzfccopy tbody, #qtjzfccopy tfoot, #qtjzfccopy tr, #qtjzfccopy td, #qtjzfccopy th {
  border-style: none;
}

#qtjzfccopy p {
  margin: 0;
  padding: 0;
}

#qtjzfccopy .gt_table {
  display: table;
  border-collapse: collapse;
  line-height: normal;
  margin-left: auto;
  margin-right: auto;
  color: #333333;
  font-size: 16px;
  font-weight: normal;
  font-style: normal;
  background-color: #FFFFFF;
  width: auto;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #A8A8A8;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #A8A8A8;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
}

#qtjzfccopy .gt_caption {
  padding-top: 4px;
  padding-bottom: 4px;
}

#qtjzfccopy .gt_title {
  color: #333333;
  font-size: 125%;
  font-weight: initial;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-color: #FFFFFF;
  border-bottom-width: 0;
}

#qtjzfccopy .gt_subtitle {
  color: #333333;
  font-size: 85%;
  font-weight: initial;
  padding-top: 3px;
  padding-bottom: 5px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-color: #FFFFFF;
  border-top-width: 0;
}

#qtjzfccopy .gt_heading {
  background-color: #FFFFFF;
  text-align: center;
  border-bottom-color: #FFFFFF;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#qtjzfccopy .gt_bottom_border {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#qtjzfccopy .gt_col_headings {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
}

#qtjzfccopy .gt_col_heading {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 6px;
  padding-left: 5px;
  padding-right: 5px;
  overflow-x: hidden;
}

#qtjzfccopy .gt_column_spanner_outer {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: normal;
  text-transform: inherit;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 4px;
  padding-right: 4px;
}

#qtjzfccopy .gt_column_spanner_outer:first-child {
  padding-left: 0;
}

#qtjzfccopy .gt_column_spanner_outer:last-child {
  padding-right: 0;
}

#qtjzfccopy .gt_column_spanner {
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: bottom;
  padding-top: 5px;
  padding-bottom: 5px;
  overflow-x: hidden;
  display: inline-block;
  width: 100%;
}

#qtjzfccopy .gt_spanner_row {
  border-bottom-style: hidden;
}

#qtjzfccopy .gt_group_heading {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  text-align: left;
}

#qtjzfccopy .gt_empty_group_heading {
  padding: 0.5px;
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  vertical-align: middle;
}

#qtjzfccopy .gt_from_md > :first-child {
  margin-top: 0;
}

#qtjzfccopy .gt_from_md > :last-child {
  margin-bottom: 0;
}

#qtjzfccopy .gt_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  margin: 10px;
  border-top-style: solid;
  border-top-width: 1px;
  border-top-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 1px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 1px;
  border-right-color: #D3D3D3;
  vertical-align: middle;
  overflow-x: hidden;
}

#qtjzfccopy .gt_stub {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
}

#qtjzfccopy .gt_stub_row_group {
  color: #333333;
  background-color: #FFFFFF;
  font-size: 100%;
  font-weight: initial;
  text-transform: inherit;
  border-right-style: solid;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
  padding-left: 5px;
  padding-right: 5px;
  vertical-align: top;
}

#qtjzfccopy .gt_row_group_first td {
  border-top-width: 2px;
}

#qtjzfccopy .gt_row_group_first th {
  border-top-width: 2px;
}

#qtjzfccopy .gt_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#qtjzfccopy .gt_first_summary_row {
  border-top-style: solid;
  border-top-color: #D3D3D3;
}

#qtjzfccopy .gt_first_summary_row.thick {
  border-top-width: 2px;
}

#qtjzfccopy .gt_last_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#qtjzfccopy .gt_grand_summary_row {
  color: #333333;
  background-color: #FFFFFF;
  text-transform: inherit;
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
}

#qtjzfccopy .gt_first_grand_summary_row {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-top-style: double;
  border-top-width: 6px;
  border-top-color: #D3D3D3;
}

#qtjzfccopy .gt_last_grand_summary_row_top {
  padding-top: 8px;
  padding-bottom: 8px;
  padding-left: 5px;
  padding-right: 5px;
  border-bottom-style: double;
  border-bottom-width: 6px;
  border-bottom-color: #D3D3D3;
}

#qtjzfccopy .gt_striped {
  background-color: rgba(128, 128, 128, 0.05);
}

#qtjzfccopy .gt_table_body {
  border-top-style: solid;
  border-top-width: 2px;
  border-top-color: #D3D3D3;
  border-bottom-style: solid;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
}

#qtjzfccopy .gt_footnotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#qtjzfccopy .gt_footnote {
  margin: 0px;
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#qtjzfccopy .gt_sourcenotes {
  color: #333333;
  background-color: #FFFFFF;
  border-bottom-style: none;
  border-bottom-width: 2px;
  border-bottom-color: #D3D3D3;
  border-left-style: none;
  border-left-width: 2px;
  border-left-color: #D3D3D3;
  border-right-style: none;
  border-right-width: 2px;
  border-right-color: #D3D3D3;
}

#qtjzfccopy .gt_sourcenote {
  font-size: 90%;
  padding-top: 4px;
  padding-bottom: 4px;
  padding-left: 5px;
  padding-right: 5px;
}

#qtjzfccopy .gt_left {
  text-align: left;
}

#qtjzfccopy .gt_center {
  text-align: center;
}

#qtjzfccopy .gt_right {
  text-align: right;
  font-variant-numeric: tabular-nums;
}

#qtjzfccopy .gt_font_normal {
  font-weight: normal;
}

#qtjzfccopy .gt_font_bold {
  font-weight: bold;
}

#qtjzfccopy .gt_font_italic {
  font-style: italic;
}

#qtjzfccopy .gt_super {
  font-size: 65%;
}

#qtjzfccopy .gt_footnote_marks {
  font-size: 75%;
  vertical-align: 0.4em;
  position: initial;
}

#qtjzfccopy .gt_asterisk {
  font-size: 100%;
  vertical-align: 0;
}

#qtjzfccopy .gt_indent_1 {
  text-indent: 5px;
}

#qtjzfccopy .gt_indent_2 {
  text-indent: 10px;
}

#qtjzfccopy .gt_indent_3 {
  text-indent: 15px;
}

#qtjzfccopy .gt_indent_4 {
  text-indent: 20px;
}

#qtjzfccopy .gt_indent_5 {
  text-indent: 25px;
}
</style>
<table class="table gt_table" data-quarto-disable-processing="false" data-quarto-bootstrap="false">
<thead><tr class="gt_col_headings">
<th class="gt_col_heading gt_columns_bottom_border gt_left" rowspan="1" colspan="1" scope="col" id="&lt;strong&gt;Characteristic&lt;/strong&gt;"><strong>Characteristic</strong></th>
      <th class="gt_col_heading gt_columns_bottom_border gt_center" rowspan="1" colspan="1" scope="col" id='&lt;strong&gt;N = 547&lt;/strong&gt;&lt;span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;"&gt;&lt;sup&gt;1&lt;/sup&gt;&lt;/span&gt;'>
<strong>N = 547</strong><span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;"><sup>1</sup></span>
</th>
    </tr></thead>
<tbody class="gt_table_body">
<tr>
<td headers="label" class="gt_row gt_left">TP53</td>
<td headers="stat_0" class="gt_row gt_center">301 (55%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">EGFR</td>
<td headers="stat_0" class="gt_row gt_center">168 (31%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">KRAS</td>
<td headers="stat_0" class="gt_row gt_center">167 (31%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">LRP1B</td>
<td headers="stat_0" class="gt_row gt_center">5 (20%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">522</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">STK11</td>
<td headers="stat_0" class="gt_row gt_center">71 (15%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">67</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">KEAP1</td>
<td headers="stat_0" class="gt_row gt_center">65 (14%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">69</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">GRM3</td>
<td headers="stat_0" class="gt_row gt_center">3 (13%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">523</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">SLIT2</td>
<td headers="stat_0" class="gt_row gt_center">3 (13%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">523</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">SPTA1</td>
<td headers="stat_0" class="gt_row gt_center">3 (13%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">523</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">RANBP2</td>
<td headers="stat_0" class="gt_row gt_center">3 (13%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">523</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">KMT2D</td>
<td headers="stat_0" class="gt_row gt_center">53 (11%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">67</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">CDKN2A.Del</td>
<td headers="stat_0" class="gt_row gt_center">52 (11%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">67</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">GLI3</td>
<td headers="stat_0" class="gt_row gt_center">11 (11%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">444</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">PRKDC</td>
<td headers="stat_0" class="gt_row gt_center">22 (11%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">339</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">SMARCA4</td>
<td headers="stat_0" class="gt_row gt_center">48 (10%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">67</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">COL7A1</td>
<td headers="stat_0" class="gt_row gt_center">8 (10%)</td>
</tr>
<tr>
<td headers="label" class="gt_row gt_left">    Unknown</td>
<td headers="stat_0" class="gt_row gt_center">467</td>
</tr>
</tbody>
<tfoot class="gt_footnotes"><tr>
<td class="gt_footnote" colspan="2">
<span class="gt_footnote_marks" style="white-space:nowrap;font-style:italic;font-weight:normal;"><sup>1</sup></span> n (%)</td>
    </tr></tfoot>
</table>
</div>
</div>
<div class="section level4">
<h4 id="data-visualizations">Data Visualizations<a class="anchor" aria-label="anchor" href="#data-visualizations"></a>
</h4>
<p>We can use the <code><a href="../reference/mutation_viz.html">mutation_viz()</a></code> function to visualize
several aspects of the mutation data, including variant classification,
variant type, SNV class and top variant genes.</p>
<p>For the purposes of this vignette we will visualize the genomic data
from cBioPortal.</p>
<p><strong>Using the genomic data from cBioPortal:</strong></p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mutation_viz_gen_dat_cbio</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/mutation_viz.html">mutation_viz</a></span><span class="op">(</span><span class="va">mutations_extended_2_0</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #BBBB00;">!</span> `samples` argument is `NULL`. We will infer your cohort inclusion and resulting data frame will include all samples with at least one alteration in <span style="color: #00BB00;">mutation</span>, <span style="color: #00BB00;">fusion</span> or <span style="color: #00BB00;">cna</span> data frames</span></span>
<span><span class="co">#&gt; Warning: To ensure gene with multiple names/aliases are correctly grouped together, the</span></span>
<span><span class="co">#&gt; following genes in your dataframe have been recoded (if you are running</span></span>
<span><span class="co">#&gt; `create_gene_binary()` you can prevent this with `alias_table = FALSE`):</span></span>
<span><span class="co">#&gt; HIST3H3 recoded to H3-4</span></span>
<span><span class="co">#&gt; HIST1H3B recoded to H3C2</span></span>
<span><span class="co">#&gt; HIST1H3I recoded to H3C11</span></span>
<span><span class="co">#&gt; HIST1H3D recoded to H3C4</span></span>
<span><span class="co">#&gt; HIST1H3F recoded to H3C7</span></span>
<span><span class="co">#&gt; HIST1H3C recoded to H3C3</span></span>
<span><span class="co">#&gt; HIST1H3G recoded to H3C8</span></span>
<span><span class="co">#&gt; H3F3C recoded to H3-5</span></span>
<span><span class="co">#&gt; HIST1H3E recoded to H3C6</span></span>
<span><span class="co">#&gt; HIST1H1C recoded to H1-2</span></span>
<span><span class="co">#&gt; H3F3B recoded to H3-3B</span></span>
<span><span class="co">#&gt; HIST1H3J recoded to H3C12</span></span>
<span><span class="co">#&gt; HIST1H2BD recoded to H2BC5</span></span>
<span><span class="co">#&gt; H3F3A recoded to H3-3A</span></span>
<span><span class="co">#&gt; HIST1H3H recoded to H3C10</span></span>
<span></span>
<span><span class="va">mutation_viz_gen_dat_cbio</span></span>
<span><span class="co">#&gt; $varclass</span></span></code></pre></div>
<p><img src="genie-bpc-vignette_files/figure-html/unnamed-chunk-18-1.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $vartype</span></span></code></pre>
<p><img src="genie-bpc-vignette_files/figure-html/unnamed-chunk-18-2.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $samplevar</span></span></code></pre>
<p><img src="genie-bpc-vignette_files/figure-html/unnamed-chunk-18-3.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $topgenes</span></span></code></pre>
<p><img src="genie-bpc-vignette_files/figure-html/unnamed-chunk-18-4.png" width="700"></p>
<pre><code><span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; $genecor</span></span></code></pre>
<p><img src="genie-bpc-vignette_files/figure-html/unnamed-chunk-18-5.png" width="700"></p>
</div>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Additional details regarding the GENIE BPC data and the {genieBPC} R
package are published in the following papers:</p>
<ul>
<li><p>Lavery, J. A., Brown, S., Curry, M. A., Martin, A., Sjoberg, D.
D., &amp; Whiting, K. (2023). <a href="https://pubmed.ncbi.nlm.nih.gov/36519837/" class="external-link">A data processing
pipeline for the AACR project GENIE biopharma collaborative data with
the {genieBPC} R package</a>. Bioinformatics (Oxford, England), 39(1),
btac796. <a href="https://doi.org/10.1093/bioinformatics/btac796" class="external-link uri">https://doi.org/10.1093/bioinformatics/btac796</a></p></li>
<li><p>Lavery, J. A., Lepisto, E. M., Brown, S., Rizvi, H., McCarthy,
C., LeNoue-Newton, M., Yu, C., Lee, J., Guo, X., Yu, T., Rudolph, J.,
Sweeney, S., AACR Project GENIE Consortium, Park, B. H., Warner, J. L.,
Bedard, P. L., Riely, G., Schrag, D., &amp; Panageas, K. S. (2022). <a href="https://pubmed.ncbi.nlm.nih.gov/35192403/" class="external-link">A Scalable Quality
Assurance Process for Curating Oncology Electronic Health Records: The
Project GENIE Biopharma Collaborative Approach</a>. JCO clinical cancer
informatics, 6, e2100105. <a href="https://doi.org/10.1200/CCI.21.00105" class="external-link uri">https://doi.org/10.1200/CCI.21.00105</a></p></li>
</ul>
<p>Technical details regarding proper analysis of this data can be found
in the following publication:</p>
<ul>
<li><p>Brown, S., Lavery, J. A., Shen, R., Martin, A. S., Kehl, K. L.,
Sweeney, S. M., Lepisto, E. M., Rizvi, H., McCarthy, C. G., Schultz, N.,
Warner, J. L., Park, B. H., Bedard, P. L., Riely, G. J., Schrag, D.,
Panageas, K. S., &amp; AACR Project GENIE Consortium (2022). <a href="https://pubmed.ncbi.nlm.nih.gov/34734967/" class="external-link">Implications of
Selection Bias Due to Delayed Study Entry in Clinical Genomic
Studies</a>. JAMA oncology, 8(2), 287–291. <a href="https://doi.org/10.1001/jamaoncol.2021.5153" class="external-link uri">https://doi.org/10.1001/jamaoncol.2021.5153</a></p></li>
<li><p>Kehl, K. L., Uno, H., Gusev, A., Groha, S., Brown, S., Lavery, J.
A., Schrag, D., &amp; Panageas, K. S. (2023). <a href="https://pubmed.ncbi.nlm.nih.gov/36626408/" class="external-link">Elucidating Analytic
Bias Due to Informative Cohort Entry in Cancer Clinico-genomic
Datasets</a>. Cancer epidemiology, biomarkers &amp; prevention: a
publication of the American Association for Cancer Research, cosponsored
by the American Society of Preventive Oncology, 32(3), 344–352. <a href="https://doi.org/10.1158/1055-9965.EPI-22-0875" class="external-link uri">https://doi.org/10.1158/1055-9965.EPI-22-0875</a></p></li>
<li><p>Kehl, K. L., Riely, G. J., Lepisto, E. M., Lavery, J. A., Warner,
J. L., LeNoue-Newton, M. L., Sweeney, S. M., Rudolph, J. E., Brown, S.,
Yu, C., Bedard, P. L., Schrag, D., Panageas, K. S., &amp; American
Association of Cancer Research (AACR) Project Genomics Evidence
Neoplasia Information Exchange (GENIE) Consortium (2021). <a href="https://pubmed.ncbi.nlm.nih.gov/34309669/" class="external-link">Correlation Between
Surrogate End Points and Overall Survival in a Multi-institutional
Clinicogenomic Cohort of Patients With Non-Small Cell Lung or Colorectal
Cancer</a>. JAMA network open, 4(7), e2117547. <a href="https://doi.org/10.1001/jamanetworkopen.2021.17547" class="external-link uri">https://doi.org/10.1001/jamanetworkopen.2021.17547</a></p></li>
</ul>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by <a href="https://www.karissawhiting.com/" class="external-link">Karissa Whiting</a>, Michael Curry, <a href="https://github.com/hfuchs5/" class="external-link">Hannah Fuchs</a>, Axel Martin, Arshi Arora.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
